{
  "cells": [
    {
      "cell_type": "markdown",
      "id": "d429ccc6",
      "metadata": {
        "id": "d429ccc6"
      },
      "source": [
        "# About the Dataset\n",
        "\n",
        "The original CORD-19 is a resource of over 1,000,000 scholarly articles, including over 400,000 with full text, about COVID-19, SARS-CoV-2, and related coronaviruses.\n",
        "\n",
        "In our project, the dataset is sampled from the CORD-19 with size ~10,000 to reduce computation burden."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "id": "UFYXtLsHeNX0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UFYXtLsHeNX0",
        "outputId": "08825601-cf94-481a-e6b6-336a0132a717"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n",
            "/content\n",
            "__MACOSX  meta_10k.csv\tsubset\tsubset.zip\n",
            "Directory /content/CORD_19_subset already exists. Skipping extraction.\n"
          ]
        }
      ],
      "source": [
        "import os\n",
        "import subprocess\n",
        "\n",
        "\n",
        "# shared link: https://drive.google.com/drive/folders/1Td_ZTUVrsKeftDE5Zll7252YLJdWiNTk?usp=share_link \n",
        "# you can download the data via the shared link, and skip Step 0 and Step 1 if you want to run the code in your local machine \n",
        "\n",
        "\n",
        "# Step 0: add the shared folder to your google drive. e.g., /content/drive/MyDrive/CORD_19\n",
        "\n",
        "# Step 1: Mount Google Drive\n",
        "from google.colab import drive\n",
        "drive.mount(\"/content/drive\")\n",
        "\n",
        "\n",
        "!echo $PWD\n",
        "\n",
        "!ls /content/drive/MyDrive/CORD_19/\n",
        "\n",
        "# Step 2: unzip json files \n",
        "subset_dir = os.path.join(os.getcwd(),  \"CORD_19_subset\")\n",
        "\n",
        "\n",
        "zip_file_path=\"/content/drive/MyDrive/CORD_19/subset.zip\"\n",
        "\n",
        "# Check if the destination directory exists\n",
        "if not os.path.exists(subset_dir):\n",
        "    # Unzip the file\n",
        "    cmd = \"unzip {} -d {}\".format(zip_file_path, subset_dir)\n",
        "    proc = subprocess.Popen(cmd, shell=True)\n",
        "else:\n",
        "    print(f\"Directory {subset_dir} already exists. Skipping extraction.\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "id": "SxWWPxCDjHF5",
      "metadata": {
        "id": "SxWWPxCDjHF5"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "id": "eeabd1eb",
      "metadata": {
        "id": "eeabd1eb"
      },
      "outputs": [],
      "source": [
        "# import packages\n",
        "\n",
        "\n",
        "import os\n",
        "import json\n",
        "import glob\n",
        "import numpy as np\n",
        "import pandas as pd\n",
        "\n",
        "import matplotlib.pyplot as plt\n"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "cc1250a1",
      "metadata": {
        "id": "cc1250a1"
      },
      "source": [
        "# Data Load & Pre-processing"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "id": "83d2222c",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 695
        },
        "id": "83d2222c",
        "outputId": "d3082464-4669-4404-b139-0300e50112d4"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "9022\n"
          ]
        },
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"meta_df\",\n  \"rows\": 9022,\n  \"fields\": [\n    {\n      \"column\": \"Unnamed: 0\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 337241,\n        \"min\": 44,\n        \"max\": 1056551,\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          811225,\n          732051,\n          763638\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cord_uid\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"h3uaj6fx\",\n          \"yz1bshv2\",\n          \"p630d49d\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sha\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"e031a4fcf00577dcc7584f7a456f83394b61ec1b\",\n          \"9ce761ee59fd4b5bc7fda2d3c51b89c6be54360b\",\n          \"e114858a40b22f7ff3ac74ab9b97cecc0ee36761\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"source_x\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 21,\n        \"samples\": [\n          \"PMC\",\n          \"BioRxiv; Medline; WHO\",\n          \"ArXiv; Medline\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"Promise and peril-defining ethical telehealth practice from the clinician and patient perspective: A qualitative study\",\n          \"DeSUMOylation of Apoptosis Inhibitor 5 by Avibirnavirus VP3 Supports Virus Replication\",\n          \"Emergent procedures for oropharyngeal bleeding during the COVID-19 pandemic: Protection of medical staff\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"doi\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8599,\n        \"samples\": [\n          \"10.1038/s41598-021-91817-9\",\n          \"10.5334/gh.786\",\n          \"10.1038/s41578-021-00305-z\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pmcid\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7942,\n        \"samples\": [\n          \"PMC3573664\",\n          \"PMC7789084\",\n          \"PMC7729151\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pubmed_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 6986,\n        \"samples\": [\n          \"35165656\",\n          \"35110546\",\n          \"33538335\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"license\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 13,\n        \"samples\": [\n          \"els-covid\",\n          \"medrxiv\",\n          \"cc-by\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8285,\n        \"samples\": [\n          \"The variety and extent of non-pharmaceutical measures implemented by the government to control COVID-19 in Argentina were exceptional, making this country the best example to analyze the evolution of COVID-19 under the most stringent and longerlasting restrictive policies- which up to May 31(st) 2021 included 119 days of strict nation-wide lock-down, 304 days of less restrictive lock-downs, and 35 days of curfews. Two of the three peaks of infection correlated with the germicidal solar flux received in Argentina, suggesting a seasonal component and a role for the virus persisting in the environment. A massive public gathering crowding the presidential square in Buenos Aires, during which nearly half of those present were without face masks, did not alter the infection curve in that city. Comparative epidemiological data (standardized per million inhabitants) shows that COVID-19 in Uruguay, a neighboring country whose capital is at a similar latitude than Buenos Aires and who did not mandate lock-downs or curfews, progressed much slower (until vaccination started) than in Argentina. The number of yearly deaths caused by respiratory diseases and influenza in Argentina before the pandemic was similar to the total number of deaths attributed to COVID-19 cumulated on April 25, 2021, more than a year after the pandemic started. The failure to detect any benefit on ameliorating COVID-19 by the long and strict nation-wide lock-downs in Argentina should raise world-wide concerns about mandating costly and ineffective restrictive measures during ongoing or future pandemics.\",\n          \"The world is currently facing a serious coronavirus pandemic (COVID-19). This novel coronavirus is a new virus for which effective drugs and vaccines have not yet been developed. Amateur and professional athletic events around the world have been stopped, and teams and athletes have had to adopt social withdrawal measures, interrupting their training and preparation routines for competitions. Given this alarming scenario, this article aims to alert athletes to the importance of maintaining a conditioning routine during this confinement period.\",\n          \"Bivalve molluscan shellfish like clams and oysters, etc., are capable to bioaccumulate surrounding contaminants from waters into their digestive systems and posing serious threats of food poisoning. Detection of rotaviruses (RVs) in shellfish is of particular importance because RVs are prone to genome reassortment resulting in the emergence of new RV variants that may compromise vaccine safety. Herein, we have detected the wild-type RVs and Rotarix/RotaTeq vaccine strains in freshwater clams collected on the riverside, Kawasaki city, from July 2019 to January 2020 and correlated the detected genotypes with that of gastroenteritis cases of nearby clinics to understand the transmission of RVs in the environment. The wild-type RVs were detected in 62 (64.6%) out of 96 freshwater clams in every study month: July, September, November, and January that are considered as off-season for RV infections. The most frequent genotypes were G2 (42.9%), G8 (28.6%), G3 (14.3%), G1 (7.1%), and G10 (7.1%), which remained comparable with genotypic distribution found in the clinical samples over the last few years indicating that these RVs may accumulate in clams since a long time. However, G10 genotype was detected in clam but not in clinical samples suggesting the presence of asymptomatic infection or RVs could be carried out from a long distance. Importantly, vaccine strains, RotaTeq (1%) but not Rotarix (0%), were also detected in a clam. Attention must be paid to monitoring the potential transmission of wild-type and vaccine RV strains in the environment to prevent the emergence of new variants generated from genome reassortment with vaccine strains.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"publish_time\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1864,\n        \"samples\": [\n          \"2020-01-02\",\n          \"2022-03-09\",\n          \"2016-04-04\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8997,\n        \"samples\": [\n          \"Marchetti, Federico; Verazza, Sara; Brambilla, Margherita; Restivo, Vincenzo\",\n          \"Kaeberlein, Matt; Tyler, Jessica K\",\n          \"Atlante, Sandra; Mongelli, Alessia; Barbi, Veronica; Martelli, Fabio; Farsetti, Antonella; Gaetano, Carlo\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"journal\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2785,\n        \"samples\": [\n          \"Adv Med\",\n          \"Microbial Forensics\",\n          \"Appl Biochem Biotechnol\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"mag_id\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"who_covidence_id\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"arxiv_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 480,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pdf_json_files\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pmc_json_files\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7414,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"s2_id\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 61913642.79275464,\n        \"min\": 15600.0,\n        \"max\": 249148763.0,\n        \"num_unique_values\": 7468,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe",
              "variable_name": "meta_df"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-fa7d4e5e-6759-4db6-a1ce-3c338f71b6e5\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Unnamed: 0</th>\n",
              "      <th>cord_uid</th>\n",
              "      <th>sha</th>\n",
              "      <th>source_x</th>\n",
              "      <th>title</th>\n",
              "      <th>doi</th>\n",
              "      <th>pmcid</th>\n",
              "      <th>pubmed_id</th>\n",
              "      <th>license</th>\n",
              "      <th>abstract</th>\n",
              "      <th>publish_time</th>\n",
              "      <th>authors</th>\n",
              "      <th>journal</th>\n",
              "      <th>mag_id</th>\n",
              "      <th>who_covidence_id</th>\n",
              "      <th>arxiv_id</th>\n",
              "      <th>pdf_json_files</th>\n",
              "      <th>pmc_json_files</th>\n",
              "      <th>url</th>\n",
              "      <th>s2_id</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>44</td>\n",
              "      <td>m71xkuo9</td>\n",
              "      <td>c6bf372c094f035a514975c35a7f9c094abbe493</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Sequence specific visual detection of LAMP rea...</td>\n",
              "      <td>10.1186/1472-6750-6-3</td>\n",
              "      <td>PMC1373654</td>\n",
              "      <td>16401354</td>\n",
              "      <td>cc-by</td>\n",
              "      <td>BACKGROUND: Development of a practical gene po...</td>\n",
              "      <td>2006-01-10</td>\n",
              "      <td>Mori, Yasuyoshi; Hirano, Tsuyoshi; Notomi, Tsu...</td>\n",
              "      <td>BMC Biotechnol</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/c6bf372c094f035a51497...</td>\n",
              "      <td>document_parses/pmc_json/PMC1373654.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>96</td>\n",
              "      <td>hqc7u9w3</td>\n",
              "      <td>c65f0939cf35a0f04bf93bd6e8f771b8521563a5</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Transmission Parameters of the 2001 Foot and M...</td>\n",
              "      <td>10.1371/journal.pone.0000502</td>\n",
              "      <td>PMC1876810</td>\n",
              "      <td>17551582</td>\n",
              "      <td>cc-by</td>\n",
              "      <td>Despite intensive ongoing research, key aspect...</td>\n",
              "      <td>2007-06-06</td>\n",
              "      <td>Chis Ster, Irina; Ferguson, Neil M.</td>\n",
              "      <td>PLoS One</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/c65f0939cf35a0f04bf93...</td>\n",
              "      <td>document_parses/pmc_json/PMC1876810.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>217</td>\n",
              "      <td>jzwcy7dr</td>\n",
              "      <td>a009c8efa4c5f13a5e604608d4e33e1dac078044</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Results From a Hypothesis Generating Case-Cont...</td>\n",
              "      <td>10.1093/schbul/sbm139</td>\n",
              "      <td>PMC2632504</td>\n",
              "      <td>18156638</td>\n",
              "      <td>bronze-oa</td>\n",
              "      <td>Background: Herpes family viruses can cause ce...</td>\n",
              "      <td>2008-08-20</td>\n",
              "      <td>Niebuhr, David W.; Millikan, Amy M.; Yolken, R...</td>\n",
              "      <td>Schizophrenia Bulletin</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/a009c8efa4c5f13a5e604...</td>\n",
              "      <td>document_parses/pmc_json/PMC2632504.xml.json</td>\n",
              "      <td>https://academic.oup.com/schizophreniabulletin...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>255</td>\n",
              "      <td>02opdk0m</td>\n",
              "      <td>b411e12b20d883ef2ee5ca19d48eff9fccedf05f</td>\n",
              "      <td>PMC</td>\n",
              "      <td>CVTree update: a newly designed phylogenetic s...</td>\n",
              "      <td>10.1093/nar/gkp278</td>\n",
              "      <td>PMC2703908</td>\n",
              "      <td>19398429</td>\n",
              "      <td>cc-by-nc</td>\n",
              "      <td>The CVTree web server (http://tlife.fudan.edu....</td>\n",
              "      <td>2009-07-01</td>\n",
              "      <td>Xu, Zhao; Hao, Bailin</td>\n",
              "      <td>Nucleic Acids Res</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/b411e12b20d883ef2ee5c...</td>\n",
              "      <td>document_parses/pmc_json/PMC2703908.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>342</td>\n",
              "      <td>094d0rn6</td>\n",
              "      <td>513d5ea4db4eb8e94c14c46b018c6041d78119cf</td>\n",
              "      <td>PMC</td>\n",
              "      <td>IPS-1 Is Essential for the Control of West Nil...</td>\n",
              "      <td>10.1371/journal.ppat.1000757</td>\n",
              "      <td>PMC2816698</td>\n",
              "      <td>20140199</td>\n",
              "      <td>cc-by</td>\n",
              "      <td>The innate immune response is essential for co...</td>\n",
              "      <td>2010-02-05</td>\n",
              "      <td>Suthar, Mehul S.; Ma, Daphne Y.; Thomas, Sunil...</td>\n",
              "      <td>PLoS Pathog</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/513d5ea4db4eb8e94c14c...</td>\n",
              "      <td>document_parses/pmc_json/PMC2816698.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-fa7d4e5e-6759-4db6-a1ce-3c338f71b6e5')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-fa7d4e5e-6759-4db6-a1ce-3c338f71b6e5 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-fa7d4e5e-6759-4db6-a1ce-3c338f71b6e5');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-a8c770c4-a773-4aab-8914-481134d2babc\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-a8c770c4-a773-4aab-8914-481134d2babc')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-a8c770c4-a773-4aab-8914-481134d2babc button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "   Unnamed: 0  cord_uid                                       sha source_x  \\\n",
              "0          44  m71xkuo9  c6bf372c094f035a514975c35a7f9c094abbe493      PMC   \n",
              "1          96  hqc7u9w3  c65f0939cf35a0f04bf93bd6e8f771b8521563a5      PMC   \n",
              "2         217  jzwcy7dr  a009c8efa4c5f13a5e604608d4e33e1dac078044      PMC   \n",
              "3         255  02opdk0m  b411e12b20d883ef2ee5ca19d48eff9fccedf05f      PMC   \n",
              "4         342  094d0rn6  513d5ea4db4eb8e94c14c46b018c6041d78119cf      PMC   \n",
              "\n",
              "                                               title  \\\n",
              "0  Sequence specific visual detection of LAMP rea...   \n",
              "1  Transmission Parameters of the 2001 Foot and M...   \n",
              "2  Results From a Hypothesis Generating Case-Cont...   \n",
              "3  CVTree update: a newly designed phylogenetic s...   \n",
              "4  IPS-1 Is Essential for the Control of West Nil...   \n",
              "\n",
              "                            doi       pmcid pubmed_id    license  \\\n",
              "0         10.1186/1472-6750-6-3  PMC1373654  16401354      cc-by   \n",
              "1  10.1371/journal.pone.0000502  PMC1876810  17551582      cc-by   \n",
              "2         10.1093/schbul/sbm139  PMC2632504  18156638  bronze-oa   \n",
              "3            10.1093/nar/gkp278  PMC2703908  19398429   cc-by-nc   \n",
              "4  10.1371/journal.ppat.1000757  PMC2816698  20140199      cc-by   \n",
              "\n",
              "                                            abstract publish_time  \\\n",
              "0  BACKGROUND: Development of a practical gene po...   2006-01-10   \n",
              "1  Despite intensive ongoing research, key aspect...   2007-06-06   \n",
              "2  Background: Herpes family viruses can cause ce...   2008-08-20   \n",
              "3  The CVTree web server (http://tlife.fudan.edu....   2009-07-01   \n",
              "4  The innate immune response is essential for co...   2010-02-05   \n",
              "\n",
              "                                             authors                 journal  \\\n",
              "0  Mori, Yasuyoshi; Hirano, Tsuyoshi; Notomi, Tsu...          BMC Biotechnol   \n",
              "1                Chis Ster, Irina; Ferguson, Neil M.                PLoS One   \n",
              "2  Niebuhr, David W.; Millikan, Amy M.; Yolken, R...  Schizophrenia Bulletin   \n",
              "3                              Xu, Zhao; Hao, Bailin       Nucleic Acids Res   \n",
              "4  Suthar, Mehul S.; Ma, Daphne Y.; Thomas, Sunil...             PLoS Pathog   \n",
              "\n",
              "   mag_id  who_covidence_id arxiv_id  \\\n",
              "0     NaN               NaN      NaN   \n",
              "1     NaN               NaN      NaN   \n",
              "2     NaN               NaN      NaN   \n",
              "3     NaN               NaN      NaN   \n",
              "4     NaN               NaN      NaN   \n",
              "\n",
              "                                      pdf_json_files  \\\n",
              "0  document_parses/pdf_json/c6bf372c094f035a51497...   \n",
              "1  document_parses/pdf_json/c65f0939cf35a0f04bf93...   \n",
              "2  document_parses/pdf_json/a009c8efa4c5f13a5e604...   \n",
              "3  document_parses/pdf_json/b411e12b20d883ef2ee5c...   \n",
              "4  document_parses/pdf_json/513d5ea4db4eb8e94c14c...   \n",
              "\n",
              "                                 pmc_json_files  \\\n",
              "0  document_parses/pmc_json/PMC1373654.xml.json   \n",
              "1  document_parses/pmc_json/PMC1876810.xml.json   \n",
              "2  document_parses/pmc_json/PMC2632504.xml.json   \n",
              "3  document_parses/pmc_json/PMC2703908.xml.json   \n",
              "4  document_parses/pmc_json/PMC2816698.xml.json   \n",
              "\n",
              "                                                 url  s2_id  \n",
              "0  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...    NaN  \n",
              "1  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...    NaN  \n",
              "2  https://academic.oup.com/schizophreniabulletin...    NaN  \n",
              "3  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...    NaN  \n",
              "4  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...    NaN  "
            ]
          },
          "execution_count": 3,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# Load Meta data from meta_10k.csv\n",
        "data_root = '/content/drive/MyDrive/CORD_19/'\n",
        "\n",
        "metadata_path = os.path.join(data_root, 'meta_10k.csv')\n",
        "meta_df = pd.read_csv(metadata_path, dtype={\n",
        "    'pubmed_id': str,\n",
        "    'Microsoft Academic Paper ID': str,\n",
        "    'doi': str\n",
        "})\n",
        "\n",
        "print(len(meta_df))\n",
        "meta_df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "id": "e1b611f0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "e1b611f0",
        "outputId": "30b26c96-07b0-4490-e5c8-b09a73d75945"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 9022 entries, 0 to 9021\n",
            "Data columns (total 20 columns):\n",
            " #   Column            Non-Null Count  Dtype  \n",
            "---  ------            --------------  -----  \n",
            " 0   Unnamed: 0        9022 non-null   int64  \n",
            " 1   cord_uid          9022 non-null   object \n",
            " 2   sha               9022 non-null   object \n",
            " 3   source_x          9022 non-null   object \n",
            " 4   title             9022 non-null   object \n",
            " 5   doi               8599 non-null   object \n",
            " 6   pmcid             7942 non-null   object \n",
            " 7   pubmed_id         6986 non-null   object \n",
            " 8   license           9022 non-null   object \n",
            " 9   abstract          8289 non-null   object \n",
            " 10  publish_time      9022 non-null   object \n",
            " 11  authors           9009 non-null   object \n",
            " 12  journal           8095 non-null   object \n",
            " 13  mag_id            0 non-null      float64\n",
            " 14  who_covidence_id  0 non-null      float64\n",
            " 15  arxiv_id          480 non-null    object \n",
            " 16  pdf_json_files    9022 non-null   object \n",
            " 17  pmc_json_files    7414 non-null   object \n",
            " 18  url               9022 non-null   object \n",
            " 19  s2_id             7469 non-null   float64\n",
            "dtypes: float64(3), int64(1), object(16)\n",
            "memory usage: 1.4+ MB\n"
          ]
        }
      ],
      "source": [
        "meta_df.info()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "id": "c3139be0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c3139be0",
        "outputId": "5cdaf76c-167c-4a56-b786-2838a18700f1"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/CORD_19_subset/subset/document_parses/pdf_json/\n",
            "total json files: 12000\n"
          ]
        }
      ],
      "source": [
        "def glob_files(path, f_type=\".json\"):\n",
        "    dst = []\n",
        "    for root, _, files in os.walk(path):\n",
        "        for f in files:\n",
        "            if f.endswith(f_type):\n",
        "                dst.append(os.path.join(root, f))\n",
        "    return dst\n",
        "\n",
        "# glob json files\n",
        "json_dir = os.path.join(subset_dir, \"subset/document_parses/pdf_json/\")\n",
        "print(json_dir)\n",
        "json_files = glob_files(json_dir, \".json\")\n",
        "\n",
        "print(\"total json files:\", len(json_files))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "id": "20fb57d2",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "20fb57d2",
        "outputId": "03c330eb-ba5a-4a05-81f6-420b007e3d43"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "4abd0c9b745b665cd9677a3de0d7e8b9a38c0f23: Stress, physical activity, and screen-related sedentary behaviour within the first month of the COVID-19 pandemic : This study investigated how stress, physical activity and sedentary behaviours, of a small sample of Canadians, changed within the first month (i.e. March/April) of the COVID-19 pandemic and the reaso... ity trackers were recruited via social media. Participants (N = 121) completed fillable calendars (March/April 2020) with their step counts and answered an online survey. Separate paired-sample t-test...\n"
          ]
        }
      ],
      "source": [
        "class FileReader:\n",
        "    def __init__(self, file_path):\n",
        "        with open(file_path) as file:\n",
        "            content = json.load(file)\n",
        "            self.paper_id = content['paper_id']\n",
        "            self.abstract = []\n",
        "            self.body_text = []\n",
        "            # Abstract\n",
        "            for entry in content['abstract']:\n",
        "                self.abstract.append(entry['text'])\n",
        "            # Body text\n",
        "            for entry in content['body_text']:\n",
        "                self.body_text.append(entry['text'])\n",
        "            self.abstract = '\\n'.join(self.abstract)\n",
        "            self.body_text = '\\n'.join(self.body_text)\n",
        "\n",
        "            self.title = content['metadata']['title']\n",
        "\n",
        "            #dict_keys(['paper_id', 'metadata', 'abstract', 'body_text',\n",
        "            #'bib_entries', 'ref_entries', 'back_matter'])\n",
        "\n",
        "\n",
        "    def __repr__(self):\n",
        "        return f\"{self.paper_id}: {self.title } : {self.abstract[:200]}... {self.body_text[:200]}...\"\n",
        "\n",
        "\n",
        "first_row = FileReader(json_files[0])\n",
        "print(first_row)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "id": "96a78cce",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 398
        },
        "id": "96a78cce",
        "outputId": "db3e15b8-5496-4752-901a-a0e8a39f3628"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "100%|██████████| 12000/12000 [00:59<00:00, 201.18it/s]\n"
          ]
        },
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"df_covid\",\n  \"rows\": 9022,\n  \"fields\": [\n    {\n      \"column\": \"paper_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"c063b2e2ccb6c080cc87cf4ce97cd3809b290dab\",\n          \"0700f7e0900668d65915b997966732ac5b35de05\",\n          \"cf7683ee7ca261c9993cbc51eda2747fac22bb3f\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"doi\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8599,\n        \"samples\": [\n          \"10.1016/j.puhe.2009.11.003\",\n          \"10.1016/j.indmarman.2020.05.018\",\n          \"10.3390/v13112329\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7281,\n        \"samples\": [\n          \"Victor M Corman et al.\",\n          \": The social distancing phase allows us to model COVID-19 in a way when the effect of latent unreported/asymptomatic cases can be reliably observed.\\nAccurate forecasts for COVID-19 are necessary for better preparedness and resource management. Specifically, deciding the response over months or several months requires accurate long-term forecasts which is particularly challenging as the model errors accumulate with time. A critical factor that can hinder accurate longterm forecasts, is the number of unreported/asymptomatic cases. While there have been early serology tests to estimate this number, more tests need to be conducted for more reliable results. To identify the number of unreported/asymptomatic cases, we take an epidemiology data-driven approach. We show that we can identify lower bounds on this ratio or upper bound on actual cases as a factor of reported cases. To do so, we propose an extension of our prior heterogeneous infection rate model, incorporating unreported/asymptomatic cases. We prove that the number of unreported cases can be reliably estimated only from a certain time period of the epidemic data. In doing so, we construct an algorithm called Fixed Infection Rate method, which identifies a reliable bound on the learned ratio. We also propose two heuristics to learn this ratio and show their effectiveness on simulated data. We use our approaches to identify the upper bounds on the ratio of actual to reported cases for New York City and several US states. Our results demonstrate with high confidence that the actual number of cases cannot be more than 35 times in New York, 40 times in Illinois, 38 times in Massachusetts and 29 times in New Jersey, than the reported cases.\",\n          \"In the Netherlands, as in many other nations, the government has proposed the use of a contact-tracing app as a means of helping to contain the spread of the corona virus. The discussion about the use of such an app has mostly been framed in terms of a tradeoff between privacy and public health. This research statement presents an analysis of the Dutch public debate on Corona-apps by using the framework of Orders of Worth by Boltanski and Th\\u00e9venot (1991). It argues that this framework can help us to move beyond the dichotomy of privacy vs. public health by recognizing a plurality of conceptions of the common good in the debate about contact-tracing apps. This statement presents six orders of worth present in the Dutch debate: civic, domestic, vitality, market, industrial and project, and argues that the identification of which common goods are at stake will contribute to discussions about the use of this technology from a standpoint with a richer ethical perspective.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"body_text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"with the low incidence of certain pulmonary conditions. Compassionate use of remdesivir has been shown to reduce time to clinical improvement in adults and has been retrospectively studied in small pediatric cohorts with promising results. There are limited studies in children with respiratory comorbidities, apart from asthma, presumably due to low morbidity of SARS-CoV-2 infection in the general paediatric population along with the low incidence of certain pulmonary conditions. [1] [2] [3] Remdesivir, a broad spectrum antiviral agent against several RNA viruses, has been shown to reduce time to clinical improvement in adults and has been retrospectively studied in small pediatric cohorts with promising results. In anticipation of randomized controlled trials for the pediatric population, an expert consensus suggests the use of antivirals in children on a case-by-case basis. 2, 3 We present a case of COVID-19 infection in a 3-year old toddler with severe postinfectious bronchiolitis obliterans (PIBO), who received compassionate treatment with 5-day-course of remdesivir, and recovered with favourable outcome.\\nThe patient is a 3-year-old toddler, without significant perinatal complications, who was diagnosed with PIBO at the age of 13 months, following an RSV and adenovirus coinfection. He was hospitalized for approximately 12 months due to difficulty in weaning off invasive and noninvasive respiratory support. He received multiple courses of pulse methylprednisolone treatment (10-30 mg/kg) and was placed on long term treatment with inhaled fluticasone and salbutamol, low dose oral azithromycin and oral montelukast (FAM).\\nHis HRCT chest demonstrates bronchial thickening, ground glass opacities, severe air trapping bilaterally and bronchiectasis ( Figure 1A ,B).\\nSince his discharge, at the age of 22 months, he remains on the FAM treatment and receives supplemental oxygen during the day (1-2 L/min) and nocturnal noninvasive ventilation (BLPAP). He is steroid-responsive and requires a pulse dose of methylprednisolone (10 mg/kg) every 4-6 weeks in line with his respiratory status.\\nTo date, he has established mixed hypercapnic and hypoxemic respiratory failure and a profound exercise intolerance. At baseline, he has tachypnea, subcostal and suprasternal retractions, diminished respiratory sounds, fine crackles on lung auscultation and hyperresonance on percussion.\\nThis patient presented at the Emergency Room due to fever 24 h before his admission, tired-appearing, with worsening tachypnea and mildly increased oxygen needs to 3 L/min. COVID-19 infection was confirmed by molecular testing in nasopharyngeal swab (cycle threshold 21) and chest X-ray showed peribronchial thickening and severe hyperinflation ( Figure 1C ). Laboratory evaluation revealed normal inflammatory biomarkers ( Table 1 ). The patient was transferred to the COVID-19 Department for further management.\\nUpon arrival, the patient was initiated on broad spectrum antibiotics and intravenous dexamethasone at a dose of 0.15 mg/kg for 5 days and weaning thereafter, for another 5 days. Following parental informed consent, a compassionate access application for Emergency Use Authorisation of remdesivir was requested and granted by the National Organisation for Health Care Provision. On day 4 from disease onset, an intravenous loading dose of remdesivir (5 mg/kg) was administered, followed by 2.5 mg/kg maintenance dose for 5 days with continuous hepatic and renal function monitoring.\\nNo associated drug toxicity was observed. Fever regressed and the patient returned to baseline oxygen requirements (2 L/min) within 24 h from remdesivir administration. He was discharged after 11 days of hospitalization, on baseline respiratory support. Follow-up examinations did not show signs of COVID-19 sequelae, his chest radiograph three months post COVID-19 course did not show significant changes and his baseline oxygen support remains unaltered ( Figure 1D ). Similarly, there is no pediatric-specific evidence that remdesivir mitigates disease severity.\\nAs the great majority of children suffer minimal symptoms, anyone could question if the potential benefits of antivirals are also limited and, therefore, the threshold of using it should be much higher, given the unknown side effects. In our case, remdesivir was well tolerated, although, it is impossible to determine its contribution to our patient's uneventful recovery.\\nIn conclusion, this unprecedented pandemic has put many clinicians under intense pressure to provide equivocal therapies based on compelling preclinical data. We feel that remdesivir may have contributed to our patient's relatively quick recovery, \",\n          \"The terms minimal absent words (MAWs) and nullomers both describe the shortest sequences that do not occur in the entire genome or proteome of an organism. Although many biotechnological applications have been envisioned, from potential selective drugs (1) (2) to forensic practice (3) , the actual role of nullomers has intensely been debated (4) and still remains enigmatic. Lately, fast tools and efficient algorithms have been introduced making the discovery of globally missing sequences practical (5-8, 42, 56-57) . In 2012, Alileche and colleagues demonstrated that two absent 5-amino-acid peptides cause fatal damage to cancer cells (2) , while 5 years later Alileche and Hampikian showed that the same MAWs have a broad lethal effect on cancer cell lines derived from nine organs represented in the NCI-60 panel (1) . Additionally, Silva and colleagues have reported that three minimal 12-nucleotide fragments entirely absent from the human genome, appear consistently at the same location in two protein-coding genes of Ebola virus genomes (9) . Although there exists an unambiguous sequence conservation in their findings, it is not clear whether the absence of these oligomers is statistically expected. In the same study, the term minimal relative absent words (RAWs) is introduced, describing sequences that are present in a pathogenic organism but absent from its host. On one hand, all the above findings ideally support the conjecture that MAWs may have gone extinct due to evolutionary pressure or putative deleterious effects. Perhaps the unfavourable properties of nullomers are linked with forbidden spatial conformations followed by functional consequences incompatible with life. Hence, the structural arrangements of globally absent motifs and the putative perturbation of molecules upon their appearance (i.e. emergence of a nullomer upon a mutation) form an interesting area for future research. Conversely, a finite set of sequences, for example the entire genome or proteome of a species, does not include all the different combinations of elements from the alphabet it is composed of, due to the fact that the combination of residues in a sequence increases exponentially with its length. Therefore, it still remains a riddle whether the absence of a nullomer is, in fact, an evolutionary consequence linked with adverse effects or a product of randomness. NCBI Genome (35; ftp://ftp.ncbi.nlm.nih.gov/genomes/). We developed custom Python scripts to discard headers and concatenate sequences from multiple fasta files, to produce files with one header and one-line sequence for each organism. The above step was applied to protein sequences as well, because MAW and em-MAW are developed in a way to calculate nullomers of each individual record in a fasta file. For the identification of peptide nullomers, we combined information both from the Swiss-Prot and TrEMBL sections of UniProtKB including protein isoforms. The redundant but slightly different sequences (isoforms) that were included in this step can contribute to the elimination of incorrectly identified MAWs. Moreover, the incorporation of predicted with manually reviewed records produces a list of more confident non-occurring sequences. The final pre-processing step included the removal of any ambiguous residues (character N in genomes or B, J, X, Z in proteins). Eventually, we generated lists of global nullomers from various species which were used as an input for downstream analysis in Nullomers Assessor.\\nStatistical models represent the observed variability in data by probability distributions. A simple model of a sequence = ( 1, 2, 3, . . . ) is a first-order Markov chain, where each position is dependent only on its immediate precursor. For example, the probability of observing a 'G' at one position depends (only) on whether there is a 'C' in the previous position. This can be expressed as:\\nIn our method, we consider genomes and proteomes (hereinafter background sequence) as Markov chains. A first order Markov chain is a model where each position is contingent merely on its previous position. Likewise, in an -th order Markov chain, each position depends on the previous positions.\\nIn order to decide whether a nullomer is statistically expected to exist, we estimate Markov probabilities.\\nFirst, the frequencies of elements (nucleotides or amino acids) of the background sequence are calculated. Then, three Markov probability matrices are generated, one for each of the first three Markov model orders. In general, a substitution matrix for distinct letters (e.g. = 4 for DNA) and order , is a grid of * probabilities. As an example, a stochastic matrix of third-order for all the naturally occurring amino acid residues in a bacterium (20 distinct amino acids assuming that neither selenocysteine nor pyrrolysine are present) requires a matrix of 20 * 20 3 = 160.000 cells (namely 20 rows and 20 3 columns, or vice versa). Each of these 160.000 probabilities indicate the likelihood for a specific amino acid to occur given the three preceding amino acids.\\nIn non-mathematical terms, each residue of a biological sequence is dependent on the previous elements, where defines the order of a Markovian process. Implicitly, a stochastic process in a biological sequence can reveal the sequential preferences among neighboring residues as well as reflect avoided motifs which may introduce an unfavorable structural folding. We utilised the abovedescribed fundamental mathematical notion and developed a custom Python script (Nullomers Assessor) which approximates the likelihood for a nullomer to occur zero times based on the first four orders of Markovian chains (including zeroth order). More precisely, four distinct p-values are assigned to each nullomer of the list. To elucidate and put this probabilistic property into a more biological context, we provide the following example. Assuming the peptide 'PTILA' is an absent minimal 5-mer, then the probability of being entirely absent based on a second-order Markov chain can be calculated as illustrated below:\\nInitially, the probability of 'PTILA' to occur is estimated (Equation 2). In this formula, P(P) denotes the observed frequency of proline in the entire background sequence. Subsequently, P(T|P) signifies the probability of threonine to arise after a proline. Similarly, P(I|TP) indicates the probability of isoleucine to occur given the previous 2 adjacent amino acids are a proline followed by a threonine, and so forth.\\nNext, the expected number of occurrences for the specific actual nullomer is estimated (Equation 3), a value that is used for the final estimation of the second order zero-occurrence probability (Equation 4).\\nIn a similar manner, the probabilities of first and third orders are computed, while the zeroth order simply mirrors the frequency of residues in the background sequence. In simple words, the calculated p-values represent the probability of a minimal absent word to be indeed absent based on the rate at which the residues occur as well as three additional transition probabilities which reflect the frequency of 2-, 3-, and 4-mers of the examined genome or proteome.\\nAlthough 4 different probabilities are calculated, the maximum p-value is kept and assigned to each examined nullomer. Since a p-value denotes the chance of a nullomer to occur exactly zero times (namely to be absent), then the lower the value, the more expected the nullomer is to exist (or equivalently the less expected it is to be, indeed, absent). By keeping the highest probability amongst the 4 calculated p-values, we expect to end up with fewer but more confidently true-positive results.\\nThere is a large number of short sequences that each have a chance of being absent. Therefore, the emergence of false positive results must be controlled (46) . More specifically, all the 4 calculated p- Alternatively, in the Benjamini-Hochberg procedure, the probabilities are sorted in descending order and sequentially rejected if the product of a p-value and the number of remaining tests is greater than a cut-off limit. The FDR method though, which constitutes a milder alternative, performs markedly loosely when applied to large eukaryotic genomes resulting in thousands of significant nullomers. This motivated us to incorporate a third correction option, the Tarone methodology. This is a special case of the Bonferroni method, where we make it less conservative by doing fewer tests. For a given wordlength k, we calculate the zero-occurrence probability (the maximum of the four Markov probabilities)\\nfor each of the 4 k-mers. Then, these 4 k-mers are ordered in descending order of zero-occurrence probability. Next, we exclude from testing any k-mer whose absence would not be significant because the Bonferroni-adjusted p-value is above a cut-off threshold:\\nEquation (6) represents the mathematical notation of the Tarone method, where counter denotes a number which progressively increases by 1 when a k-mer is excluded from testing. Finally, the \\\"testable\\\" k-mers that remain are compared to the actual nullomers in the list, and those in the intersection are output. In this way, the stringent nature of Bonferroni remains, whilst a milder adjustment is performed every time a test is excluded due to impossibility of significance.\\nNext, we tested the accuracy and rigour of the 3 different correction methods using a common dataset and a constant threshold of false discovery control. The results show that Bonferroni correction performs the strictest cleansing of false positive nullomers (very likely at the cost of increased false negatives), followed by the Tarone method, whereas the FDR approach constitutes the least stringent alternative.\\nMore precisely, any result-set derived by Bonferroni is a subset of the corresponding Tarone set, while the results of the FDR method almost always include all the above outcomes. Throughout our analysis, a fixed false discovery threshold of 1% has been applied, both when searching for genomic or peptide nullomers. In order to eliminate the emergence of Type I errors to the utmost degree, we report a nullomer as significant only when all four corrected probabilities are lower than the user specified cutoff. However, we advise users to apply Bonferroni correction to relatively large genomes, in order to control as much as possible the emergence of false-positive results, whilst starting with an alternative method when dealing with either small genomes (i.e. viruses or microbe datasets) or proteomes in general.\\nRandom sequence shuffling is a widely used approach to evaluate stochasticity as well as statistical significance of results. In order to evaluate the rigour of our method, especially because Equation (4) is only approximately true, we performed permutation tests by randomly shuffling the human proteome.\\nSince Nullomers Assessor calculates up to third-order probabilities, we sought to retain unaltered not only the counts of distinct amino acids, but also higher-order statistics, such as the frequency of adjacent letters (doublets and triplets) of the entire proteome. For this purpose, we used a sophisticated shuffling algorithm, uShuffle (21) , which performs random shuffling of sequences while preserving -let counts. The C# software package of the uShuffle method was used to shuffle the human proteome 10 times while preserving the frequency of amino acids, doublet occurrences and tripeptides. Next, for each of the 10 shuffled proteomes we re-generated lists of nullomers in order to examine whether any of the new random absent sequences would come into view as significant. By keeping the singlet, doublet, and triplet amino acid frequencies unchanged but not their order, we expected to end up with utterly different lists of nullomers. We used the original human proteome as a background sequence and re-ran our method 10 times (once per each new list of nullomers) in order to assess the newly created lists of 'counterfeit' absent words. Even though the script was executed using identical parameters (background proteome, correction method, threshold of statistical significance), no significant results emerged in any of the 10 attempts, demonstrating the stringency of our methodology.\\nThus, we assessed 10 sets of random missing sequences using the real transition probabilities of the reference human proteome given the fact our method scans any background sequence by considering frequencies up to 4-mers (precisely up to third-order Markov chains). The outcome of this step strongly suggests that Nullomers Assessor is able to disclose truly significant nullomers.\\nIn the present study, we rigorously assess absent sequences for their statistical significance. We examine genomes and proteomes from hundreds of organisms (Table 1 ) and show lists of nullomers which are unexpected to be absent, in contrast with other missing sequences. Our findings demonstrate that several thousands of absent sequences are statistically expected to occur in various genomes. The longest significant human genomic MAW is composed of 13 nucleotides, whilst the lengthiest significant peptide nullomer from the same organism is 5 residues long. After applying our method to the entire human genome, 13 significant genomic nullomers stood out (Bonferroni correction at 1% cut-off) from a set of more than 27 million non-occurring oligomers. In essence, the specific 13 missing words are highly statistically foreseen to occur somewhere in the human genome but, in reality, they are totally absent. In a similar manner, we analysed peptide nullomers from the human and mouse proteomes.\\nThirteen absent peptides from the human proteome were classified as significant when the Tarone correction method was used, while eight peptides emerged when we applied identical parameters to the mouse proteome (Table 3) . Moving the hypothesis of harmfulness a step forward, we systematically explored nullomer-making mutations which are one residue away from the reference sequence. More specifically, we calculated all the possible single amino acid substitutions in all protein records of each proteome that can give rise to any of the total 21 significant absent words. This might offer useful insights for unravelling plausible mechanisms of evolvability that underlie peptide nullomers. Prior research suggests that different residues differ in respect of their mutational preference (41) and reports implications in phosphorylation sites (37) . Therefore, the mutational landscape of nullomers presented in this work, may provide a useful resource for future sequencing studies, especially in the field of proteomics (40) . To this end, we highlight and share more than 30 Table 2 ).\\nTo assess whether nullomers have an ancestral origin, we examined a plethora of diverse organisms ranging from bacteria to human. To this end, we hypothesized that distantly-related genomes would share fewer similar sequence features and therefore one would not be surprised to find fewer or no nullomers in common. In contrast, despite the stringent filtering criteria of our methodology, we would expect to end up with identical significant nullomers from closely-related organisms due to the existing high similarity both in genomic and protein sequences. Surprisingly, while some significant nullomers are sporadically shared by some mammals, most are not shared by closely-related mammals ( Figure   1 ) Figure   2A ). In a like manner, the estimated median frequency of chimp-derived nullomers in the human genome is 53 ( Figure 2B ) while the detailed dataset is provided in Supplementary Table 3 . Figure 2C and 2D outline a similar trend in nullomers of Mus musculus and Canis lupus familiaris, respectively, where the expected number of occurrences is again more than the actual observations. This finding suggests an alternative hypothesis in which significant nullomers are not completely forbidden, but they are strongly suppressed. Strongly suppressed sequences are expected to occur just a few times, so by chance fluctuations some of them could appear zero times.\\nIn contrast, none of the roughly 2.850 unique significant nullomers in vertebrates are shared with any of the other species which belong to archaea, bacteria, protozoa, or fungi ( Figure 3A ). To some extent, this may be due to the heterogenous complexity among species of different kingdoms because significant nullomers of vertebrates are usually longer ( Figure 3B ). Overall, these observations are in accordance with findings reported by Acquisti et al. (4) . The authors of that study have demonstrated that species with a more-recent common ancestor share more MAWs in common in comparison with more distantly-related organisms. Here, we provide evidence that more-closely related organisms share not only common deficits in general, but also some identical significant missing patterns, supporting the hypothesis of evolutionarily-conserved aversion to these sequences.\\nFurthermore, we observed multiple genomic MAWs in common across species ranging from 6 to 14 residues. Table 2 presents the most frequent significant nucleotide sequences whose absence is shared between at least 18 species, while a complete list is in Supplementary Table 4 \\nAccording to the definition, a nullomer is a minimal-length sequence that does not occur within a longer sequence. Consequently, if we think of a \\u2212 1 peptide derived from a nullomer of length without either the first or the last amino acid, then the remaining part of \\u2212 1 residues certainly occurs somewhere in the background sequence. This information is enough to identify regions in proteins that are susceptible to create a significant nullomer upon a mutation. Additionally, single amino acid alterations capable of generating a nullomer can occur in non-edge positions of a k-mer. To illustrate this point further, we provide the following example. BRCA1 is a human gene that produces a tumour suppressor protein of damage repair and embryonic development (38, 39) . What can be seen in Figure 5 is that dominant mutational trends remain unaffected either with or without predicted records.\\nRecently, massive efforts have been put forth to prioritize the functional importance of phosphorylation sites (54, 55) as well as decipher correlation between mutation and phosphorylation in cancer (75) . For this reason, we compiled a curated list of susceptible nullomer-making phosphosites retrieved from PhosphoSite Plus database (36) . Our analysis revealed 176 phosphorylation sites of high confidence which are prone to give rise to a nullomer (Supplementary Table 1) , whilst a similar tendency is notable where valine is the most prominent mutational target ( Figure 6 ). Future research should be devoted to characterizing the impact of nullomer-making mutations, especially when a substitution occurs in post translational modification (PTM) sites.\\nRecent studies have shown that viral sequences mimic those of their hosts, to some extent, in part to evade immune responses (32, (50) (51) , and this can be used to predict viral hosts (33) . To this end, we set out to survey virus genomes and proteins and investigate whether the scenario of RAWs fits with our previous human-derived findings. Viral sequences and complementary annotation were downloaded from NCBI Virus (43) in March 2020. We filtered out incomplete sequences, whilst we further discarded records that have been isolated from non-human species, resulting to 33.610 genomes and 233.178 proteins. Next, we queried whether significant human nullomers appear in any of the viral sequences. We observed that 5 genomic nullomers (out of the total 13) were present in 39 unique sequences of 3 distinct virus families ( Figure 7A ). What stands out is the remarkably low number of significant human-RAWs which are present in the virus genomes (Supplementary Table 2 ). In other words, there is a general absence of absent human sequences that are present in viruses.\\nLikewise, we investigated whether human-derived peptide nullomers are present in viral protein sequences. We observed that 10 out of the 13 significant human MAWs emerge in various proteins of to 10 residues were revealed significant upon Bonferroni correction at 1% cut-off. A wide species coverage is observed since the overall motifs are absent from more than 6.000 distinct species whereas, interestingly enough, the most frequent missing sequences are recognition sites of host restriction enzymes, according to the Restriction Enzyme Database (65; http://rebase.neb.com/). Restriction enzymes present in prokaryotes recognise specific sequential patterns in viruses and cleave their DNA sequences into fragments (66) (67) . These cutter enzymes provide a defence mechanism in host species against virus invasion and have been successfully utilized in a range of biotechnological applications and research studies (68) (69) (70) (71) (72) . Although it is known that restriction sites confer a selective disadvantage to viruses, this has not been previously linked to nullomers. The fact that our top hits include several palindrome sequences ranging from 4 to 6 bases in length offers potential for virus nullomers to be utilised as predictors of previously unknown restriction motifs in host organisms. Furthermore, given the robustness of our evaluation method, the entire dataset can offer a valuable resource for future in silico research. Table 5 presents the most frequent virus nullomers which are shared between hundreds of viral species while the entire list is provided in the form of a spreadsheet (Supplementary Table 5 ) as well as through Nullomers Database.\\nTo make the nullomers of the current study easily accessible and explorable, we developed an online portal available at https://www.nullomers.org. Nullomers Database aims to serve as a central hub of information for further investigation of minimal absent words. The provided results are significant identified genomic and peptide nullomers assessed by Nullomers Assessor. Main emphasis has been given to peptide nullomers and, more specifically, to regions of proteins that are prone to give rise to a significant nullomer upon a single amino acid alteration. Assuming that the deficit of the shortest absent words is governed by evolutionary factors with unfavourable consequences, then one would expect that the emergence of a nullomer upon a mutation would possibly have a negative effect. To prevent outdated information remaining in Nullomers Database, complex stored procedures have been developed in conjunction with an automated communication channel which retrieves information from\\nUniProt. The gene name, protein sequence, description, sequential annotation as well as protein status (i.e. whether a protein is still active or has been deprecated and moved to UniProt Archive) are asynchronously collected from UniProt via a REST web-service. Moreover, in order to assess the significance and determine the effect of putative nullomer-making mutations, we provide functional impact annotation by Mutation Assessor (22, 23) , a server which predicts functional consequences of amino acid alterations. Given the dynamic nature of the UniProt database, the information retrieval of all the above-described steps has been automated, making Nullomers Database a fully autonomous, scalable, and frequently updated repository. Additionally, the integration of MolArt (24) , an interactive visualization plugin (Figure 9 ), allows for the simultaneous exploration of multiple sequential and structural features in protein nullomers. The interconnected and synchronized panels of MolArt permit users to identify co-occurrent elements in regions that are prone to engender a missing word upon a single amino acid substitution. The entire sequence annotation of the plugin is retrieved from UniProt in real-time whereas the corresponding experimental structures are dynamically fetched from Protein DataBank (or from Swiss Model Repository when a structure is not available). Therefore, the specific functionality of Nullomers Database ensures that the provided information can always be in line with major protein databases and automatically be enriched over time.\\nThis paper introduces Nullomers Assessor, a probabilistic protocol provided as an open-source software tool, for the assessment of any set of minimal absent genomic or peptide sequences. The software offers a rigorous way of filtering missing words by Markov chains, while three statistical correction methods are available to control false positive results. We applied the script to entire genomes of hundreds of species and observed that numerous nullomers are statistically significant in multiple organisms. Moreover, we systematically examined more than 147.000 individual virus sequences and observed that the most frequent significant absent motifs are restriction recognition sites. In addition to the prevailing hypothesis that minimal absent words have gone extinct due to negative selection, we suggest that nullomers may have been replaced by more specialized sequences, which execute similar or even optimized functions.\\nWe analysed the human and mouse proteomes and identified positions that are prone to introduce a significant missing peptide. We found that more than one fourth of human proteins can give rise to a significant nullomer upon a single amino acid substitution. We freely provide our findings in a visual, interactive, and user-friendly way via Nullomers Database. Taking advantage of the powerful functionalities that modern web technologies provide, we highlight protein positions which can generate a minimal absent word in their sequences.\\nIn summary, the present study reveals significant nullomers that are unlikely to be absent by chance.\\nFurther studies should be conducted to experimentally validate and determine the actual role of nullomers as well as the extent of harmfulness behind nullomer-making mutations; hence, we anticipate that both Nullomers Assessor and Nullomers Database can be useful resources and facilitate research towards a better understanding of the still mysterious role of minimal absent words. 12-mer 0 0 0 1 13-mer 0 0 0 0 Table 5 . List of the most frequent (present in >1.000 distinct sequences) significant nullomers in viruses.\",\n          \"Bee propolis is a viscous natural product produced by honeybees to seal their hive. Although its chemical composition varies according to environmental conditions, geographical location and available plant populations, the biological health benefits of propolis long have been appreciated throughout the world as part of traditional medicines. For instance, ancient Egyptians used propolis in embalming procedures because of its antimicrobial properties [1, 2] . The extensive health properties arise from the compounds in propolis. Notwithstanding the variations described earlier, bee propolis consistently contains a high concentration of phenolic Experimental section X-ray. Galangin (Indofine Chemical Co, Hillsborough, NJ, USA) was recrystallized from ethanol by slow evaporation. Table 1 shows the crystal data. Data collection was taken with a Bruker Smart APEXII single crystal X-ray diffractometer, with a Charge Coupled Device Detector and the experiment was done at 125 K using a cold liquid nitrogen stream from Oxford cryosystems. The X-ray source emitted Mo K\\u03b1 radiation at 0.71073 \\u00c5. We used the SHELX program to determine and refine antioxidant crystal structure [19] , and input the Xray data into the MERCURY program to produce images of the molecules and crystal packing [20] . Crystal data of galangin have been deposited at the Cambridge Structural Database (CSD) and are available at https://www.ccdc.cam.ac.uk/structures/? using Identifier CCDC number 2123390.\\nRRDE measurement of antioxidant activity. Materials used to determine the antioxidant activity of galangin and propolis were tetrabutylammonium bromide (TBAB; Sigma Aldrich) and 99% anhydrous Dimethyl Sulfoxide (DMSO; Sigma Aldrich. Galangin (Indofine Chemical Co., Hillsborough, NJ, USA). Propolis was provided by a co-author (C. Smart), from his own hive in the Hudson Valley. The 0.1 M TBAB/DMSO solution was used to produce electric current and enhance the occurrence of redox reactions. Antioxidant activity was measured via the hydrodynamic voltammetry technique with a rotating ring disk electrode (RRDE). The equipment used in this experiment was an MSR electrode rotator with CE and ETL marks, together with a WaveDriver 20 benchtop USB from Pine Instrumentation. The main electrode tip was an E6RI ChangeDisk with a rigid gold ring and gold disk (Au/Au) insert. Before and after each experiment, 0.3\\u03bcL Alumina suspension was used to clean the disk electrode tip (Allied High Tech Products, Inc, Rancho Dominguez, CA, USA) on a moistened polishing microcloth to eliminate potential film formation. A Platinum (Pt) reference electrode and Pt counter electrode were also used in this experiment. All electrodes were obtained from Pine Research Durham, NC, USA. [21] . Cyclic voltammagrams were run using a Solartron SI 1287 Potentiostat/galvanostat (Solartron Analytical, Oakridge, TN, USA) controlled through Coreware\\u00a9 software.\\nThe antioxidant activity of galangin was determined based on its superoxide radical scavenging ability that was measured using the protocol developed in our lab [22] . A stock solution of galangin, 0.02 M, in anhydrous DMSO was used in trials, whereas 0.335 g of propolis were dissolved in 10 ml of DMSO and used as stock solution. For the experiment, the electrolytic cell was bubbled for 5 minutes with a dry O 2 /N 2 (35%/65%) gas mixture to establish its dissolved oxygen level. The Au disk electrode was then rotated at 1000 rpm while the disk was swept from 0.2 V to -1.2 Volts and the ring was held constant at 0.0 Volts, the disk voltage sweep rate was set to 25 mV/s. In summary, 4 runs plus blank were performed in the RRDE experiment for galangin to determine the antioxidant activity with equal addition of 5 \\u03bcL of galangin stock solution, whereas 5 aliquots, were added for propolis study.\\nResults from each run were collected on Aftermath software and represented as voltammograms showing current vs. potential graphs that were later analyzed using Microsoft Excel. In an RRDE voltammetry experiment, the generation of the superoxide radicals occurs at the disk \\nReverse Reaction 2: Oxidation of superoxide radicals at the ring\\nThus, the rate at which increasing concentrations of galangin scavenged the generated superoxide radicals during the electrolytic reaction was determined by obtaining the quotient of the ring current and the disk current (percent value) at each concentration. These values were denoted as the collection efficiency of the scavenger at different concentrations. Using Microsoft Excel, collective efficiency values were plotted against the corresponding concentrations of galangin to produce a graph illustrating the effect of increasing concentrations of galangin on the scavenging of superoxide radicals in the electrolytic solution. Ultimately, the slope of the curves served as a quantitative measure of the antioxidant activity of galangin and propolis.\\nComputational experiments. Antioxidant activity of galangin studied using DFT. Calculations were performed using programs from Biovia (SanDiego, CA, USA). Density functional theory (DFT) code DMol3 was applied to calculate energy, geometry, TS and frequencies implemented in Materials Studio 7.0 [23] . We employed the double numerical polarized (DNP) basis set that included all the occupied atomic orbitals plus a second set of valence atomic orbitals, and polarized d-valence orbitals [24] ; the correlation generalized gradient approximation (GGA) was applied including Becke exchange [25] , plus BLYP-D correlation including Grimme's correction when van der Waals interactions were involved [26] . We used the DNP basis set, which consists of a double-numeric quality basis set for approximately 2 atomic orbitals for each one occupied in free atom (DN), plus basis with polarization functions, that is, functions with angular momentum one higher than that of highest occupied orbital in free atom.\\nCalculations were performed with DMSO solvent inclusion, using the continuous model of Dmol 3 [27] . We used the DNP basis set, which consists in a double-numeric quality basis set for approximately 2 atomic orbitals for each one occupied in free atom (DN) plus basis with polarization functions, that is, functions with angular momentum one higher than that of highest occupied orbital in free atom. All electrons were treated explicitly and the real space cutoff of 5 \\u00c5 was imposed for numerical integration of the Hamiltonian matrix elements. The self-consistent field convergence criterion was set to the root mean square change in the electronic density to be less than 10 \\u22126 electron/\\u00c5 3 .\\nNo human, animal or cell studies were used in research described in this manuscript.\\nThe galangin crystal grown after a few days from an ethanol solution showed color variation: it was clear when observed in one direction but reddish-yellow when observed at an approximate perpendicular direction. A data crystal was chosen (Fig 1) and an APEX2 DUO platform X-ray diffractometer from Bruker Advanced X-ray Solutions was used to obtain X-ray data measurements at 125 K. The crystal structure was solved and refined using ShelX programs [19] . Views of the crystal structure down the crystallographic axes, a, b and c are shown in S1 Although galangin was crystallized from ethanol, a water molecule is present in the asymmetric unit. The water molecules function as hydrogen bonded bridges between layers of galangin molecules. Additionally, the distance between water oxygen atoms (viewed down the a-axis) is equal to the crystallographic a-axis ( Fig 2B) . The hydrogen bond parameters are listed in Table 2 The results from our experimental RRDE studies show that galangin has very strong antioxidant activity (described later) towards the superoxide radical anion and are compatible with earlier work using the DPPH method to evaluate galangin radical scavenging ability [29] . The galangin crystal structure with its rich network of hydrogen bonds and stacking interactions supports this conclusion. We then sought to explain this strong antioxidant behavior by using computational methods to explore galangin scavenging behavior with the superoxide radical anion.\\nGalangin X-ray coordinates were input into Materials Studio program DMOL 3 , and Fig 3A shows relevant distances obtained after geometry optimization. This structure was obtained as an energy minimum, i.e. having only positive frequencies after vibrational analysis calculation; these distances will be of interest for comparison with the structural changes occurring after \\nscavenging the superoxide radical. As will be described later, galangin is a good scavenger of superoxide and here we propose a potential mechanism of scavenging. The common mechanism by which polyphenols scavenge radicals is through H-atom transfer from a polyphenol hydroxyl to the radical. In galangin, there are 3 potential hydrogen atoms donors from the \\nhydroxyl groups located in position 3, 5 and 7. Initially, one might imagine that the 7OH would be favored for this process, since the H atoms in the remaining hydroxyls of galangin form strong intramolecular H-bonds with the O(carbonyl), making these H atoms more difficult to extract from the polyphenol. Fig 3B shows the result of DFT geometry optimization for superoxide separated from H7, initially set as 2.60 \\u00c5, which is the distance corresponding to the sum of van der Waals distances for O and H atoms. It is seen that H7 is not captured by superoxide radical. However, this result may be due to an energy barrier and so we minimized the potential product to exclude that possibility: HO 2 was placed at van der Waals separation from the H7 excluded radical galangin, and the whole system becomes a radical species. This results in no recapture of H7 by the galangin radical, Fig 3C. Comparing energies of the resulting product structures shown in Fig 3C and 3B , we see that the latter is 1.7 kcal/mol lower and A: Galangin geometry obtained from X-ray coordinates after DFT minimization; B: Initial van der Waals separation between H7 and superoxide, 2.60 \\u00c5, decreases to 1.465 \\u00c5 after DFT minimization, while hydroxyl separation is still a bond length, 1.065 \\u00c5. Superoxide: stick style; galangin 7-OH: ball and stick style; C: H7 was eliminated from the hydroxyl and the remaining species, the galangin radical, was DFT minimized and later placed 2.60 \\u00c5 from HO 2 for further geometry minimization. Fig 3C shows However, it is known that the presence of a 3-hydroxyl in the flavonoid system induces antioxidant activity [30] . Here we show that the interaction between superoxide and this hydroxyl substituent in galangin confirms this experimental fact. Minimization of the van der Waals approached superoxide to H3(hydroxyl), Fig 5A, converges to a structure very similar to the one approaching HO 2 and the H3 excluded galangin radical, Fig 5B. More importantly, the energies of these 2 minima are exactly the same (\\u0394G = 0.01 kcal/mol). We conclude that the initial interaction between galangin H3 and superoxide generates a complex. It is interesting to analyze the outcome when a proton approaches this complex. From the structural complex created in the related minimization, shown in Fig 5C, it is seen that H 2 O 2 is formed.\\nThe galangin-superoxide complex shown in Fig 5A was van der Waals placed nearby a proton. This is a neutral radical that after minimization shows formation of H 2 O 2 plus the galangin derivative radical, which locates the unpaired electron in the polyphenol, Fig 426 \\u00c5) . We can consider this radical having an extended conjugation. Ring A shows 2 bonds (yellow 1.390 \\u00c5 and 1.399\\u00c5) shorter than lengthened bonds, (purple 1.414 \\u00c5, 1.406 \\u00c5, 1.438 \\u00c5 and 1.412 \\u00c5), but this effect is already present in free galangin, Fig 3A and in the X-ray crystal structure (1.381(2) \\u00c5 and 1.377(2) \\u00c5).\\nUsing our computational results, we provide mechanistic evidence that galangin can act as a superoxide dismutase (SOD) mimic. From Fig 5C, H 2 O 2 was excluded, and superoxide \\u03c0approached to ring B, which was chosen because it has some quinone characteristics and so likely to be a good target for capturing the superoxide unpaired electron. Indeed, the superoxide electron is captured by the ring, while O 2 is formed (O-O bond distance = 1.273 \\u00c5) but the latter does not penetrate galangin \\u03c0 system, as its separation from ring B is still similar to van der Waals value, 3.508 \\u00c5, Fig 6A. Hence, molecular oxygen was eliminated, and the residual structure minimized, Fig 6B. Finally, a proton was van der Waals approached to galangin O3 and, upon minimization, galangin was reformed. This suggests galangin acts as a SOD mimic for scavenging superoxide with H3 supporting reaction (a), corresponding to SOD activity: (1) Incoming superoxide (red) interacts with H3; (2) H3 is sequestered, forming HO 2 ; (3) a proton arrives and adds to HO 2 forming, and eliminating, H 2 O 2 (light blue); (4) A second incoming superoxide (green) interacts \\u03c0-\\u03c0 with galangin ring B; (5) the superoxide donates its electron and O 2 is released (light blue); (6) an incoming proton interacts with O3, re- Finally, we explore if hydroxyl H5 can also detach from galangin when approached by superoxide, by performing DFT-D minimizations. Although there is some shortening between H5 and superoxide, it is much weaker than that seen for H3, and additional approach by a proton does not induce H 2 O 2 formation (not shown). Hence, we studied the behavior of the potential product, which is an HO 2 moiety interacting with galangin after H5 exclusion. The energy of this product is lower and a TS was explored. Fig 6C and 6D , show that the capture of H5 by superoxide is feasible: \\u0394G = -2.067 kcal/mol, E(barrier) = 1.3 kcal/mol.\\nWe conclude that, abstraction of a galangin H atom by superoxide is feasible for hydroxyl hydrogens H3 and H5, with H3 slightly favored because it is a thermodynamically spontaneous process, whereas H5 needs to overcome an energy barrier. However, this H5 barrier is easily affordable (1.3 kcal/mol). Also, in contrast with chrysin [22] , H7 galangin is not captured by superoxide.\\nThe antioxidant ability of flavonoids has been studied fro some time [11] [12] [13] [14] [15] [16] [17] . Recently, computational studies on flavonoid derivatives, including galangin and quercetin, have focused primarily on the importance of hydroxyl groups in exocyclic phenyl ring B (which galangin does not have) and pyrone ring C (the C-3 hydroxyl group) present in both galangin and quercetin [18] . Our DFT results are in agreement with findings in this study, than the hydroxyl H3 atom is slightly favored for hydrogen abstraction and antioxidant activity, over hydroxyl H5.\\nOur previous investigations on superoxide scavenging by embelin [31] emodin [32] , and chalcones [33] described an additional mode of sequestering superoxide, through the \\u03c0-\\u03c0 interaction between superoxide radical and the polyphenol. This scavenging mechanism is suggested for galangin as well, since, as described later in RRDE results, galangin is (S4 Fig, the input configuration) shows the incoming group stabilized with a centroid-centroid separation of 3.134 \\u00c5. The original separation between superoxide O atoms becomes much shorter, 1.246 \\u00c5, corresponding to the expected length of a O 2 molecule, but still bound to the polyphenol. It is clear that the superoxide unpaired electron is now located within the polyphenol structure and this species is not a radical, as the number of electrons is now even due to 2 superoxides incorporated into the complex. It is noteworthy that one H from H 2 O 2 now is within H-bonding distance (2.820 \\u00c5) to the polyphenol O1. To analyze the corresponding potential H 2 O 2 transfer to the polyphenol plane, a transition state calculation was performed after observing that the product has lower energy than the reagent, Fig  8A and 8B . Attempts to explore if the newly formed O 2 species, trans to H 2 O 2 , or H 2 O 2 itself, are able to detach from the Fig 9B complex failed, although additional superoxide radicals were incorporated as seen in the arrangement of Fig 9C. Indeed, the complex shown in Fig 9C contains the maximum number of superoxide radicals captured by galangin in this study. It is strongly \\nsuspected that if additional protons are available they may be incorporated to stabilize further H 2 O 2 units. This DFT study is summarized in the following Scheme 2.\\nScheme 2. The \\u03c0-\\u03c0 scavenging of superoxide by galangin: (a) Superoxide interacts with galangin ring A (\\u03c0-\\u03c0 approach); (b) the unpaired electron enters the galangin ring; (c) (d) two protons are incorporated within the O 2 moiety forming H 2 O 2 ; (e) a 2 nd superoxide is incorporated, (f) a proton captured by O 2 forms a HO 2 moiety; (g) after an additional proton the original H 2 O 2 (red) detaches from O-1 galangin position.\\nWe see that the \\u03c0-\\u03c0 approach of superoxide to galangin is favored and even 4 superoxide radicals can be stabilized along with a H 2 O 2 molecule on galangin (Fig 9C) , with total charge = -2, which is surprisingly high. This complex is a result of 5 incorporated superoxide and 2 protons, but before formation of this complex, H 2 O 2 formation and release was described. Thus, the number of superoxide anions associated with galangin in this \\u03c0-\\u03c0 superoxide-galangin study is six, which when added to 4 protons, defines the final charge of the complex (-2) shown in Fig 9C. From this DFT study, we conclude that 1) galangin hydroxyls H3 and H5 can be captured by superoxide, suggesting galangin acts as a mimic of SOD catalysis; 2) H7 is not captured by superoxide; 3) galangin \\u03c0-\\u03c0 interactions can incorporate several superoxide radicals, and form a detached H 2 O 2 product. These conclusions are further discussed when analyzing the RRDE experimental results of galangin scavenging of superoxide, vide infra.\\nThe antioxidant activity of galangin was determined based on its superoxide radical scavenging ability that was measured using a protocol developed in our lab [22] . In this cyclic voltammetry method superoxide is developed in situ by bubbling O 2 into the electrovoltaic cell containing anhydrous DMSO solvent, see details in Experimental section. After the blank experiment, aliquots taken from a stock 0.02M galangin solution, were added to the voltaic cell. Fig 10A shows voltammograms of blank and all runs for both electrodes. The lower part of these voltammograms develops at the disk electrode, where O 2 incorporates an electron to generate superoxide O 2 \\u2022 -. The opposite effect, oxidation of O 2 \\u2022 -, takes place at the ring electrode, which has a fixed potential enough positive for this action, upper part. As scavenger aliquots are added, they consume part of the superoxide and the ring disc detects less superoxide, lowering the current. The collection efficiency is shown in Fig 10B, for all data, showing the decrease of superoxide. All scavengers studied by us so far behaved in a linear manner for the initial aliquots used. Beyond the initial aliquots, 2 possibilities were seen: (1) complete linear behavior, as in, chrysin [22] , eriodictyol [22] , BHT [31] , emodin [32] , celastrol [34] or (2) initial linear behavior followed by asymptotic performance, such as in quercetin [22] , embelin [31] and clovamide [35] . In the present study, we also calculated a linear trend for all data, with regression factor of 0.9296, whereas the initial 3 data (using the blank plus 2 initial added aliquots) have a linear regression factor of 0.9635, slightly better, but still not ideal. Since we find that several superoxide anions are incorporated in one galangin unit, Fig 9C, galangin is a very efficient scavenger, and indeed this is demonstrated by the slope in Fig 10B, -19x10 4 . In \\nfact, when comparing with previous studies following the same protocol, quercetin [31] , (slope = -6.0x10 4 ) was the radical scavenger showing the steepest slope; in this study, galangin shows an even stronger superoxide scavenging capacity than quercetin. This is in contrast with calculations reported by Spiegel et al. [18] where thermodynamic parameters showed quercetin as a better radical scavenger, and which was explained as due to the catechol moiety present in phenyl ring B of quercetin.\\nOur DFT study demonstrates the strong capability of galangin to capture superoxide radicals via \\u03c0-\\u03c0 interaction. This is in agreement with our experimental RRDE showing galangin to be a better antioxidant than quercetin by having a steeper slope (Fig 10B) compared with quercetin. It is interesting that the commercially used antioxidant, BHT, has a much milder slope, -1.6x10 3 , [31] . \\nPropolis was also analyzed using the RRDE method. Fig 11A shows the related voltammograms and Fig 11B shows the antioxidant efficiency of propolis when scavenging the superoxide radical where the x-axis denotes the total volume of added aliquots; the linear form shows a non-ideal regression factor, 0.8604. Fig 11C only considers the initial 4 points of Fig 11B shows a linear behavior with an excellent regression factor, 0.9958. When comparing with our previously described RRDE analysis of olive oil [36] , which is also a mixed substrate, propolis shows higher antioxidant activity, as shown by an almost complete elimination of superoxide after the last aliquot.\\nResults from cyclic voltammetry using a ring disk electrode (RRDE), show that galangin is a powerful scavenger of the superoxide radical anion. This is compatible with the galangin crystal structure that shows a rich network of hydrogen bonds and stacking interactions. Using DFT methods, \\u03c0-\\u03c0 interaction between superoxide and galangin are further confirmed; these interactions are in part responsible for superoxide oxidation and capture of the superoxide electron by the aromatic rings of galangin. Additionally, abstraction of galangin H3 and H5 by superoxide is also demonstrated and associated with SOD action by galangin. This SOD action may be responsible for the RRDE steeper slope of galangin compared with quercetin [22] . When analyzing propolis, RRDE results show very strong antioxidant capability, including almost complete elimination of superoxide from the electrochemical cell. Since propolis contains other scavengers besides galangin, a main component of propolis, the total amount of effective compounds in the propolis contribute to the antioxidant activity.\\nSupporting information S1 Fig. Unit cell packing as viewed down a-axis, down b-axis and down c- \"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8997,\n        \"samples\": [\n          \" Zuercher, S. J..  Banzer, C..  Adamus, C.. <br>Lehmann, A. I..  Richter, D..  Kerksieck, P.\",\n          \" De, Shankhyajyoti.  Dey, Arabin Kumar.  Gauda,<br>Deepak\",\n          \" Jairak, Waleemas.  Chamsai, Ekkapat.  Udom,<br>Kitikhun.  Charoenkul, Kamonpan.  Chaiyawong,<br>Supassama.  Techakriengkrai, Navapon.  Tangwangvivat,<br>Ratanaporn.  Suwannakarn, Kamol.  Amonsin, Alongkorn\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \" A toddler diagnosed with severe<br>postinfectious bronchiolitis obliterans and COVID\\u201019<br>infection\",\n          \" Significant non-existence of sequences in<br>genomes and proteomes\",\n          \" Antioxidant properties of bee propolis and an<br>important component, galangin, described by X-ray<br>crystal structure, DFT-D and hydrodynamic<br>voltammetry\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"journal\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2785,\n        \"samples\": [\n          \"J Adv Nurs\",\n          \"Health Promotion International\",\n          \"SAGE Open Med\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract_summary\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7281,\n        \"samples\": [\n          \" Victor M Corman et al.\",\n          \" : The social distancing phase allows us to model<br>COVID-19 in a way when the effect of latent<br>unreported/asymptomatic cases can be reliably observed.\\nAccurate<br>forecasts for COVID-19 are necessary for better<br>preparedness and resource management. Specifically,<br>deciding the response over months or several months<br>requires accurate long-term forecasts which is<br>particularly challenging as the model errors accumulate<br>with time. A critical factor that can hinder<br>accurate longterm forecasts, is the number of<br>unreported/asymptomatic cases. While there have been early serology<br>tests to estimate this number, more tests need to be<br>conducted for more reliable results. To identify the<br>number of...\",\n          \" In the Netherlands, as in many other nations,<br>the government has proposed the use of a<br>contact-tracing app as a means of helping to contain the spread of<br>the corona virus. The discussion about the use of<br>such an app has mostly been framed in terms of a<br>tradeoff between privacy and public health. This<br>research statement presents an analysis of the Dutch<br>public debate on Corona-apps by using the framework of<br>Orders of Worth by Boltanski and Th\\u00e9venot (1991). It<br>argues that this framework can help us to move beyond<br>the dichotomy of privacy vs. public health by<br>recognizing...\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe",
              "variable_name": "df_covid"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-7f37552a-cc59-4838-86c9-dae4ea07659e\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>paper_id</th>\n",
              "      <th>doi</th>\n",
              "      <th>abstract</th>\n",
              "      <th>body_text</th>\n",
              "      <th>authors</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>abstract_summary</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>4abd0c9b745b665cd9677a3de0d7e8b9a38c0f23</td>\n",
              "      <td>10.1111/aphw.12261</td>\n",
              "      <td>This study investigated how stress, physical a...</td>\n",
              "      <td>ity trackers were recruited via social media. ...</td>\n",
              "      <td>Woodruff, Sarah J..  Coyne, Paige.  St‐Pierre...</td>\n",
              "      <td>Stress, physical activity, and&lt;br&gt;screen‐rela...</td>\n",
              "      <td>Appl Psychol Health Well Being</td>\n",
              "      <td>This study investigated how stress, physical&lt;...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>c2aa3e817499fe7e08dc32dc8be7757ad710a50f</td>\n",
              "      <td>10.3762/bjoc.18.22</td>\n",
              "      <td></td>\n",
              "      <td>RNA-targeting oligonucleotides (e.g., antisens...</td>\n",
              "      <td>Kumar, Pawan.  Brown, Tom</td>\n",
              "      <td>The role of chemistry in the success of&lt;br&gt;ol...</td>\n",
              "      <td>Beilstein J Org Chem</td>\n",
              "      <td>Not provided.</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>691a53fbf2a6ab87757dc315f558b52ddd1ff7c4</td>\n",
              "      <td>10.1177/1473325020973394</td>\n",
              "      <td>My emotional responses to this moment include ...</td>\n",
              "      <td>under the former Soviet Union with signs that ...</td>\n",
              "      <td>Crampton, Alexandra</td>\n",
              "      <td>The lie of pandemic pivot and essential work</td>\n",
              "      <td>Qual Soc Work</td>\n",
              "      <td>My emotional responses to this moment include...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>975e30a2e5e456c653a5be91cde1e7c6cf765071</td>\n",
              "      <td>10.4093/dmj.2020.0266</td>\n",
              "      <td></td>\n",
              "      <td>First of all, we would like to thank Kim et al...</td>\n",
              "      <td>Moon, Sun Joon.  Rhee, Eun-Jung.  Lee,&lt;br&gt;Won...</td>\n",
              "      <td>Independent Impact of Diabetes on the Severit...</td>\n",
              "      <td>Diabetes Metab J</td>\n",
              "      <td>Not provided.</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>6fed2795f636338b067cb926ec4750f983dea435</td>\n",
              "      <td>10.1007/978-3-030-60039-6_8</td>\n",
              "      <td>In the current century, the novel coronavirus ...</td>\n",
              "      <td>Another aspect of wearable devices is general ...</td>\n",
              "      <td>Krishnamurthi, Rajalakshmi.  Gopinathan,&lt;br&gt;D...</td>\n",
              "      <td>Wearable Devices and COVID-19: State of the&lt;b...</td>\n",
              "      <td>Emerging Technologies for Battling Covid-19</td>\n",
              "      <td>In the current century, the novel coronavirus...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-7f37552a-cc59-4838-86c9-dae4ea07659e')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-7f37552a-cc59-4838-86c9-dae4ea07659e button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-7f37552a-cc59-4838-86c9-dae4ea07659e');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-15a188a9-28d3-4300-93de-78e97d65ae1a\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-15a188a9-28d3-4300-93de-78e97d65ae1a')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-15a188a9-28d3-4300-93de-78e97d65ae1a button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "                                   paper_id                          doi  \\\n",
              "0  4abd0c9b745b665cd9677a3de0d7e8b9a38c0f23           10.1111/aphw.12261   \n",
              "1  c2aa3e817499fe7e08dc32dc8be7757ad710a50f           10.3762/bjoc.18.22   \n",
              "2  691a53fbf2a6ab87757dc315f558b52ddd1ff7c4     10.1177/1473325020973394   \n",
              "3  975e30a2e5e456c653a5be91cde1e7c6cf765071        10.4093/dmj.2020.0266   \n",
              "4  6fed2795f636338b067cb926ec4750f983dea435  10.1007/978-3-030-60039-6_8   \n",
              "\n",
              "                                            abstract  \\\n",
              "0  This study investigated how stress, physical a...   \n",
              "1                                                      \n",
              "2  My emotional responses to this moment include ...   \n",
              "3                                                      \n",
              "4  In the current century, the novel coronavirus ...   \n",
              "\n",
              "                                           body_text  \\\n",
              "0  ity trackers were recruited via social media. ...   \n",
              "1  RNA-targeting oligonucleotides (e.g., antisens...   \n",
              "2  under the former Soviet Union with signs that ...   \n",
              "3  First of all, we would like to thank Kim et al...   \n",
              "4  Another aspect of wearable devices is general ...   \n",
              "\n",
              "                                             authors  \\\n",
              "0   Woodruff, Sarah J..  Coyne, Paige.  St‐Pierre...   \n",
              "1                          Kumar, Pawan.  Brown, Tom   \n",
              "2                                Crampton, Alexandra   \n",
              "3   Moon, Sun Joon.  Rhee, Eun-Jung.  Lee,<br>Won...   \n",
              "4   Krishnamurthi, Rajalakshmi.  Gopinathan,<br>D...   \n",
              "\n",
              "                                               title  \\\n",
              "0   Stress, physical activity, and<br>screen‐rela...   \n",
              "1   The role of chemistry in the success of<br>ol...   \n",
              "2       The lie of pandemic pivot and essential work   \n",
              "3   Independent Impact of Diabetes on the Severit...   \n",
              "4   Wearable Devices and COVID-19: State of the<b...   \n",
              "\n",
              "                                       journal  \\\n",
              "0               Appl Psychol Health Well Being   \n",
              "1                         Beilstein J Org Chem   \n",
              "2                                Qual Soc Work   \n",
              "3                             Diabetes Metab J   \n",
              "4  Emerging Technologies for Battling Covid-19   \n",
              "\n",
              "                                    abstract_summary  \n",
              "0   This study investigated how stress, physical<...  \n",
              "1                                      Not provided.  \n",
              "2   My emotional responses to this moment include...  \n",
              "3                                      Not provided.  \n",
              "4   In the current century, the novel coronavirus...  "
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "from tqdm import tqdm\n",
        "\n",
        "def get_breaks(content, length):\n",
        "    data = \"\"\n",
        "    words = content.split(' ')\n",
        "    total_chars = 0\n",
        "\n",
        "    # add break every length characters\n",
        "    for i in range(len(words)):\n",
        "        total_chars += len(words[i])\n",
        "        if total_chars > length:\n",
        "            data = data + \"<br>\" + words[i]\n",
        "            total_chars = 0\n",
        "        else:\n",
        "            data = data + \" \" + words[i]\n",
        "    return data\n",
        "\n",
        "\n",
        "dict_ = {'paper_id': [], 'doi':[], 'abstract': [], 'body_text': [],\n",
        "         'authors': [], 'title': [], 'journal': [], 'abstract_summary': []}\n",
        "\n",
        "\n",
        "for idx, entry in tqdm(enumerate(json_files), total=len(json_files)):\n",
        "    try:\n",
        "        content = FileReader(entry)\n",
        "    except Exception as e:\n",
        "        continue  # invalid paper format, skip\n",
        "\n",
        "    # get metadata information\n",
        "    meta_data = meta_df.loc[meta_df['sha'] == content.paper_id]\n",
        "    # no metadata, skip this paper\n",
        "    if len(meta_data) == 0:\n",
        "        continue\n",
        "    if len(content.body_text) == 0:\n",
        "        continue\n",
        "    dict_['abstract'].append(content.abstract)\n",
        "    dict_['paper_id'].append(content.paper_id)\n",
        "    dict_['body_text'].append(content.body_text)\n",
        "    # also create a column for the summary of abstract to be used in a plot\n",
        "    if len(content.abstract) == 0:\n",
        "        # no abstract provided\n",
        "        dict_['abstract_summary'].append(\"Not provided.\")\n",
        "    elif len(content.abstract.split(' ')) > 100:\n",
        "        # abstract provided is too long for plot, take first 300 words append with ...\n",
        "        info = content.abstract.split(' ')[:100]\n",
        "        summary = get_breaks(' '.join(info), 40)\n",
        "        dict_['abstract_summary'].append(summary + \"...\")\n",
        "    else:\n",
        "        # abstract is short enough\n",
        "        summary = get_breaks(content.abstract, 40)\n",
        "        dict_['abstract_summary'].append(summary)\n",
        "\n",
        "    # get metadata information\n",
        "    meta_data = meta_df.loc[meta_df['sha'] == content.paper_id]\n",
        "\n",
        "    try:\n",
        "        # if more than one author\n",
        "        authors = meta_data['authors'].values[0].split(';')\n",
        "        if len(authors) > 2:\n",
        "            # more than 2 authors, may be problem when plotting, so take first 2 append with ...\n",
        "            dict_['authors'].append(get_breaks('. '.join(authors), 40))\n",
        "        else:\n",
        "            # authors will fit in plot\n",
        "            dict_['authors'].append(\". \".join(authors))\n",
        "    except Exception as e:\n",
        "        # if only one author - or Null valie\n",
        "        dict_['authors'].append(meta_data['authors'].values[0])\n",
        "\n",
        "    # add the title information, add breaks when needed\n",
        "    try:\n",
        "        title = get_breaks(meta_data['title'].values[0], 40)\n",
        "        dict_['title'].append(title)\n",
        "    # if title was not provided\n",
        "    except Exception as e:\n",
        "        dict_['title'].append(meta_data['title'].values[0])\n",
        "\n",
        "    # add the journal information\n",
        "    dict_['journal'].append(meta_data['journal'].values[0])\n",
        "\n",
        "    # add doi\n",
        "    dict_['doi'].append(meta_data['doi'].values[0])\n",
        "\n",
        "\n",
        "df_covid = pd.DataFrame(dict_, columns=['paper_id', 'doi', 'abstract', 'body_text',\n",
        "                                        'authors', 'title', 'journal', 'abstract_summary'])\n",
        "df_covid.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "id": "2c2974a3",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2c2974a3",
        "outputId": "9bc51e71-1d4d-4404-cb1d-b71b658b9349"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 9022 entries, 0 to 9021\n",
            "Data columns (total 8 columns):\n",
            " #   Column            Non-Null Count  Dtype \n",
            "---  ------            --------------  ----- \n",
            " 0   paper_id          9022 non-null   object\n",
            " 1   doi               8599 non-null   object\n",
            " 2   abstract          9022 non-null   object\n",
            " 3   body_text         9022 non-null   object\n",
            " 4   authors           9009 non-null   object\n",
            " 5   title             9022 non-null   object\n",
            " 6   journal           8095 non-null   object\n",
            " 7   abstract_summary  9022 non-null   object\n",
            "dtypes: object(8)\n",
            "memory usage: 564.0+ KB\n"
          ]
        }
      ],
      "source": [
        "df_covid.info()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "id": "4f9af672",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4f9af672",
        "outputId": "eceec098-206e-4fe8-d85b-f848a93e4720"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "Index: 8070 entries, 0 to 9020\n",
            "Data columns (total 8 columns):\n",
            " #   Column            Non-Null Count  Dtype \n",
            "---  ------            --------------  ----- \n",
            " 0   paper_id          8070 non-null   object\n",
            " 1   doi               8070 non-null   object\n",
            " 2   abstract          8070 non-null   object\n",
            " 3   body_text         8070 non-null   object\n",
            " 4   authors           8070 non-null   object\n",
            " 5   title             8070 non-null   object\n",
            " 6   journal           8070 non-null   object\n",
            " 7   abstract_summary  8070 non-null   object\n",
            "dtypes: object(8)\n",
            "memory usage: 567.4+ KB\n"
          ]
        }
      ],
      "source": [
        "df = df_covid\n",
        "df.dropna(inplace=True)\n",
        "df.info()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "id": "89d3b3b8",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "89d3b3b8",
        "outputId": "20fe501a-4c47-4f93-8e8b-b441089f2862"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: langdetect in /usr/local/lib/python3.10/dist-packages (1.0.9)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from langdetect) (1.16.0)\n"
          ]
        }
      ],
      "source": [
        "!pip install langdetect"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "id": "f2eacd72",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f2eacd72",
        "outputId": "0b8d3e3f-4841-4e5f-fd23-55297ac27fb6"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "100%|██████████| 8070/8070 [01:08<00:00, 117.20it/s]\n"
          ]
        }
      ],
      "source": [
        "from tqdm import tqdm\n",
        "from langdetect import detect\n",
        "from langdetect import DetectorFactory\n",
        "\n",
        "# set seed\n",
        "DetectorFactory.seed = 0\n",
        "\n",
        "# hold label - language\n",
        "languages = []\n",
        "\n",
        "# go through each text\n",
        "for ii in tqdm(range(0,len(df))):\n",
        "    # split by space into list, take the first x intex, join with space\n",
        "    text = df.iloc[ii]['body_text'].split(\" \")\n",
        "\n",
        "    lang = \"en\"\n",
        "    try:\n",
        "        if len(text) > 50:\n",
        "            lang = detect(\" \".join(text[:50]))\n",
        "        elif len(text) > 0:\n",
        "            lang = detect(\" \".join(text[:len(text)]))\n",
        "    # ught... beginning of the document was not in a good format\n",
        "    except Exception as e:\n",
        "        all_words = set(text)\n",
        "        try:\n",
        "            lang = detect(\" \".join(all_words))\n",
        "        # what!! :( let's see if we can find any text in abstract...\n",
        "        except Exception as e:\n",
        "\n",
        "            try:\n",
        "                # let's try to label it through the abstract then\n",
        "                lang = detect(df.iloc[ii]['abstract_summary'])\n",
        "            except Exception as e:\n",
        "                lang = \"unknown\"\n",
        "                pass\n",
        "\n",
        "    # get the language\n",
        "    languages.append(lang)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "id": "7e7396f0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7e7396f0",
        "outputId": "d54ed418-c5dd-4cb3-b9ab-2ed07790843a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Total: 8070\n",
            "\n",
            "{'de': 20, 'en': 8028, 'es': 13, 'fr': 6, 'id': 1, 'nl': 2}\n"
          ]
        }
      ],
      "source": [
        "from pprint import pprint\n",
        "\n",
        "languages_dict = {}\n",
        "for lang in set(languages):\n",
        "    languages_dict[lang] = languages.count(lang)\n",
        "\n",
        "print(\"Total: {}\\n\".format(len(languages)))\n",
        "pprint(languages_dict)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "id": "e667fa35",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "e667fa35",
        "outputId": "38d0a17b-79c7-42de-be9b-e02922c23f11"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "Index: 8028 entries, 0 to 9020\n",
            "Data columns (total 9 columns):\n",
            " #   Column            Non-Null Count  Dtype \n",
            "---  ------            --------------  ----- \n",
            " 0   paper_id          8028 non-null   object\n",
            " 1   doi               8028 non-null   object\n",
            " 2   abstract          8028 non-null   object\n",
            " 3   body_text         8028 non-null   object\n",
            " 4   authors           8028 non-null   object\n",
            " 5   title             8028 non-null   object\n",
            " 6   journal           8028 non-null   object\n",
            " 7   abstract_summary  8028 non-null   object\n",
            " 8   language          8028 non-null   object\n",
            "dtypes: object(9)\n",
            "memory usage: 627.2+ KB\n"
          ]
        }
      ],
      "source": [
        "df['language'] = languages\n",
        "df = df[df['language'] == 'en']\n",
        "df.info()"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "ad4bf462",
      "metadata": {},
      "source": [
        "# Histogram of year / journal"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "5b707bb3",
      "metadata": {},
      "outputs": [],
      "source": [
        "# Enhanced code for histogram of publication years\n",
        "# Original: None\n",
        "\n",
        "# Convert publish_time to datetime\n",
        "meta_df['publish_time'] = pd.to_datetime(meta_df['publish_time'], errors='coerce')\n",
        "\n",
        "# Plot histogram of publication years\n",
        "plt.figure(figsize=(12, 6))\n",
        "meta_df['publish_time'].dt.year.value_counts().sort_index().plot(kind='bar')\n",
        "plt.title('Histogram of Publication Years')\n",
        "plt.xlabel('Year')\n",
        "plt.ylabel('Number of Papers')\n",
        "plt.show()\n",
        "\n",
        "# Enhanced code for histogram of journals\n",
        "# Original: None\n",
        "\n",
        "# Plot histogram of journals\n",
        "plt.figure(figsize=(12, 6))\n",
        "df['journal'].value_counts().head(20).plot(kind='bar')\n",
        "plt.title('Top 20 Journals by Number of Papers')\n",
        "plt.xlabel('Journal')\n",
        "plt.ylabel('Number of Papers')\n",
        "plt.xticks(rotation=90)\n",
        "plt.show()\n"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "cae116c4",
      "metadata": {},
      "source": [
        "# Map-Reduce "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "1d54ec16",
      "metadata": {},
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "id": "ac547210",
      "metadata": {},
      "source": [
        "# Association Analysis"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "f3b12c03",
      "metadata": {},
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "id": "48ec6f2e",
      "metadata": {},
      "source": [
        "# Similarity Analysis"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "4823d5b4",
      "metadata": {},
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "id": "adc2e25a",
      "metadata": {},
      "source": [
        "# Clustering Analysis"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.9.16"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}
