{
  "cells": [
    {
      "cell_type": "markdown",
      "id": "d429ccc6",
      "metadata": {
        "id": "d429ccc6"
      },
      "source": [
        "# About the Dataset\n",
        "\n",
        "The original CORD-19 is a resource of over 1,000,000 scholarly articles, including over 400,000 with full text, about COVID-19, SARS-CoV-2, and related coronaviruses.\n",
        "\n",
        "In our project, the dataset is sampled from the CORD-19 with size ~10,000 to reduce computation burden."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "dcd57c91-923e-4682-96a2-e1ee4eb9ef71",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dcd57c91-923e-4682-96a2-e1ee4eb9ef71",
        "outputId": "caa80b6d-c68d-4075-f21f-238c156c7aa7"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n",
            "/content\n",
            "meta_10k.csv  subset  subset.zip\n",
            "Directory /content/CORD_19_subset already exists. Skipping extraction.\n"
          ]
        }
      ],
      "source": [
        "import os\n",
        "import subprocess\n",
        "\n",
        "# shared link: https://drive.google.com/drive/folders/1Td_ZTUVrsKeftDE5Zll7252YLJdWiNTk?usp=share_link\n",
        "# you can download the data via the shared link, and skip Step 0 and Step 1 if you want to run the code in your local machine\n",
        "\n",
        "\n",
        "# Step 0: add the shared folder to your google drive. e.g., /content/drive/MyDrive/CORD_19\n",
        "\n",
        "# Step 1: Mount Google Drive\n",
        "from google.colab import drive\n",
        "drive.mount(\"/content/drive\")\n",
        "\n",
        "\n",
        "!echo $PWD\n",
        "\n",
        "!ls /content/drive/MyDrive/CORD_19/\n",
        "\n",
        "# Step 2: unzip json files\n",
        "subset_dir = os.path.join(os.getcwd(),  \"CORD_19_subset\")\n",
        "\n",
        "\n",
        "zip_file_path=\"/content/drive/MyDrive/CORD_19/subset.zip\"\n",
        "\n",
        "# Check if the destination directory exists\n",
        "if not os.path.exists(subset_dir):\n",
        "    # Unzip the file\n",
        "    cmd = \"unzip {} -d {}\".format(zip_file_path, subset_dir)\n",
        "    proc = subprocess.Popen(cmd, shell=True)\n",
        "else:\n",
        "    print(f\"Directory {subset_dir} already exists. Skipping extraction.\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "SxWWPxCDjHF5",
      "metadata": {
        "id": "SxWWPxCDjHF5"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "eeabd1eb",
      "metadata": {
        "id": "eeabd1eb"
      },
      "outputs": [],
      "source": [
        "# import packages\n",
        "\n",
        "\n",
        "import os\n",
        "import json\n",
        "import glob\n",
        "import numpy as np\n",
        "import pandas as pd\n",
        "\n",
        "import matplotlib.pyplot as plt\n"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "cc1250a1",
      "metadata": {
        "id": "cc1250a1"
      },
      "source": [
        "# Data Load & Pre-processing"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "83d2222c",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 677
        },
        "id": "83d2222c",
        "outputId": "970ba990-f59f-4974-8a81-113f1d4d597d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "9022\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   Unnamed: 0  cord_uid                                       sha source_x  \\\n",
              "0          44  m71xkuo9  c6bf372c094f035a514975c35a7f9c094abbe493      PMC   \n",
              "1          96  hqc7u9w3  c65f0939cf35a0f04bf93bd6e8f771b8521563a5      PMC   \n",
              "2         217  jzwcy7dr  a009c8efa4c5f13a5e604608d4e33e1dac078044      PMC   \n",
              "3         255  02opdk0m  b411e12b20d883ef2ee5ca19d48eff9fccedf05f      PMC   \n",
              "4         342  094d0rn6  513d5ea4db4eb8e94c14c46b018c6041d78119cf      PMC   \n",
              "\n",
              "                                               title  \\\n",
              "0  Sequence specific visual detection of LAMP rea...   \n",
              "1  Transmission Parameters of the 2001 Foot and M...   \n",
              "2  Results From a Hypothesis Generating Case-Cont...   \n",
              "3  CVTree update: a newly designed phylogenetic s...   \n",
              "4  IPS-1 Is Essential for the Control of West Nil...   \n",
              "\n",
              "                            doi       pmcid pubmed_id    license  \\\n",
              "0         10.1186/1472-6750-6-3  PMC1373654  16401354      cc-by   \n",
              "1  10.1371/journal.pone.0000502  PMC1876810  17551582      cc-by   \n",
              "2         10.1093/schbul/sbm139  PMC2632504  18156638  bronze-oa   \n",
              "3            10.1093/nar/gkp278  PMC2703908  19398429   cc-by-nc   \n",
              "4  10.1371/journal.ppat.1000757  PMC2816698  20140199      cc-by   \n",
              "\n",
              "                                            abstract publish_time  \\\n",
              "0  BACKGROUND: Development of a practical gene po...   2006-01-10   \n",
              "1  Despite intensive ongoing research, key aspect...   2007-06-06   \n",
              "2  Background: Herpes family viruses can cause ce...   2008-08-20   \n",
              "3  The CVTree web server (http://tlife.fudan.edu....   2009-07-01   \n",
              "4  The innate immune response is essential for co...   2010-02-05   \n",
              "\n",
              "                                             authors                 journal  \\\n",
              "0  Mori, Yasuyoshi; Hirano, Tsuyoshi; Notomi, Tsu...          BMC Biotechnol   \n",
              "1                Chis Ster, Irina; Ferguson, Neil M.                PLoS One   \n",
              "2  Niebuhr, David W.; Millikan, Amy M.; Yolken, R...  Schizophrenia Bulletin   \n",
              "3                              Xu, Zhao; Hao, Bailin       Nucleic Acids Res   \n",
              "4  Suthar, Mehul S.; Ma, Daphne Y.; Thomas, Sunil...             PLoS Pathog   \n",
              "\n",
              "   mag_id  who_covidence_id arxiv_id  \\\n",
              "0     NaN               NaN      NaN   \n",
              "1     NaN               NaN      NaN   \n",
              "2     NaN               NaN      NaN   \n",
              "3     NaN               NaN      NaN   \n",
              "4     NaN               NaN      NaN   \n",
              "\n",
              "                                      pdf_json_files  \\\n",
              "0  document_parses/pdf_json/c6bf372c094f035a51497...   \n",
              "1  document_parses/pdf_json/c65f0939cf35a0f04bf93...   \n",
              "2  document_parses/pdf_json/a009c8efa4c5f13a5e604...   \n",
              "3  document_parses/pdf_json/b411e12b20d883ef2ee5c...   \n",
              "4  document_parses/pdf_json/513d5ea4db4eb8e94c14c...   \n",
              "\n",
              "                                 pmc_json_files  \\\n",
              "0  document_parses/pmc_json/PMC1373654.xml.json   \n",
              "1  document_parses/pmc_json/PMC1876810.xml.json   \n",
              "2  document_parses/pmc_json/PMC2632504.xml.json   \n",
              "3  document_parses/pmc_json/PMC2703908.xml.json   \n",
              "4  document_parses/pmc_json/PMC2816698.xml.json   \n",
              "\n",
              "                                                 url  s2_id  \n",
              "0  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...    NaN  \n",
              "1  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...    NaN  \n",
              "2  https://academic.oup.com/schizophreniabulletin...    NaN  \n",
              "3  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...    NaN  \n",
              "4  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...    NaN  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-75902fa7-1c8c-479d-8961-c90fe9b3ce91\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Unnamed: 0</th>\n",
              "      <th>cord_uid</th>\n",
              "      <th>sha</th>\n",
              "      <th>source_x</th>\n",
              "      <th>title</th>\n",
              "      <th>doi</th>\n",
              "      <th>pmcid</th>\n",
              "      <th>pubmed_id</th>\n",
              "      <th>license</th>\n",
              "      <th>abstract</th>\n",
              "      <th>publish_time</th>\n",
              "      <th>authors</th>\n",
              "      <th>journal</th>\n",
              "      <th>mag_id</th>\n",
              "      <th>who_covidence_id</th>\n",
              "      <th>arxiv_id</th>\n",
              "      <th>pdf_json_files</th>\n",
              "      <th>pmc_json_files</th>\n",
              "      <th>url</th>\n",
              "      <th>s2_id</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>44</td>\n",
              "      <td>m71xkuo9</td>\n",
              "      <td>c6bf372c094f035a514975c35a7f9c094abbe493</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Sequence specific visual detection of LAMP rea...</td>\n",
              "      <td>10.1186/1472-6750-6-3</td>\n",
              "      <td>PMC1373654</td>\n",
              "      <td>16401354</td>\n",
              "      <td>cc-by</td>\n",
              "      <td>BACKGROUND: Development of a practical gene po...</td>\n",
              "      <td>2006-01-10</td>\n",
              "      <td>Mori, Yasuyoshi; Hirano, Tsuyoshi; Notomi, Tsu...</td>\n",
              "      <td>BMC Biotechnol</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/c6bf372c094f035a51497...</td>\n",
              "      <td>document_parses/pmc_json/PMC1373654.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>96</td>\n",
              "      <td>hqc7u9w3</td>\n",
              "      <td>c65f0939cf35a0f04bf93bd6e8f771b8521563a5</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Transmission Parameters of the 2001 Foot and M...</td>\n",
              "      <td>10.1371/journal.pone.0000502</td>\n",
              "      <td>PMC1876810</td>\n",
              "      <td>17551582</td>\n",
              "      <td>cc-by</td>\n",
              "      <td>Despite intensive ongoing research, key aspect...</td>\n",
              "      <td>2007-06-06</td>\n",
              "      <td>Chis Ster, Irina; Ferguson, Neil M.</td>\n",
              "      <td>PLoS One</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/c65f0939cf35a0f04bf93...</td>\n",
              "      <td>document_parses/pmc_json/PMC1876810.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>217</td>\n",
              "      <td>jzwcy7dr</td>\n",
              "      <td>a009c8efa4c5f13a5e604608d4e33e1dac078044</td>\n",
              "      <td>PMC</td>\n",
              "      <td>Results From a Hypothesis Generating Case-Cont...</td>\n",
              "      <td>10.1093/schbul/sbm139</td>\n",
              "      <td>PMC2632504</td>\n",
              "      <td>18156638</td>\n",
              "      <td>bronze-oa</td>\n",
              "      <td>Background: Herpes family viruses can cause ce...</td>\n",
              "      <td>2008-08-20</td>\n",
              "      <td>Niebuhr, David W.; Millikan, Amy M.; Yolken, R...</td>\n",
              "      <td>Schizophrenia Bulletin</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/a009c8efa4c5f13a5e604...</td>\n",
              "      <td>document_parses/pmc_json/PMC2632504.xml.json</td>\n",
              "      <td>https://academic.oup.com/schizophreniabulletin...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>255</td>\n",
              "      <td>02opdk0m</td>\n",
              "      <td>b411e12b20d883ef2ee5ca19d48eff9fccedf05f</td>\n",
              "      <td>PMC</td>\n",
              "      <td>CVTree update: a newly designed phylogenetic s...</td>\n",
              "      <td>10.1093/nar/gkp278</td>\n",
              "      <td>PMC2703908</td>\n",
              "      <td>19398429</td>\n",
              "      <td>cc-by-nc</td>\n",
              "      <td>The CVTree web server (http://tlife.fudan.edu....</td>\n",
              "      <td>2009-07-01</td>\n",
              "      <td>Xu, Zhao; Hao, Bailin</td>\n",
              "      <td>Nucleic Acids Res</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/b411e12b20d883ef2ee5c...</td>\n",
              "      <td>document_parses/pmc_json/PMC2703908.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>342</td>\n",
              "      <td>094d0rn6</td>\n",
              "      <td>513d5ea4db4eb8e94c14c46b018c6041d78119cf</td>\n",
              "      <td>PMC</td>\n",
              "      <td>IPS-1 Is Essential for the Control of West Nil...</td>\n",
              "      <td>10.1371/journal.ppat.1000757</td>\n",
              "      <td>PMC2816698</td>\n",
              "      <td>20140199</td>\n",
              "      <td>cc-by</td>\n",
              "      <td>The innate immune response is essential for co...</td>\n",
              "      <td>2010-02-05</td>\n",
              "      <td>Suthar, Mehul S.; Ma, Daphne Y.; Thomas, Sunil...</td>\n",
              "      <td>PLoS Pathog</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>document_parses/pdf_json/513d5ea4db4eb8e94c14c...</td>\n",
              "      <td>document_parses/pmc_json/PMC2816698.xml.json</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-75902fa7-1c8c-479d-8961-c90fe9b3ce91')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-75902fa7-1c8c-479d-8961-c90fe9b3ce91 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-75902fa7-1c8c-479d-8961-c90fe9b3ce91');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-e18cacbb-9a7c-46da-afbb-f58707fa229d\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-e18cacbb-9a7c-46da-afbb-f58707fa229d')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-e18cacbb-9a7c-46da-afbb-f58707fa229d button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "meta_df",
              "summary": "{\n  \"name\": \"meta_df\",\n  \"rows\": 9022,\n  \"fields\": [\n    {\n      \"column\": \"Unnamed: 0\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 337241,\n        \"min\": 44,\n        \"max\": 1056551,\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          811225,\n          732051,\n          763638\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cord_uid\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"h3uaj6fx\",\n          \"yz1bshv2\",\n          \"p630d49d\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sha\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"e031a4fcf00577dcc7584f7a456f83394b61ec1b\",\n          \"9ce761ee59fd4b5bc7fda2d3c51b89c6be54360b\",\n          \"e114858a40b22f7ff3ac74ab9b97cecc0ee36761\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"source_x\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 21,\n        \"samples\": [\n          \"PMC\",\n          \"BioRxiv; Medline; WHO\",\n          \"ArXiv; Medline\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"Promise and peril-defining ethical telehealth practice from the clinician and patient perspective: A qualitative study\",\n          \"DeSUMOylation of Apoptosis Inhibitor 5 by Avibirnavirus VP3 Supports Virus Replication\",\n          \"Emergent procedures for oropharyngeal bleeding during the COVID-19 pandemic: Protection of medical staff\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"doi\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8599,\n        \"samples\": [\n          \"10.1038/s41598-021-91817-9\",\n          \"10.5334/gh.786\",\n          \"10.1038/s41578-021-00305-z\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pmcid\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7942,\n        \"samples\": [\n          \"PMC3573664\",\n          \"PMC7789084\",\n          \"PMC7729151\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pubmed_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 6986,\n        \"samples\": [\n          \"35165656\",\n          \"35110546\",\n          \"33538335\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"license\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 13,\n        \"samples\": [\n          \"els-covid\",\n          \"medrxiv\",\n          \"cc-by\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8285,\n        \"samples\": [\n          \"The variety and extent of non-pharmaceutical measures implemented by the government to control COVID-19 in Argentina were exceptional, making this country the best example to analyze the evolution of COVID-19 under the most stringent and longerlasting restrictive policies- which up to May 31(st) 2021 included 119 days of strict nation-wide lock-down, 304 days of less restrictive lock-downs, and 35 days of curfews. Two of the three peaks of infection correlated with the germicidal solar flux received in Argentina, suggesting a seasonal component and a role for the virus persisting in the environment. A massive public gathering crowding the presidential square in Buenos Aires, during which nearly half of those present were without face masks, did not alter the infection curve in that city. Comparative epidemiological data (standardized per million inhabitants) shows that COVID-19 in Uruguay, a neighboring country whose capital is at a similar latitude than Buenos Aires and who did not mandate lock-downs or curfews, progressed much slower (until vaccination started) than in Argentina. The number of yearly deaths caused by respiratory diseases and influenza in Argentina before the pandemic was similar to the total number of deaths attributed to COVID-19 cumulated on April 25, 2021, more than a year after the pandemic started. The failure to detect any benefit on ameliorating COVID-19 by the long and strict nation-wide lock-downs in Argentina should raise world-wide concerns about mandating costly and ineffective restrictive measures during ongoing or future pandemics.\",\n          \"The world is currently facing a serious coronavirus pandemic (COVID-19). This novel coronavirus is a new virus for which effective drugs and vaccines have not yet been developed. Amateur and professional athletic events around the world have been stopped, and teams and athletes have had to adopt social withdrawal measures, interrupting their training and preparation routines for competitions. Given this alarming scenario, this article aims to alert athletes to the importance of maintaining a conditioning routine during this confinement period.\",\n          \"Bivalve molluscan shellfish like clams and oysters, etc., are capable to bioaccumulate surrounding contaminants from waters into their digestive systems and posing serious threats of food poisoning. Detection of rotaviruses (RVs) in shellfish is of particular importance because RVs are prone to genome reassortment resulting in the emergence of new RV variants that may compromise vaccine safety. Herein, we have detected the wild-type RVs and Rotarix/RotaTeq vaccine strains in freshwater clams collected on the riverside, Kawasaki city, from July 2019 to January 2020 and correlated the detected genotypes with that of gastroenteritis cases of nearby clinics to understand the transmission of RVs in the environment. The wild-type RVs were detected in 62 (64.6%) out of 96 freshwater clams in every study month: July, September, November, and January that are considered as off-season for RV infections. The most frequent genotypes were G2 (42.9%), G8 (28.6%), G3 (14.3%), G1 (7.1%), and G10 (7.1%), which remained comparable with genotypic distribution found in the clinical samples over the last few years indicating that these RVs may accumulate in clams since a long time. However, G10 genotype was detected in clam but not in clinical samples suggesting the presence of asymptomatic infection or RVs could be carried out from a long distance. Importantly, vaccine strains, RotaTeq (1%) but not Rotarix (0%), were also detected in a clam. Attention must be paid to monitoring the potential transmission of wild-type and vaccine RV strains in the environment to prevent the emergence of new variants generated from genome reassortment with vaccine strains.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"publish_time\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1864,\n        \"samples\": [\n          \"2020-01-02\",\n          \"2022-03-09\",\n          \"2016-04-04\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8997,\n        \"samples\": [\n          \"Marchetti, Federico; Verazza, Sara; Brambilla, Margherita; Restivo, Vincenzo\",\n          \"Kaeberlein, Matt; Tyler, Jessica K\",\n          \"Atlante, Sandra; Mongelli, Alessia; Barbi, Veronica; Martelli, Fabio; Farsetti, Antonella; Gaetano, Carlo\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"journal\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2785,\n        \"samples\": [\n          \"Adv Med\",\n          \"Microbial Forensics\",\n          \"Appl Biochem Biotechnol\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"mag_id\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"who_covidence_id\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"arxiv_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 480,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pdf_json_files\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pmc_json_files\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7414,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"s2_id\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 61913642.79275464,\n        \"min\": 15600.0,\n        \"max\": 249148763.0,\n        \"num_unique_values\": 7468,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 148
        }
      ],
      "source": [
        "# Load Meta data from meta_10k.csv\n",
        "data_root = '/content/drive/MyDrive/CORD_19/'\n",
        "\n",
        "metadata_path = os.path.join(data_root, 'meta_10k.csv')\n",
        "meta_df = pd.read_csv(metadata_path, dtype={\n",
        "    'pubmed_id': str,\n",
        "    'Microsoft Academic Paper ID': str,\n",
        "    'doi': str\n",
        "})\n",
        "\n",
        "print(len(meta_df))\n",
        "meta_df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "e1b611f0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "e1b611f0",
        "outputId": "45fab904-a988-4b20-d245-ac754f80375b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 9022 entries, 0 to 9021\n",
            "Data columns (total 20 columns):\n",
            " #   Column            Non-Null Count  Dtype  \n",
            "---  ------            --------------  -----  \n",
            " 0   Unnamed: 0        9022 non-null   int64  \n",
            " 1   cord_uid          9022 non-null   object \n",
            " 2   sha               9022 non-null   object \n",
            " 3   source_x          9022 non-null   object \n",
            " 4   title             9022 non-null   object \n",
            " 5   doi               8599 non-null   object \n",
            " 6   pmcid             7942 non-null   object \n",
            " 7   pubmed_id         6986 non-null   object \n",
            " 8   license           9022 non-null   object \n",
            " 9   abstract          8289 non-null   object \n",
            " 10  publish_time      9022 non-null   object \n",
            " 11  authors           9009 non-null   object \n",
            " 12  journal           8095 non-null   object \n",
            " 13  mag_id            0 non-null      float64\n",
            " 14  who_covidence_id  0 non-null      float64\n",
            " 15  arxiv_id          480 non-null    object \n",
            " 16  pdf_json_files    9022 non-null   object \n",
            " 17  pmc_json_files    7414 non-null   object \n",
            " 18  url               9022 non-null   object \n",
            " 19  s2_id             7469 non-null   float64\n",
            "dtypes: float64(3), int64(1), object(16)\n",
            "memory usage: 1.4+ MB\n"
          ]
        }
      ],
      "source": [
        "meta_df.info()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "c3139be0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c3139be0",
        "outputId": "d14a2fdc-a7d4-4208-b053-ea74de8cf824"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/CORD_19_subset/subset/document_parses/pdf_json/\n",
            "total json files: 12000\n"
          ]
        }
      ],
      "source": [
        "def glob_files(path, f_type=\".json\"):\n",
        "    dst = []\n",
        "    for root, _, files in os.walk(path):\n",
        "        for f in files:\n",
        "            if f.endswith(f_type):\n",
        "                dst.append(os.path.join(root, f))\n",
        "    return dst\n",
        "\n",
        "# glob json files\n",
        "json_dir = os.path.join(subset_dir, \"subset/document_parses/pdf_json/\")\n",
        "print(json_dir)\n",
        "json_files = glob_files(json_dir, \".json\")\n",
        "\n",
        "print(\"total json files:\", len(json_files))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "20fb57d2",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "20fb57d2",
        "outputId": "b83e9bd7-f7ec-4034-db5e-9bfaf6f52d8f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "3184fc47aab4ea35ab911413018a3a86c56368f1: COVID-19 pneumonia and an indelible ground-glass nodule : ... A 59-year-old man who complained of fever, general malaise, and olfactory and taste disorders was admitted to hospital five days after onset. Coronavirus disease 2019 (COVID-19) pneumonia was diagnose...\n"
          ]
        }
      ],
      "source": [
        "class FileReader:\n",
        "    def __init__(self, file_path):\n",
        "        with open(file_path) as file:\n",
        "            content = json.load(file)\n",
        "            self.paper_id = content['paper_id']\n",
        "            self.abstract = []\n",
        "            self.body_text = []\n",
        "            # Abstract\n",
        "            for entry in content['abstract']:\n",
        "                self.abstract.append(entry['text'])\n",
        "            # Body text\n",
        "            for entry in content['body_text']:\n",
        "                self.body_text.append(entry['text'])\n",
        "            self.abstract = '\\n'.join(self.abstract)\n",
        "            self.body_text = '\\n'.join(self.body_text)\n",
        "\n",
        "            self.title = content['metadata']['title']\n",
        "\n",
        "            #dict_keys(['paper_id', 'metadata', 'abstract', 'body_text',\n",
        "            #'bib_entries', 'ref_entries', 'back_matter'])\n",
        "\n",
        "\n",
        "    def __repr__(self):\n",
        "        return f\"{self.paper_id}: {self.title } : {self.abstract[:200]}... {self.body_text[:200]}...\"\n",
        "\n",
        "\n",
        "first_row = FileReader(json_files[0])\n",
        "print(first_row)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "96a78cce",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 484
        },
        "id": "96a78cce",
        "outputId": "2658442d-1ac5-4976-c332-c65233267a3d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "100%|██████████| 12000/12000 [01:05<00:00, 184.15it/s]\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                   paper_id                        doi  \\\n",
              "0  3184fc47aab4ea35ab911413018a3a86c56368f1           10.1002/rcr2.751   \n",
              "1  97079288bf8a9afb1ef3ee088135df630b2ec2b9  10.1017/s0714980820000343   \n",
              "2  4dfc59a66e7a86f6a40c5fafad4c12f483ce40bc   10.3389/fimmu.2012.00378   \n",
              "3  4bec44342ee1111591970c04aadcff79275981b6         10.4317/jced.58776   \n",
              "4  6fa0f00ec5cbe4b9b113408885e5c84898176143  10.3389/fphys.2021.723224   \n",
              "\n",
              "                                            abstract  \\\n",
              "0                                                      \n",
              "1  This article describes priority areas for rese...   \n",
              "2  Pentraxin 3 (PTX3) is a soluble pattern recogn...   \n",
              "3  Background: With the spread of the COVID-19 vi...   \n",
              "4  In this article we discuss the concept of acce...   \n",
              "\n",
              "                                           body_text  \\\n",
              "0  A 59-year-old man who complained of fever, gen...   \n",
              "1  changements, il devint évident que les personn...   \n",
              "2  The release of neutrophil extracellular traps ...   \n",
              "3  The COVID-19 pandemic imposed challenges for h...   \n",
              "4  Cholesterol is the dominant sterol in animal c...   \n",
              "\n",
              "                                             authors  \\\n",
              "0   Yamanaka, Sumitaka.  Ota, Shinichiro. <br>Yos...   \n",
              "1   Rylett, R. Jane.  Alary, Flamine.  Goldberg,<...   \n",
              "2                     Daigo, Kenji.  Hamakubo, Takao   \n",
              "3   Garcés-Elías, María-Claudia.  León-Manco,<br>...   \n",
              "4                Griffiths, William J..  Wang, Yuqin   \n",
              "\n",
              "                                               title  \\\n",
              "0   COVID‐19 pneumonia and an indelible<br>ground...   \n",
              "1   La COVID-19 et les priorités de recherche sur...   \n",
              "2   Host-protective effect of circulating<br>pent...   \n",
              "3   Impact of mandatory social isolation measures...   \n",
              "4   Sterols, Oxysterols, and Accessible<br>Choles...   \n",
              "\n",
              "                                             journal  \\\n",
              "0                                  Respirol Case Rep   \n",
              "1  Canadian journal on aging = La revue canadienn...   \n",
              "2                                      Front Immunol   \n",
              "3                                    J Clin Exp Dent   \n",
              "4                                      Front Physiol   \n",
              "\n",
              "                                    abstract_summary  \n",
              "0                                      Not provided.  \n",
              "1   This article describes priority areas for<br>...  \n",
              "2   Pentraxin 3 (PTX3) is a soluble pattern<br>re...  \n",
              "3   Background: With the spread of the COVID-19<b...  \n",
              "4   In this article we discuss the concept of<br>...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-0c93e462-6277-46bc-b8a6-8d49494fa7d8\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>paper_id</th>\n",
              "      <th>doi</th>\n",
              "      <th>abstract</th>\n",
              "      <th>body_text</th>\n",
              "      <th>authors</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>abstract_summary</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>3184fc47aab4ea35ab911413018a3a86c56368f1</td>\n",
              "      <td>10.1002/rcr2.751</td>\n",
              "      <td></td>\n",
              "      <td>A 59-year-old man who complained of fever, gen...</td>\n",
              "      <td>Yamanaka, Sumitaka.  Ota, Shinichiro. &lt;br&gt;Yos...</td>\n",
              "      <td>COVID‐19 pneumonia and an indelible&lt;br&gt;ground...</td>\n",
              "      <td>Respirol Case Rep</td>\n",
              "      <td>Not provided.</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>97079288bf8a9afb1ef3ee088135df630b2ec2b9</td>\n",
              "      <td>10.1017/s0714980820000343</td>\n",
              "      <td>This article describes priority areas for rese...</td>\n",
              "      <td>changements, il devint évident que les personn...</td>\n",
              "      <td>Rylett, R. Jane.  Alary, Flamine.  Goldberg,&lt;...</td>\n",
              "      <td>La COVID-19 et les priorités de recherche sur...</td>\n",
              "      <td>Canadian journal on aging = La revue canadienn...</td>\n",
              "      <td>This article describes priority areas for&lt;br&gt;...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>4dfc59a66e7a86f6a40c5fafad4c12f483ce40bc</td>\n",
              "      <td>10.3389/fimmu.2012.00378</td>\n",
              "      <td>Pentraxin 3 (PTX3) is a soluble pattern recogn...</td>\n",
              "      <td>The release of neutrophil extracellular traps ...</td>\n",
              "      <td>Daigo, Kenji.  Hamakubo, Takao</td>\n",
              "      <td>Host-protective effect of circulating&lt;br&gt;pent...</td>\n",
              "      <td>Front Immunol</td>\n",
              "      <td>Pentraxin 3 (PTX3) is a soluble pattern&lt;br&gt;re...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>4bec44342ee1111591970c04aadcff79275981b6</td>\n",
              "      <td>10.4317/jced.58776</td>\n",
              "      <td>Background: With the spread of the COVID-19 vi...</td>\n",
              "      <td>The COVID-19 pandemic imposed challenges for h...</td>\n",
              "      <td>Garcés-Elías, María-Claudia.  León-Manco,&lt;br&gt;...</td>\n",
              "      <td>Impact of mandatory social isolation measures...</td>\n",
              "      <td>J Clin Exp Dent</td>\n",
              "      <td>Background: With the spread of the COVID-19&lt;b...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>6fa0f00ec5cbe4b9b113408885e5c84898176143</td>\n",
              "      <td>10.3389/fphys.2021.723224</td>\n",
              "      <td>In this article we discuss the concept of acce...</td>\n",
              "      <td>Cholesterol is the dominant sterol in animal c...</td>\n",
              "      <td>Griffiths, William J..  Wang, Yuqin</td>\n",
              "      <td>Sterols, Oxysterols, and Accessible&lt;br&gt;Choles...</td>\n",
              "      <td>Front Physiol</td>\n",
              "      <td>In this article we discuss the concept of&lt;br&gt;...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-0c93e462-6277-46bc-b8a6-8d49494fa7d8')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-0c93e462-6277-46bc-b8a6-8d49494fa7d8 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-0c93e462-6277-46bc-b8a6-8d49494fa7d8');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-8fef7a74-3ad0-4f3c-9bda-e32557011a74\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-8fef7a74-3ad0-4f3c-9bda-e32557011a74')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-8fef7a74-3ad0-4f3c-9bda-e32557011a74 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_covid",
              "summary": "{\n  \"name\": \"df_covid\",\n  \"rows\": 9022,\n  \"fields\": [\n    {\n      \"column\": \"paper_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"c6781e2907c505c15a8f82b97bcf4062f6301545\",\n          \"3bbffbd116ff1bb9a9ee3259c5af61546ebb4f01\",\n          \"1f0474d2e8dc4f5e06114e919512b4bcefcba410\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"doi\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8599,\n        \"samples\": [\n          \"10.1007/s11239-021-02477-5\",\n          \"10.3389/fpsyg.2021.651844\",\n          \"10.1371/journal.pone.0165326\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7281,\n        \"samples\": [\n          \"This study projects the economic impacts of COVID-19 outbreak on the Brazilian economy using a dynamic interregional computable general equilibrium model. We considered two scenarios. The first scenario has two channels of transmission over the economic system: a negative shock of labor supply due to the rates of morbidity and mortality caused by the pandemic, and a temporary shutdown of nonessential economic activities. The second scenario adds to the first the effects of the government fiscal package adopted to counteract the effects of COVID-19 on the economy. Furthermore, in both scenarios, a sensitive analysis related to the temporality of the shutdown is considered by assuming 3 and 6 months of shutdown.\\nThe results indicate a reduction of 3.78% in the national GDP growth rate in Scenario 1 and a reduction of 0.48% in Scenario 2, in 2020, with 3 months of shutdown. With 6 months, the reduction would be greater, 10.90% and 7.64% in Scenarios 1 and 2, respectively. Thus, the government fiscal stimulus considered in this study partially mitigates the reduction in GDP projected under the COVID-19 outbreak. The study also presents sectoral projections at the national and state levels. The estimates indicate reductions in the GDP of most of Brazilian states in both scenarios.\",\n          \"The management of comprehensive and extensive, multi-lateral and multi-national programmes and projects in the field of coastal and ocean management poses numerous organisational problems encompassing co-ordination between: the actions of individual participating countries at the regional level; the national level actions of institutions from different sectors; and actions that are designed to address issues as diverse as: biological diversity conservation and sustainable use; fisheries management; maritime transport; and the control of land based pollution. Most large multi-lateral projects focus on sound scientific knowledge and information, and pay less attention to the design of a management structure that will ensure coherence and co-ordination of the interventions once the programme or project is under implementation. The project entitled \\\"Reversing Environmental Degradation Trends in the South China Sea and Gulf of Thailand\\\" was complex since it addressed three priority areas of concern namely the loss and degradation of coastal habitats, over-exploitation of fisheries in the Gulf of Thailand, and land-based pollution. It is suggested that the success of the management framework reflects the following key design elements:\\nThe framework permitted and encouraged both \\\"horizontal\\\" (inter-country) and \\\"vertical\\\" (intra-country) interactions and networking between individuals at all levels of project implementation and execution; Inclusion of a body (the Regional Scientific and Technical Committee) that served as a forum for reconciling both sectorial and national interests and priorities; The clear separation between discussions of scientific and technical matters from discussions dealing with policy and principles at both the national and regional levels; The framework facilitated the incorporation of sound scientific and technical advice and information into politically based decision-making; The use of regional experts and consultants from the participating countries fostered \\\"ownership\\\" of the activities and outputs; Restriction of the membership of the Project Steering Committee to government representatives only, and exclusion of observers from regional and international agencies and institutions other than UNEP; and, The framework allowed for adaptive management and was not a rigid unchanging structure.\",\n          \"Citation: Lytton, S.D.; Yeasmin, M.; Ghosh, A.K.; Bulbul, M.R.H.; Molla, M.M.A.; Herr, M.; Duchmann, H.; Sharif, M.M.; Nafisa, T.; Amin, M.R.; et al.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"body_text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \"Approximately 40% of all adults in the USA are obese [1] . The prevalence of the most severe form of obesity [class III obesity: body mass index (BMI) > 40 kg/ m 2 ] is approaching 10% (> 30 million Americans) [2] . Little has been done in the intensive care unit (ICU) to study this healthcare epidemic, which is associated with overall reduced life expectancy [3] . A common cause of ICU admission for patients with class III obesity is acute respiratory distress syndrome (ARDS) [4] , often leading to dependency on mechanical ventilation, high incidence of tracheostomy [5] , severe kidney failure [6] , multiple organ failure, and significantly higher all-cause mortality [7, 8] .\\nAppropriate protective mechanical ventilation is the cornerstone for treatment of patients with ARDS [9] [10] [11] [12] [13] . To improve lung healing and survival, several ventilation strategies have been tested with different degrees of success. Positive end-expiratory pressure (PEEP) and lung recruitment maneuvers (LRM) are two ventilation strategies aimed to decrease overstretching of lung parenchyma and cyclic opening and closing of small airways and alveoli (i.e., barotrauma, volutrauma, and atelectrauma). Obesity has been an exclusion criterion in most of the major ARDS trials testing different modalities to titrate mechanical ventilation [9, 11, 14, 15] . Despite a lack of evidence of benefit to patients with obesity, most clinicians use the ARDSnet PEEP/inspiratory fraction of oxygen (F i O 2 ) protocol [9, 11] to titrate mechanical ventilation in all ARDS patients.\\nOur recent studies confirm that pleural pressure in patients with class III obesity is higher than in patients with lean body habitus [16] [17] [18] [19] . Increased pleural pressure significantly reduces lung volume (especially functional residual capacity) and leads to formation of atelectasis, which is associated with shunting and hypoxemia [17] [18] [19] . Patients with obesity often have highly recruitable lungs, and common PEEP levels used in the ICU are not sufficient to prevent atelectasis [17] [18] [19] . As a result, in this population, only an individualized physiological titration of PEEP with a LRM is effective to counter the detrimental effects of increased pleural pressure, resulting in lung re-expansion [17] [18] [19] .\\nWe tested the hypothesis that implementation of a specialized team (the lung rescue team) exclusively assessing patients with both obesity and ARDS would reduce mortality due to individualized physiologic treatment of such unique patients. To provide individualized titration of mechanical ventilation in patients with class III obesity, the Massachusetts General Hospital (MGH) Respiratory Care Service together with the Critical Care Group implemented a lung rescue team in 2014.\\nThe Institutional Review Board approved the study (MGH-IRB-2017P000544) with waiver of patient consent.\\nAdult patients (\\u2265 18 years of age) with the following entry criteria were enrolled: (I) class III obesity (BMI > 40 kg/m 2 ), (II) diagnosis of ARDS [4] , (III) mechanical ventilation for > 48 h, and (IV) admission to MGH surgical or medical ICU from January 1, 2012, to December 31, 2017. None of these patients participated in any MGH ARDS ongoing interventional trials.\\nThis retrospective study compared two cohorts of patients. During 2012-2014, mechanical ventilation settings of the first cohort of patients (standard protocolbased cohort) was titrated according to ARDSnet PEEP/ F i O 2 guidelines [9, 11] . During 2015-2017, mechanical ventilation of the second cohort of patients (lung rescue team cohort) was treated by the lung rescue team.\\nDuring ICU admission and the first 4 days of mechanical ventilation, patient characteristics together with cardiopulmonary and hemodynamic were recorded. Patients' outcomes up to 1-year follow-up were also recorded.\\nAll ARDS patients at MGH were ventilated in volumecontrolled or pressure-controlled ventilation mode, with tidal volumes of 4-8 mL/kg of predicted body weight while maintaining plateau pressure of < 28 cmH 2 O and respiratory rate titrated to maintain 88%-95% SpO 2 and permissive hypercapnia with pH > 7.25 and PaCO 2 < 60 mmHg [20, 21] . To prevent ventilation asynchrony and for ARDS treatment, patients were paralyzed with cisatracurium to suppress train-of-four to 0 to 1 [22] .\\nFor the standard protocol-based cohort ( Fig. 1 , panel 1), ventilation of patients with class III obesity was managed using ARDSnet PEEP/F i O 2 tables [11] . Due to the absence of benefit associated with the higher PEEP/ lower F i O 2 table [11] , the lower PEEP/higher F i O 2 table was used.\\nFor the lung rescue team cohort (Fig. 1, panel 2) , ventilation was started based on the lower PEEP/higher F i O 2 table. Subsequently, ventilation was titrated by the lung rescue team, composed of a critical care physician, two critical care fellows, and an ICU respiratory therapist. The lung rescue team represents the clinical evolution of the research activity performed by our group. In recent years, our group extensively investigated ventilatory management of patients with obesity and acute respiratory failure, and these studies often led to a dramatic improvement in the levels of hypoxemia. Consequently, the ICUs of MGH started to request a respiratory consult and a clinical team was implemented. The lung rescue team evaluated patients within the first day from the beginning of mechanical ventilation. The clinical decisions of the lung rescue team were based on multiple measurements of respiratory mechanics and hemodynamics; mechanical ventilation was accurately titrated using LRM and choosing the correct value of PEEP matching information from decremental PEEP trial, end-expiratory transpulmonary pressure measurements, and the use of electrical impedance tomography. Hemodynamics was carefully analyzed through standard hemodynamic parameters and right heart echocardiography with the aim to intensively study the interaction between lung and heart/vascular function [23] .\\nIn both cohorts of patients, the weaning process from mechanical ventilation was performed according to the recommendations of the 2005 International Consensus Conference [24] .\\nBaseline characteristics, respiratory mechanics, and hemodynamics during the first 4 days of ICU admission and outcomes were compared between the two groups with two-sample parametric or nonparametric tests as appropriate. Normality of distribution was assessed using the Shapiro-Wilk test. T-test/Wilcoxon rank-sum and chi-square test were used for group Fig. 1 Standard protocol-based approach versus lung rescue team approach. According to the standard protocol-based approach, ARDS patients were essentially treated setting the mechanical ventilator in accordance with the indications provided by ARDSnet tables (panel 1). Conversely, an individualized lung rescue team approach (panel 2) involved a thorough (multidisciplinary) assessment of respiratory mechanics, including esophageal pressure monitoring (2, A), as well as the patient's response to lung recruitment. The best-PEEP was titrated based on a decremental PEEP trial, while hemodynamics was assessed by means of transthoracic echocardiography (2, B). PEEP, positive end-expiratory pressure; F i O 2 , inspiratory fraction of oxygen; P L , transpulmonary pressure comparison among continuous/categorical and categorical/categorical variables, respectively.\\nThe primary outcome was mortality at 28 days and was decided a priori during the study design process. The effect of lung rescue ventilation strategy on mortality up to 1-year follow-up was assessed using Kaplan-Meier curves, and hazard ratio was calculated using the Cox proportional hazard model. Kaplan-Meier analysis was used to examine unadjusted differences in survival in the two groups. The Cox proportional hazard model was used to examine differences in survival after adjusting for predetermined potential confounders (BMI, age, APACHE, PaO2/FiO2 ratio). A two-sided P value of < 0.05 was considered statistically significant. Data from all patients admitted to the ICU that met inclusion criteria were collected and included in the analysis; an a priori sample size calculation was not performed. All statistical analysis and all graphs were performed using STATA version 13 (STATA Corp., USA).\\nPlease see supplementary materials for details regarding patient screening, recordings, measurements, and interventions.\\nFrom 2012 to 2014, 70 ARDS patients (BMI = 49 \\u00b1 9 kg/ m 2 ) were managed according to a standard protocolbased approach. From 2015 to 2017, 50 ARDS patients (BMI = 54 \\u00b1 13 kg/m 2 ) were managed by the lung rescue team. Upon ICU admission, patients in the first cohort were slightly older (57 \\u00b1 13 years versus 52 \\u00b1 14 years, P = 0.03). No other baseline comorbidities differed between the groups (Table 1) .\\nAt ICU admission, P a O 2 /F i O 2 was higher in the standard protocol-based cohort (PaO 2 /FiO 2 ratio of 197 mmHg [CI95% 177-217], compared to 154 mmHg [CI95% 127-179] in the lung rescue group, P = 0.003) ( Table 2 and Additional file 1: Table S1 ). No other differences were observed in baseline lung mechanics between the two cohorts ( Table 2 and Additional file 1: Table S1 ).\\nPatients in the standard protocol-based cohort were all ventilated according to the ARDSnet lower PEEP/higher F i O 2 table [11] for > 48 h (average 198 \\u00b1 278 h). PEEP levels and respiratory mechanics did not change during the first 4 days of ventilation. By day 4, only 10 patients (14%) had improved oxygenation to > 300 mmHg ( Table 2 and Additional file 1: Table S1 ).\\nUpon admission to the ICU, patients in the lung rescue group were also initially ventilated according to the ARDSnet lower PEEP/higher F i O 2 table [11] . Within 24 h after initiation of mechanical ventilation, the lung rescue team performed esophageal manometry, LRM, and a decremental PEEP trial. As a result of the lung rescue approach, PEEP increased an average of 10 cmH 2 O (9 \\u00b1 2 cmH 2 O on day 1 versus 19 \\u00b1 4 cmH 2 O on day 2, P < 0.001) and end-expiratory transpulmonary pressure passed from \\u2212 6.3 \\u00b1 3.7 cmH 2 O to + 1.7 \\u00b1 3.2 cmH 2 O (P < 0.001). All patients in the lung rescue team cohort were ventilated for > 48 h (299 \\u00b1 322 h). Comparison of time of ventilation, measured as ventilation-free days, did not reveal a difference between the two cohorts (Table 3) .\\nIn contrast to the standard protocol-based cohort, the lung rescue team cohort showed a remarkable improvement in respiratory mechanics and oxygenation throughout the first 4 days of ventilation. Driving pressure decreased an average of 3.4 cmH 2 O, while compliance of the respiratory system improved an average of 12 mL/ cmH 2 O, suggesting considerable lung recruitment. This result was also documented by improved PaO 2 /FiO 2 ratio from 153 \\u00b1 88 mmHg at admission to 282 \\u00b1 102 mmHg on day 2, after titration of PEEP (Table 2) . On day 4, 28 patients (56%) improved oxygenation to > 300 mmHg.\\nA large proportion of patients in both groups (70% of patients in the standard protocol-based cohort and 76% of patients in lung rescue team cohort) presented in shock, requiring similar intravenous doses of inotropic and vasopressor agents (16, CI95% 11-21 in the standard protocol-based cohort versus 15, CI95% 9-21 in the lung rescue team cohort, P = 0.79). In the first 4 days of ICU admission, the average dose of required inotropics and vasopressors in the standard protocol-based cohort did not substantially change (Table 2 ). In the lung rescue team cohort, despite the PEEP increase of 10 cmH 2 O, average VIS decreased during the first 4 days of ICU admission. By day 4, the proportion of patients requiring vasopressors decreased to 30%, requiring lower doses of intravenous inotropic agents and vasopressors (Table 2) .\\nTo monitor right heart function of hemodynamically unstable patients, the lung rescue team performed bedside transthoracic echocardiography (TTE) before and after titration of ventilation. Both tricuspid annular plane systolic excursion (TAPSE) and peak systolic velocity (S\\u2032) were unchanged by LRM and PEEP titration (TAPSE measured in 27 \\nPatients in the standard protocol-based cohort compared to those in the lung rescue team had almost double the risk of dying from an ARDS diagnosis at 28 days and 3 months (Fig. 2 and Table 3) .\\nNo deaths occurred in either group after 3 months, and the increased risk of mortality persisted in the standard protocol-based cohort at 6 months and at 1 year after admission. The mortality difference between cohorts was even greater when corrected for potential confounders (BMI, APACHE, age, PaO 2 /FiO 2 ratio), suggesting that a lung rescue approach is a strong independent variable of improved survival (Fig. 2 and Table 3 ).\\nThe main cause of death was multi-organ failure in the standard protocol-based cohort (93%; Additional file 1: Table S2 ) and lung rescue team cohort (64%; Additional file 1: Table S3 ). In the remaining cases, care was withdrawn in the standard protocol-based cohort for severe and diffuse ischemic brain injury (two patients, 7%) and in the lung rescue team cohort for advanced metastatic cancer (four patients, 36%). All treatments were withdrawn based on patient's and proxy's wishes, with the exception of palliative and comfort care. Driving pressure is difference between plateau pressure (measured at the end of an end-inspiratory pause) and total positive end-expiratory pressure (measured at the end of an end-expiratory pause) b Respiratory system compliance is the ratio of tidal volume to driving pressure Information at days 3 and 4 were available for more than 80% of patients and statistics were performed only on available data To evaluate the effects of time on mortality of patients with obesity and ARDS, the two cohorts of patients were subdivided in two further subgroups of equal number of patients. The standard protocol-based cohort mortality was 40% in the first 35 patients (2012-2013) and 42% in the second 35 patients (2013-2014) indicating that mortality did not change over the 3 years. The mortality in the first 25 patients of the lung rescue team cohort was 28% (2015-2016) and 16% in the remaining 25 patients (2016-2017) , showing a decrease compared to the standard protocol-based cohort.\\nWhen mortality was accounted for, days not in hospital at day 28 [25] and days not in ICU at day 28 [25] were similar in the two groups. The increased ICU length of stay and hospital length of stay in the lung rescue team cohort have to be ascribed exclusively to improved survival (Table 3) . No differences were observed in the incidence of reintubation, tracheostomy, and renal acute injury between cohorts.\\nNo safety concerns were recorded associated with the lung rescue procedures (please see Additional file 1: Supplementary Materials).\\nThe implementation of a lung rescue team to individually titrate mechanical ventilation according to physiological parameters in patients with class III obesity and ARDS was associated with significantly decreased mortality at 28 days and at 3 months compared with use of the ARDSnet lower PEEP/higher F i O 2 table. The mortality difference persisted at 1-year follow-up.\\nThe largest epidemiologic study on ARDS was conducted in 2016 and reported an overall ICU mortality of 35.3% (95%CI 33.3%-37.2%) and hospital mortality of 40.0% (95CI 38.1%-42.1%) [7] , similar to what we observed in the standard protocol-based cohort treated according to the ARDSnet protocol. In 2019, two large randomized US trials in ARDS reported a mortality rate at 1 year of 44% [26] and at 90 days of 42% [25] . MGH was part of those two studies. Thus, the improved survival observed in the lung rescue group might be attributed to a prompt and sustained improvement in cardiopulmonary physiology following the individualized titration of mechanical ventilation.\\nIn the USA, the largest trial in ARDS patients that systematically changed the common practice of mechanical ventilation was the original ARDSnet trial [9] , a study sponsored by NIH and completed in 2000. This trial developed simple and clear mechanical ventilation protocols (ARDSnet tables) to guide clinicians to deliver mechanical ventilation for ARDS patients [11] . However, obesity was a criterion of exclusion from the trial and for many subsequent ARDS trials focused on best practice of mechanical ventilation [9, 11, 14, 15, 27] .\\nDespite the high prevalence of class III obesity [2] in the USA and increasing health issues related to this condition, to date no study has primarily evaluated this population with ARDS. The PROBESE study [28] showed no difference between two protocolized ventilator strategies (low PEEP, [4 cmH 2 O] versus lung recruitment maneuvers and high PEEP [12 cmH 2 O]) during general anesthesia for surgery in patients with obesity and without ARDS. To our knowledge, our observational study is the first to investigate the effects on survival of an alternative individualized and physiologically driven approach of care, rather than use of ARDSnet protocols in patients with class III obesity and ARDS. In the absence of definitive guidance, over the past years we have meticulously studied pulmonary physiology and hemodynamics in mechanically ventilated patients with obesity [17] [18] [19] .\\nThe question in ARDS patients is whether nonfunctional atelectatic lung can be re-opened without subjecting the normal lung to further injury.\\nARDS patients are often said to have a \\\"baby lung,\\\" a reduced lung volume but with a highly variable amount of recruitable lung parenchyma [29] . The increased pleural pressure in patients with class III obesity [16] [17] [18] [19] causes atelectasis of > 40% of lung parenchyma. Atelectasis in patients with class III obesity can easily be recruited [17, 19] . Our study intervention sought to maximize recruitment of atelectatic lung. This occurred, as shown by the decrease in driving pressure in the lung rescue group. Amato et al. [12] documented that low driving pressure predicts increased survival in ARDS patients. Driving pressure declined only in patients who received individualized physiologic measurements in the second cohort. Decreased driving pressure, improved compliance of the respiratory system, and improved oxygenation all confirm lung recruitment [30] .\\nIn both the cohorts of patients, pressure-volume curves were not performed, thus careful airway closure was not estimated; however, as shown by Grieco [31] in people with obesity, airway closure might be a common phenomenon in the obese population and could co-exist with alveolar derecruitment in our patients as well. Despite the real value of alveolar pressure, it is unknown when airway closure is detected, it was shown [31] that theoretically it could be close to the airway opening pressure; consequently, the alveolar pressure at the end of expiration is independent of the applied PEEP when its value is below the opening airway pressure.\\nContrary to the common association between high levels of airway pressure and reduced right heart function with hemodynamic impairment, the lung rescue approach was associated with a decreased proportion of patients requiring vasoactive and inotropic agents. During LRM and after titration of PEEP, most patients' hemodynamics remained unchanged, as shown by TTE right systolic measurements and unchanged doses of vasoactive and inotropic drug infusions. In contrast to the recently published ART trial [13] , a large study of ARDS patients, we found neither barotrauma nor cardiac arrest in our population. Differences in response to increased airway pressure found in the ART trial compared to hemodynamic stability observed in the lung rescue cohort might be explained by the large amount of recruitable lung parenchyma. When atelectatic lung is recruited, there is a decrease in pulmonary vascular resistance and right heart workload. While our study did not invasively measure cardiac output, pulmonary pressures, or filling pressures, our prior work using a porcine model of obesity documented unchanged pulmonary vascular resistance and hemodynamics with both LRM and decremental PEEP trial [18] . In patients treated according to the lung rescue approach, hemodynamic stability continues if ventilation can establish a homogeneous distribution of ventilation, physiological lung volumes with low transpulmonary pressures, and minimal alveolar overstretch, even when higher levels of airway pressures are required. Prior physiological studies have confirmed that hemodynamics of critically ill patients with high pleural pressure and obesity tolerate LRM and increased airway pressures [17, 18] .\\nFirst, this report is not a randomized controlled trial but a single-center retrospective study with a limited number of patients, evaluating two cohorts of patients treated with different approaches to mechanical ventilation. Notably, at ICU admission, patients in the lung rescue team cohort had worse oxygenation, which is associated with a higher severity of illness, than patients in the standard protocol-based cohort [32] . Despite increased critical illness, the lung rescue team cohort had decreased mortality in multivariate analysis after adjusting for common ICU mortality confounders, including age, BMI, APACHE, and PaO 2 /FiO 2 ratio (Additional file 1: Table S4 ). The strength of the physiological rationale and improvement in mortality suggests future multicenter prospective randomized trials should be done to confirm these findings.\\nSecond, survival benefits observed in the lung rescue team cohort might result from recent improvements in care of patients with obesity and novel ARDS therapies. However, since 2012 at MGH, there have not been changes in the care of patients with class III obesity, in titration of mechanical ventilation, or in treatment of septic shock, except those discussed in this study. Further, as mentioned in the \\\"Methods\\\", none of our patients were enrolled in any MGH clinical trials. Although we cannot exclude other factors beyond our knowledge that might have affected the outcomes in the two cohorts, we know that, accounting for patients enrolled in trials, ARDS mortality did not change at MGH over the past 10 years (unpublished data) and did not change in the most recent US ARDS trials [25, 26] .\\nThird, benefits associated with the lung rescue approach might be difficult to reproduce in other centers, unless a dedicated team has expertise in advanced measurements of lung physiology and hemodynamics. In 2014, we implemented a lung rescue team at MGH to optimize mechanical ventilation in patients with class III obesity. The research fellows that participated in the specialized team received ongoing training over a year in measurements of respiratory and cardiac physiology, including use of TTE, transpulmonary pressure measurement, and respiratory mechanics, which allowed personalized assessment of each patient in the lung rescue team cohort.\\nFourth, despite we considered patients with an average BMI higher than 50 kg/m 2 , in the present study, we did not take into account the possible differences between abdominal and non-abdominal obesity and their correlation with BMI.\\nTo our knowledge, this is the first observational study to specifically investigate the impact of different mechanical ventilation approaches in patients with class III obesity and ARDS on survival. We found that in patients with an average BMI of > 50 kg/m 2 , an individualized lung rescue approach based on individualized cardiopulmonary physiology is associated with a decreased in-hospital mortality. Based on our findings and considering the increasingly large group of hypoxic ARDS patients with obesity, the present study justifies the conduction of a randomized control trial testing whether a titration of mechanical ventilation based on an individualized strategy with a dedicated health professionals' team might be superior to a fixed protocol based on the ARDSnet lower PEEP/higher F i O 2 table.\\nThe online version of this article (https://doi.org/10.1186/s13054-019-2709-x) contains supplementary material, which is available to authorized users. \",\n          \"The novel severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), which causes coronavirus disease (COVID- 19) , first appeared in December 2019 in Wuhan, China, and quickly spread worldwide. On 30 January 2020, the Director-General of the World Health Organization (WHO) declared the outbreak of COVID-19 to be a public health emergency of international concern and a set of recommendations was issued [1] . In Lithuania, the first case of COVID-19 was diagnosed on 28 February 2020. As the number of cases was increasing rapidly, the Lithuanian Government decided to declare quarantine from 16 to 30 March [2] . This was extended several times and ended on 16 June. All public indoor and outdoor gatherings were prohibited. Educational institutions began to work remotely. Shops excluding grocery shops and pharmacies were closed. Restaurants and bars were also closed, leaving the option for food takeaway. This situation forced people to cook\\nThe study questionnaire included 44 questions regarding various sociodemographic criteria, the Mediterranean Diet Adherence Screen (MEDAS) [12, 13] , and changes in eating behaviours during the quarantine: snacking and intake of some foods with possible answers: 'Higher', 'Lower' or 'As usual'; perception of eating more during the quarantine with possible answers 'Yes' or 'No'; changes in physical activity with possible answers: 'It has increased', 'It has decreased', 'It remains as usual', 'I do not practice physical activity'; and changes in alcohol consumption with possible answers: 'Yes, my intake of alcoholic beverages is higher', 'No, my intake of alcoholic beverages is lower', 'My intake of alcoholic beverages remains as usual'. The question about weight change was: 'Have you gained weight during the quarantine' with possible answers: 'Yes', 'No', 'I don't know'. By weight change, the respondents were grouped in those who gained weight ('Yes') and others ('No' or 'I don't know'). Self-reported current weight and height were used to calculate body mass index (BMI) as weight in kilograms divided by height in meters squared. According to WHO guidelines, overweight was defined as BMI 25-29 kg/m 2 and obesity as BMI \\u2265 30 kg/m 2 [14] .\\nThe respondents were grouped into three age groups: 18-35 years, 36-50 years and 51 years and older. By education level, respondents were categorized into two groups: with university education and lower level of education.\\nData analysis was performed using the statistical package IBM SPSS Statistics for Windows, Version 20.0 (IBM Corp.: Armonk, NY, USA, released 2011). The categorical variables were presented as percentages and compared using the chi-square test and z test with Bonferroni correction. Univariate and Nutrients 2020, 12, 3119 3 of 9 multivariate logistic regression analysis was used to evaluate the associations between weight gain during COVID-19 quarantine (dependent variable) and social factors, nutrition, physical activity and alcohol consumption habits.\\nThe international COVIDiet study was approved by the Research Ethics Committee of the University of Granada (1526/CEIH/2020) and was registered in ClinicalTrials.gov (https://clinicaltrials. gov/ct2/show/NCT04449731). Additionally, the Lithuanian study was approved by the Bioethics Centre of the Lithuanian University of Health Sciences. No personal data was collected and participation was voluntary and anonymous. The participants were informed about the objectives of the study.\\nThe main characteristics of the study population are presented in Table 1 . The majority of the participants were females, 18-35 years old and having a university education. Prevalence of overweight was 27.8% and obesity 12.4%. Every third respondent (31.5%) gained weight during the quarantine. A considerable number of participants reported that they changed their eating habits during the quarantine (Table 2 ). Almost half of the respondents (49.4%) answered that they ate more than usual. A similar proportion of participants (45.1%) increased snacking. However, 62.1% of respondents replied that they cooked at home more often than before quarantine. Every fifth participant (20.6%) increased consumption of fried food. Fast food became less popular, as 41.3% of respondents decreased its consumption. More than one-third of participants (37.7%) reported that they increased consumption of homemade pastries. Intake in carbonated and sugary drinks was decreased by 19.4% of respondents and consumption of commercial pastries by 26%. More people reduced read meat and fish consumption than increased, while a higher proportion of participants increased fruit and vegetable consumption than decreased.\\nMost of the participants (69.9%) declared that their intake of alcoholic beverages (wine, beer, strong alcoholic drinks) remained the same as before the COVID-19 quarantine. The proportion of those who increased (14.2%) and decreased (15.9%) consumption of alcohol was similar. A large number of participants (60.6%) reported that they reduced physical activity (Table 2) . The changes in health behaviours differed according to changes in body weight ( Table 3) . Diet of those who gained weight changed in a more negative direction compared to those who reported no changes or did not notice the increase in body weight. More weigh-gaining respondents reduced fruit and vegetable consumption and increased red meat, carbonated or sugary drinks, pastries, fast food and fried food consumption. The majority of participants (84.3%) whose weight increased declared that they ate more compared to 33.4% of individuals without changes in body weight. More weigh-gaining people increased snacking and cooked more often than before quarantine. Increase in alcohol consumption was reported by every fifth participant (20.9%) who gained weight, and every tenth (11.2%), who did not. A large proportion of weight-gaining respondents (85.2%) decreased their physical activity during quarantine (Table 3) . Table 3 . The proportion of participants who reported changes in health behaviours by changes in weight (%). The weight gain was associated with BMI of participants. Those with higher BMI gained weight more often compared to those having normal BMI (Figure 1 ). The weight gain was associated with BMI of participants. Those with higher BMI gained weight more often compared to those having normal BMI (Figure 1 ). The binary univariate logistic regression showed that women were more likely to gain weight compared to men (Table 4 ). No associations were found with the age and education of respondents. The odds of weight gain were higher for those who decreased consumption of fresh fruits and vegetables and increased consumption of red meat, carbonated or sugary drinks, pastries, snacks, fast or fried food and ate more than usual. Increased alcohol consumption and decreased physical activity were also related to higher odds of weight gain. In multivariate logistic regression analyses, the association of weight gain with increased intake of carbonated or sugary drinks, homemade pastries and fast food, increased snacking, eating more than usual, decreased physical activity and increased alcohol consumption remained statistically significant. The odds of weight gain also increased with age. The binary univariate logistic regression showed that women were more likely to gain weight compared to men (Table 4 ). No associations were found with the age and education of respondents. The odds of weight gain were higher for those who decreased consumption of fresh fruits and vegetables and increased consumption of red meat, carbonated or sugary drinks, pastries, snacks, fast or fried food and ate more than usual. Increased alcohol consumption and decreased physical activity were also related to higher odds of weight gain. In multivariate logistic regression analyses, the association of weight gain with increased intake of carbonated or sugary drinks, homemade pastries and fast food, increased snacking, eating more than usual, decreased physical activity and increased alcohol consumption remained statistically significant. The odds of weight gain also increased with age. \\nOur study demonstrated that Lithuanians changed their eating habits during the quarantine. Almost half of the respondents reported that they ate more than usual, increased snacking and cooked more often at home. The intake of foods changed in positive and negative directions. More respondents increased than reduced fruit and vegetable consumption. The intake of carbonated and sugary drinks, fast food, and commercial pastries decreased; however, the consumption of homemade pastries and fried food increased. Negative changes in eating, physical activity, and alcohol consumption habits were associated with weight gain which was reported by one third of respondents.\\nOur data support previous findings showing that restrictions during COVID-19 increased the number of main meals, frequency of cooking, and snacking [12, [15] [16] [17] [18] . Spending more time on cooking offers the possibilities to eat healthier, try new recipes and enjoy food without rushing. To remind people of the most important principles of healthy nutrition, WHO experts developed recommendations on food and nutrition during self-quarantine [1] . However, restrictions during COVID-19 pandemic decreased the availability of some foods. People were urged to stay at home, all restaurants and bars were closed, and only some of them offered food takeaway. Ordering food online was overloaded [12] . The variety of food products in grocery stores has declined [12, 19] . A Spanish study highlighted that people reported difficulties of finding meat in the supermarkets and grocery stores, which could be related to a decrease in consumption of meat during quarantine [12] . At the beginning of the quarantine, people experienced panic and uncertainty, which led to extra buying of all long-lasting food products and caused a shortage of grain and some other products in grocery stores [19] .\\nSocial isolation with or without loneliness and boredom can be associated with a negative impact on nutrition habits, obesity, physical activity, and poor physical and mental health [6, 15, 18, 20, 21] . Some authors showed that high-level of depression was related to poor diet, increased intake of saturated fat, energy-dense and salty foods [6, 22] . Women who tended to act impulsively in a stressing environment were more likely to eat more candies [23] . People with anxiety consumed snacks 2.45 times more often [24] . The studies performed during the COVID-19 pandemic revealed more frequent and increased consumption of sweets, biscuits and cakes, decreased intake of fruits and vegetables and increased consumption of frozen and canned foods, more frequent cooking and eating out of control [15] [16] [17] [18] 24, 25] . On the other hand, some positive changes such as decreased consumption of processed meat, carbonated or sugary drinks and increased consumption of fresh fruits and vegetables, fish, legumes and white meat were identified [12, 16] . Our data support the previous findings that both positive and negative changes in dietary habits were observed during the quarantine.\\nIt is well documented that physical activity has many positive health effects [26] . Together with the changes in diet, a strong impact of quarantine on physical activity was observed. Many studies showed that all levels of physical activity decreased and sitting or screen time increased [12, [15] [16] [17] [18] 22, 24] . Few studies found opposite results, showing an increase of physical activity during COVID-19 crisis [16, 27] . These differences may be related to different governmental policy on movement restrictions during this period.\\nThe data on alcohol consumption is controversial. Some studies reported an increase in alcohol consumption and higher drinking tendency among alcohol addicts [17, 25] , while other studies reported decrease in binge drinking and alcohol consumption [12, 15, 16, 22] . A British study found decreased alcohol consumption only in the younger cohort [27] . Our study demonstrated similar proportions of those who increased and decreased alcohol consumption.\\nEvidence suggests that even a small positive energy balance over time is sufficient to cause weight gain in many individuals [28] . Our data showed that those respondents who gained weight were more likely to change their diet in an unhealthy direction, decrease physical activity, and increase alcoholic beverages consumption than those who did not report changes in body weight. Furthermore, more respondents with higher BMI at the beginning of quarantine gained weight compare to those having a normal weight.\\nPrevious studies have also reported weight gain during the quarantine [16] , providing estimated weight changes from 1.6 kg to 2.07 kg in different countries [16] [17] [18] 24, 25] . People following an unhealthy diet had 4.5 times higher odds of weight increase [24] . Weight gain was associated with higher consumption of most foods: meat, dairy, fast foods, and even fruits, vegetables and legumes [25] . An Italian study found no associations between BMI and healthy food intake, while higher consumption of junk food such as snacks, dressings, sweet beverages and sweets was associated with higher BMI [16] . Lower exercise, consumption of snacks, unhealthy foods, cereals, and sweets were associated with significant weight gain in adults with obesity a month after the beginning of the quarantine [18] .\\nThe latest studies report that patients with obesity are at increased risk of complications of COVID-19 infection and higher mortality rates [8, 29] . As there is not yet effective treatment and vaccination against COVID-19 and there may be new virus outbreaks, healthy eating and physical activity promotion should be incorporated in government initiatives to reduce the negative effects of the COVID-19 pandemic [30] . Encouraging people to follow a balanced diet, especially targeting those with obesity, could help to maintain an effective immune system and may provide protection against infections and other diseases.\\nThe strengths of our study include the online survey, which allowed us quickly to reach a sufficiently large sample of the population from different municipalities of the whole country during the quarantine. Additionally, we used the COVIDiet questionnaire prepared by researchers from the University of Granada for the international study so we will be able to compare our data with other countries in the future. Several limitations of our study should be mentioned for the accurate interpretation of its results. First, we used non-random sampling to reach participants. The majority of respondents were women and people with higher education. The higher participation of women was also reported by other researchers [12, 18] . However, our study covered all municipalities of Lithuania and a large age interval. Furthermore, we used self-reported online survey data which may be less reliable and biased. BMI was calculated from self-reported data on weight and height, which could lead to underestimation of overweight and obesity. A question about weight loss was not included in the questionnaire. Finally, we did not collect data on employment status and stress during the quarantine. Stress could impact eating and alcohol consumption habits.\\nCOVID-19 quarantine caused both positive and negative changes in nutrition, physical activity, and alcohol consumption habits in the Lithuanian population. Weight gain was associated with unhealthy dietary changes, decreased physical activity, and increased alcohol consumption. Our data highlighted the need for investigations into which dietary and physical activity guidelines might be appropriate to help counteract potentially negative impacts of COVID-19 on health behaviour and body weight.\\nAuthor Contributions: V.K. and J.P. made substantial contributions to conception and design of the manuscript. V.K. and L.B. carried out statistical analysis. V.K., L.B. and J.P. were involved in drafting of the manuscript. C.R.-P. reviewed and edited the manuscript. All authors have read and agreed to the published version of the manuscript.\\nFunding: This research received no external funding.\",\n          \"Public health is the combination of sciences and skills that aims to protect, promote and restore the wellbeing of a population. 1 Public health research in developing countries is important to quantify health conditions, assess interventions and control measures, as well as inform health policy decisions. There are several aspects of public health research in developing countries that differentiate it from that conducted in more affluent settings.\\nFirst, the major health problems in poor populations are infectious, perinatal and nutritional disorders and the highest burden of these problems is found in children. In poor countries, not only are children more vulnerable to disease and death than adults, compared with industrialized nations, they constitute a much larger proportion of the population. Most developing countries have pyramidal populations reflecting high birth rates and short-life spans. By contrast, the age structure of industrialized countries tends to be onion shaped due to decreasing birth rates and longer life expectancy. Recent socioeconomic changes in many poor countries have resulted in a shift in the patterns of disease. As a consequence of lifestyle and behaviour changes, as well as a shift from rural to urban living, developing countries have to cope with chronic noncommunicable illnesses such as adult cardiovascular disease, diabetes and depression, while continuing to struggle with childhood infectious, perinatal and nutritional disorders. 2 Despite these recent developments, the public health sector still tends to focus on children. This is not only because the greatest disease burden still occurs in the youngest age groups, but also because childhood interventions tend to be the most costeffective and sustainable, and the development of children determines the quality of future populations.\\nSecond, in many parts of sub-Saharan Africa, tropical Asia and Latin America, there is a profound lack of health infrastructure and reliable routinely collected data. Large proportions of births and deaths occur at home and remain unregistered. Many ill patients do not or are unable to seek health care. Treatment facilities are often understaffed and have limited capacity for laboratory-confirmed diagnoses. These factors pose significant challenges for recording accurate morbidity and mortality data. Thus a large fraction of public health research in these settings focuses on health burden assessments to generate the most urgently needed data for public health delivery.\\nThird, populations in developing countries should be considered as vulnerable in the sense that they have limited financial and political power. It is important that they not be left out of health-related research from which they would benefit, but this research has to be done in accordance with ethical guidelines and principles.\\nFinally, the bulk of the worldwide disease burden is in developing regions, where only a small fraction of global healthcare funds is available. Epidemiological data are crucial for the rational allocation of these limited resources and to inform decisions about strategies to be implemented. Public health research in developing countries emphasizes the search for cost-effective control and preventive strategies that could potentially benefit large segments of the community rather than expensive treatments for individual patients.\\nIn this chapter, we discuss some important issues and challenges of public-health research in resource-poor settings. This chapter does not cover all aspects of health research in developing countries but will discuss the following: disease burden assessment, outbreak investigation, measuring protection and cost-effectiveness of interventions, good clinical practice in research and using research findings to guide health policy decisions. The focus of all sections will be on infectious diseases.\\nImprovement of health conditions in a country requires recognition of the main problems, selecting the most appropriate and cost-effective interventions, implementing services efficiently and a continuous assessment of results. On-going analysis of disease burden is essential to the formulation of responsive health policies. There are several epidemiological methods to assess disease burden, the most relevant of which are discussed below.\\n\\u2022 There are several aspects of public health research in developing countries that differentiate it from that conducted in more affluent settings, including the need to focus on infectious, perinatal and nutritional disorders especially in children.\\n\\u2022 Several methods to assess the burden of and evaluate interventions against infectious diseases are discussed and illustrated with examples.\\n\\u2022 Health research in developing countries should comply with current international research standards, so as to assure that the rights, safety and wellbeing of participants are protected and that the study data are credible.\\n\\u2022 Public health resources in developing countries are limited. An understanding of the burden of health conditions and the potential impact of interventions and control measures is crucial for rational priority-setting.\\n6 Issues and Challenges of Public-Health Research in Developing Countries 41 create 5-year projections until 2020. 6 Several parameters are required to calculate DALYs, including age-and gender-specific mortality estimates; incidence of disease estimates; proportion of time individuals are disabled; and severity and duration of disability. Two further adjustments are made: discounting and age-weighing. Future years of healthy life lost are often discounted by 3% per year, i.e. years in the future count for less compared to those in the present. Discounting future health reduces the relative impact of a child death compared with an adult death. The value that is accorded for a year of life lost is also age-weighed, based on the assumption that the relative value of a year of life rises rapidly from zero at birth to a peak in the early 20s, after which it steadily declines. The strengths of this approach are that it incorporates disability and it allows comparison between diseases, populations and time periods.\\nDALYs are now being used in other types of studies, notably economic analyses (see below). In addition, the WHO is also using a new metric, healthy-adjusted life expectancy (HALE) at birth, which adds up expectation of life for different health states, adjusted for severity distribution, making it sensitive to changes over time or differences between countries in the severity distribution of health states. HALE is defined as the average number of years that a person can expect to live in 'full health' by taking into account years lived in less than full health due to disease and/or injury.\\nTo continue the important work on global health statistics, the Institute for Health Metrics and Evaluation (IHME) was launched at the University of Washington in June 2007 (see: http://www.healthmetricsandevaluation.org). The IHME measures population health status and disease burden, identifies the factors that determine health outcomes and evaluates health policies and interventions. Among their earliest projects were new estimates of mortality rates.\\nIn a separate effort, the Child Health Epidemiology Reference Group of WHO and UNICEF estimated global, regional and national causes of child mortality in 2008. 7 They used multi-cause proportionate mortality models to estimate deaths in neonates and under-5-year-old children and selected singlecause disease models and analysis of available vital registration data to estimate causes of child deaths. They found that of the calculated 8.795 million deaths in children younger than 5 years of age worldwide in 2008, infectious diseases caused 68%, with the largest percentages due to pneumonia, diarrhoea and malaria.\\nThese estimates of global disease burden are useful to guide global programmes and donor assistance, but there are uncertainties about the accuracy of these calculations and the estimates may not be applicable to specific locations. Thus, there remains the continuing need for special field research studies to validate these approximations.\\nProspective surveillance studies detect a disease of interest in a cohort to calculate incidence, case fraction, case fatality rate or other measures of frequency. These studies provide data on the risk for illnesses or death in a population but may also be implemented in preparation for intervention trials (see below). Prospective surveillance studies are expensive due to the large costs for case-capture, diagnostic verification, treatment and data management. Frequently, it is necessary to conduct a census to have an accurate and up-to-date denominator. The conduct of RETROSPECTIVE REVIEWS OF EXISTING DATA An assessment of available data provides a general idea about health and disease statistics in a country or region. Collating and analysing existing information is relatively cheap and quick, may be useful to assess trends and may provide a nation-or region-wide picture but its limitations need to be kept in mind. The quality of the assessment depends on the data used to reach it. For example, reliance on public agencies' estimates of a disease, which may be inflated to increase public attention and consequently increase funding, will distort global calculations. The World Health Organization (WHO) compiles more comprehensive health data annually but the accuracy of reporting varies between diseases and across countries. 3 These estimates rely on routine notification, which may be weakened by over-and under-diagnosis, incomplete reporting and delays. Reporting of some diseases may be suppressed due to social taboos (e.g. HIV/AIDS) or the fear of trade sanctions (e.g. cholera).\\nSystematic reviews and analyses of published and unpublished data on specific diseases may provide more accurate information. For example, Reddy and co-workers conducted a systematic review and meta-analysis of the published literature to assess the burden and the most common aetiologies of community-acquired, non-malaria bloodstream infections in Africa. 4 They found that 13% of adults and 8% of children admitted to hospital with a blood culture taken had a bloodstream infection. The most common isolate was Salmonella enterica subspecies (of these, 34% were S. typhi and 58% nontyphoidal Salmonella) overall in adults and Streptococcus pneumoniae in children. Retrospective studies of this nature provide valuable insights for public health planning and research but do not yield detailed and area-specific information. Other limitations include sparse data sources (depending on the health condition of interest) and highly variable or poorly defined methodologies of the original studies. Aside from published findings, there are large amounts of routinely collected and frequently unprocessed disease surveillance reports in health ministries and other institutions. This so-called 'grey literature' may also be included in systematic reviews.\\nDue to the many weaknesses of using existing data, triangulation of information is recommended, that is comparing and contrasting data from various sources to validate accuracy. Other than WHO publications, peer-reviewed articles and government surveillance reports, innovative sources such as outbreak information from the Program for Monitoring Emerging Diseases (ProMED), which operates as an online forum for infectious disease specialists, microbiologists and public health officials, have also been included in burden of disease assessments. 5 Traditionally, burden of disease assessments include mortality and morbidity rates, which are useful but do not reflect the total picture. An illness may be uncommon or have low death rates but can still cause considerable burden through chronic disability. In the 1990s, researchers at the Harvard School of Public Health, together with the WHO and the World Bank, estimated the global burden of disease by region and age group, in terms of disability-adjusted life years lost (DALYs). DALYs are the sum of years of life lost due to premature death and years living with disability of specified severity and duration. Using available data from around the world, disease estimates were made for 1990 and mathematical modelling was used to 42 SECTION 3 Epidemiology researchers need to understand the community's utilization of healthcare facilities for the disease of interest and design the surveillance method accordingly.\\nInformation from healthcare utilization surveys may be used to adjust disease estimates obtained through surveillance. For example, in a typhoid fever surveillance study in an urban and rural area in Kenya, Breiman and co-workers used health utilization data to adjust crude incidence rates. 9 The crude and adjusted incidence of blood culture-confirmed typhoid fever in the urban area was 247 and 822 cases per 100 000 person-years of observation (pyo), respectively, compared with 29 and 445 cases per 100 000 pyo in the rural area. The results showed dramatic differences in crude and adjusted, urban and rural typhoid incidence and showed rates similar to those in Asian urban settings, which had not been previously available from Africa.\\nMany diseases have a wide spectrum of presentations, which can range from sub-clinical to life-threatening. Different surveillance methods may capture different entities of the same disease. Active surveillance tends to detect mild illness; passive clinic-based community studies capture conditions that require a patient to present for care; whereas sentinel surveillance in secondary or tertiary hospitals detects the most severe forms of the disease. For example, dengue, a vector-borne viral illness, has a broad range of presentations. The majority of patients recover following a self-limiting febrile illness but a small proportion progress to severe disease, mostly characterized by plasma leakage leading to circulatory failure. 10 Home visits and clinic-based studies are likely to detect dengue fever, whereas hospital-based studies mainly capture the severe forms of the disease ( Figure 6 .1). If only a small fraction of cases in the population become severe, several sentinel hospitals in a large surveillance area may be needed in order to detect a sufficient number of patients to reach useful research conclusions about the disease.\\nTo calculate incidence rates (usually in terms of cases per 1000 to 100 000 population per year), the researchers need an accurate estimate of the numerator (the number of cases) as well as the source population from which the cases are captured. The study population may be enumerated through a baseline study census; demographic and healthcare utilization data may also be collected at the same time. If the study aims to determine very precise incidence rates (e.g. in preparation for or during intervention studies), baseline and follow-up censuses to monitor deaths, births and migration during the study period are required. When approximate incidence rates are to be measured, projected population size from the last government census may be sufficient. If the referral base of hospitals included in a sentinel surveillance is unclear, incidence cannot be calculated. Instead, the proportion of the disease of interest among all presentations or admissions (i.e. case fraction) may be reported. This is useful as an indicator of the burden of disease among patients who seek hospital care.\\nQuantification of sequelae and deaths may be done through follow-up of cases detected during surveillance or through general mortality surveys. In population-based studies, these are usually reported in terms of sequelae and deaths per 1000 to 100 000 population per year. In hospital-based studies, the a census requires technical know-how, a large workforce and hence considerable resources.\\nFor some populations without access to treatment facilities or for rural areas with no laboratories, clinical and diagnostic infrastructure may need to be put in place to carry out surveillance studies. This raises questions of feasibility and increases costs substantially. There is also the problem of long-term sustainability after the surveillance project is completed. However, in an impoverished setting without accurate routine reporting of disease, prospective surveillance studies remain the gold standard for providing as complete and accurate a picture of disease burden as possible.\\nGeographic sites for prospective surveillance studies should be carefully selected to ensure that they are representative of the population of interest. Generalizing findings from one site to other populations, even within the same country, may be problematic. Multi-site studies may be done to assess the burden of disease in a wider regional area. For example, a prospective surveillance study of Shigella diarrhoea was undertaken in study sites in six Asian countries (Bangladesh, China, Pakistan, Indonesia, Vietnam and Thailand), to determine disease burden and prevailing species and serotypes. 8 The overall incidence of shigellosis cases presenting for treatment was two episodes per 1000 residents per year in all ages and was highest among children under 5 years old, at 13/1000 per year. The most frequently isolated Shigella species was S. flexneri in all sites, except in Thailand, where S. sonnei was most frequently detected. Findings such as these may be used to guide potential vaccine development or other interventions.\\nA major decision when conducting prospective studies of disease burden is the choice of how and where to detect cases. Active surveillance detects the disease of interest by regularly visiting or contacting residents of a community. Active surveillance is especially appropriate when the disease of interest is characterized by mild symptoms not likely to cause the patient to present for treatment. Active surveillance, particularly if diagnosis requires laboratory testing and confirmation, is labour-intensive and expensive. In addition, field workers require rigorous training and close supervision to ensure adherence to standardized methods. These logistic complexities limit the population size that can be included in such studies. There is also the danger of fatigue or refusal by the community if the purpose of the study is incompletely understood or if the visits are not conducted in a culturally acceptable fashion.\\nPassive surveillance captures cases presenting for care at treatment facilities. Passive surveillance may be done through treatment facilities established by the researchers or through existing primary healthcare units. When the burden of disease in a very large population is to be measured, then sentinel surveillance in several selected secondary or tertiary hospitals dispersed over a large geographic area may be conducted. This method is much more cost-efficient but unlike active surveillance, is subject to potential bias, since case detection is influenced by the study population's utilization of treatment. Although passive surveillance may be enhanced by regular community dialogue and household visits to encourage consultation for the disease of interest, it may still underestimate the burden of a disease for which the population usually selfmedicates or seeks care with alternative or traditional healers who do not participate in the study. To avoid this bias, the incomplete or the reported cause of death may be unreliable. Many deaths in developing countries do not occur in hospitals, and officials who had neither treated nor seen the deceased person, may be requested to sign death certificates. Verbal autopsy or verbal postmortem is an alternative method to collect mortality data. It enables investigators to ascribe a probable cause of death, retrospectively. Verbal autopsy consists of a detailed interview of the deceased's next of kin or caregiver and a review of relevant records (e.g. clinic visit records) to determine symptoms and signs of illness before death, so as to establish the most likely cause of death.\\nThere are specific recommendations about the design of verbal autopsies for mortality surveillance. 11 The data collection tool should include structured and unstructured questions; forms for adults and children exist and can be adapted, piloted and validated on-site. The interviewers should be specially trained. The interval between death and interview should be culturally appropriate but not overly long as to affect recall. Algorithms for decoding the completed interviews into causes of death must be clearly pre-defined. For example, two medically trained individuals may independently assess the completed verbal autopsy forms to identify the likely cause of each death. If there is disagreement between the two diagnoses, a third physician may adjudicate the decision. If the physicians cannot determine the cause of death, the death may be recorded as unspecified. Computer-automated methods for assigning cause of death have also been proposed and used. In most studies, the cause of death is assigned according to the International Classification of Diseases, Injuries and Causes of Death codes as recommended by the WHO. 12 Metrics for assessing the performance of different verbal autopsy cause assignment methods have been developed. 13 Verbal autopsy studies are becoming increasingly common, with the largest to date conducted in India. 14 In this 'One million deaths study' , all deaths occurring in 2001-2003 in 1.1 million nationally representative Indian households, were surveyed. fraction of presenting or admitted patients who die (case fatality rate) or who develop complications and disability is reported. In prospective surveillance studies, cases have to be immediately and appropriately treated, thus the rates of complications, disability and death are often lower than would be noted outside a research setting. This inherent bias has to be kept in mind when drawing conclusions about sequelae and deaths from prospective research studies.\\nCross-sectional studies survey a sample of the population at one point in time and estimate the prevalence of a condition, an infection or a disease. The survey tool may be a questionnaire; a physical assessment (e.g. of weight, height or blood pressure); a blood test (e.g. for malaria parasites or HIV infection); or a diagnostic procedure (e.g. chest X-ray). Unlike prospective surveillance studies, cross-sectional surveys do not provide information on incidence, i.e. the number of new cases per population during a specific time period. However, they may provide other important public health information. For example, for diseases that induce life-long antibodies (e.g. HIV, hepatitis A and B), sero-epidemiological studies may show the age groups most affected and when done at different time periods and geographic locations, may indicate the effectiveness of prevention and control strategies. HIV seroprevalence in pregnant women is often used as an indicator of the burden of HIV/AIDS in a community. A major challenge of cross-sectional studies is ensuring that the sample selected and included in the survey is representative of the population of interest.\\nFatalities in developing countries often go unregistered. Even when death certificates are available and accessible, they may be withdrawal and sleep problems. The results suggested a need for more psychological support and special counselling services, increased public education about HIV/AIDS to decrease discrimination and financial programmes to assist these children.\\nRigorous empirical research on the costs of illness is important for rational deployment of strategies to mitigate economic effects. For example, the public, provider and patient costs of culture-confirmed cholera were measured in four choleraendemic study sites using a combination of hospital-and community-based studies. 20 Families with culture-proven cases were surveyed at home, 7 and 14 days after confirmation of illness. Hospital-based studies found that the costs of severe cholera were US$32 and US$47 in Matlab (Bangladesh) and Beira (Mozambique), respectively. Community-based studies in North Jakarta (Indonesia) and Kolkata (India) found that cholera cases cost between US$28 and US$206, depending on hospitalization. Patients' cost of illness, as a percentage of average monthly income, was 21% and 65% for hospitalized cases in Kolkata and North Jakarta, respectively. This analysis highlighted the financial burden of an acute diarrhoeal disease on households, often contributing to further poverty. The impact of chronic conditions such as HIV/AIDS on individuals, households and countries is more difficult to quantify. Presumably, HIV/AIDS is an important cause of poverty in many parts of sub-Saharan Africa. And yet, the impoverishing effects of AIDS have been inadequately assessed by existing studies, likely because of methodological challenges. 21 Policy-maker surveys may be conducted to elicit government opinions about diseases. For example, DeRoeck and co-workers interviewed policy-makers and other influential professionals in four South-east Asian countries (Cambodia, Indonesia, Philippines and Vietnam) to determine their views on the public health importance of dengue, the need for a vaccine and the determinants influencing its potential introduction. 22 \\nAn outbreak is the occurrence of disease episodes in greater numbers than would be expected at a particular time and place. The population at risk may range from a small, localized group to large populations. Infectious disease pathogens may cause epidemics that affect regional areas or pandemics that spread around the world. Recently, outbreak investigation and response have received unprecedented prominence with the severe acute respiratory syndrome (SARS) epidemic and the influenza A (H1N1) pandemic.\\nOnce a report of an outbreak is received, there are several steps in its investigation. 23 Specimens are collected for laboratory verification of the diagnosis. Researchers develop a clinical case Field staff completed verbal autopsies and two of 130 physicians independently assigned a cause to each death. The estimated 1.5 million child deaths in India in 2005 were attributed to five conditions: prematurity/low birth weight, neonatal infections, birth asphyxia/neonatal trauma, diarrhoeal diseases and pneumonia. 15 Each of these can be addressed with known, highly effective and widely practicable interventions.\\nIn estimating disease burden, the detection of a specific pathogen through diagnostic tests may be a major problem. The best available tests may fail to confirm many cases of illness caused by some pathogens. If a vaccine is highly effective in preventing disease in such undiagnosed syndromes, vaccine trials may be used to 'probe' the burden of disease that has been missed using the available diagnostic tests. For example, only a proportion of invasive Haemophilus influenzae type B (Hib)-associated illness is detectable through blood cultures or even more sensitive diagnostic testing, such as polymerase chain reaction. In a large, randomized trial of the Hib-tetanus protein conjugate vaccine in Gambian infants, protection was shown not only against culture-positive invasive disease, but also against culturenegative pneumonia, presumably because of the insensitivity of cultures in confirming Hib-pneumonia. 16, 17 This has also been shown in a pneumococcal vaccine trial. 18 Vaccine probe assessments such as these, demonstrate that the burden of some diseases may be much greater than can be proven with the available diagnostic tools (Table 6 .1).\\nParticularly in less-developed countries, where laboratory confirmation of diagnoses may be difficult and where widespread over-the-counter use of antibiotics may result in false-negative diagnoses, highly protective vaccines may be used to probe the total burden of difficult-to-confirm infectious diseases.\\nThe burden of disease in a population may not only be characterized by rates of disease, death and disability; there are also the social impacts and financial costs of illness. The psychosocial consequences of an illness may be evaluated through sociobehavioural studies including rapid and in-depth qualitative surveys, focus group discussions and interviews on knowledge, attitudes and perceptions. For example, in-depth interviews were conducted with 47 children (ages 8-17 years) experiencing the loss of one or both parents due to HIV/AIDS in two rural counties of central China. 19 The majority of the participants reported some level of stigmatization and described feelings of sadness, fear, anxiety, anger, loneliness, low self-esteem, social \\nThe randomized controlled trial (RCT) is the gold standard method for evaluating the efficacy of an intervention. The method involves randomly allocating participants into study and control groups, to receive or not receive the intervention being evaluated. 1 The results are assessed by a comparison of rates of disease or other appropriate outcome in the study and control group. There are practical challenges to doing RCTs in less developed countries, where infrastructure and expertise may not be available. But many infections are geographically limited to developing countries and data on protection against naturally occurring disease can only be obtained in these sites. Even for diseases occurring in both industrialized and developing countries, study results may not necessarily be generalizable because of differences in population characteristics. For example, it has been shown that the immunogenicity of vaccines can be much lower in populations in developing compared with industrialized countries. Poorer performance has been especially problematic for orally administered live vaccines. The efficacy of the Rotarix\\u2122 and RotaTeq\\u2122 vaccines against severe rotavirus related disease in developing countries of Asia, Africa, and Central America does not appear to be as high as that seen in developed countries. [25] [26] [27] Another frequently cited example is the finding that three doses of oral polio vaccine, as formerly used in the USA, resulted in sustained, probably lifelong immunity, whereas many children in developing countries may require >3 doses for adequate seroconversion. 28 The poor performance of these vaccines in developing country populations is not well understood, but could be due to several factors, including high levels of pre-existing natural immunity (either maternal or infection-derived), poor nutritional status, tropical enteropathy and co-existing infections. 29 Even within the developing world, findings from a randomized, controlled trial done in one region may not be generalizable to another region because of differences in the epidemiology of disease. For example, to seek new and improved treatments for severe falciparum malaria, a large multinational randomized comparison of parenteral artesunate versus the standard therapy of parenteral quinine was conducted in South-east Asia. 30 The trial proved that severe falciparum malaria mortality could be reduced by 30% in Asian adults when artesunate is used, but key decision-makers in Africa felt the results of this study were not generalizable to their populations where severe falciparum malaria tends to occur in children, rather than in adults. Over a 5 year period 5000 children with severe malaria in nine African countries participated in a large, multicentre, open-label randomized trial that established the superior efficacy of parenteral artesunate over quinine and led to a change in treatment guidelines. 31, 32 \\nConventional efficacy studies focus on the performance of interventions under ideal conditions whereas effectiveness trials address the protection afforded by interventions under real public health conditions. 33 Evidence from efficacy trials may not suffice to convince policy makers to allocate limited resources for new interventions. Effectiveness studies of licensed drugs and vaccines may be conducted in developing countries to collect evidence on feasibility, acceptability, and practical definition. Using this definition, cases and deaths are identified and the outbreak is analysed by time, place and person. Starting with the first case identified (index case), the number of cases by day or week is plotted to create an outbreak curve. Cases may be mapped and affected persons described in terms of age, sex and other relevant characteristics. Prevention and control procedures are implemented as soon as possible. Treatment centres may be set-up, guidelines for management disseminated and supplies and other logistics provided.\\nThere are several outbreak patterns, each associated with a distinctive epidemic curve. In a common source outbreak, cases acquire the infection from the same source (e.g. a contaminated water supply). This may be a point source outbreak when the exposure occurs in less than one incubation period or a continuous source outbreak when the exposure occurs over multiple incubation periods. In a propagated outbreak, the pathogen is transmitted from person to person.\\nIt is important to determine how the disease is transmitted in an outbreak, so that interventions can be taken to stop the current epidemic and to prevent future epidemics. A cohort study or a case-control study may be conducted to identify the risk factors that would cause an individual to become ill with the disease causing the outbreak. Cohort studies work best for well-defined populations (e.g. an outbreak that occurs among people who attended a gathering such as a funeral), while casecontrol studies work best for outbreaks where the population is not well-defined. The decision regarding the type of study that would be appropriate to investigate an outbreak also rests on the magnitude of risk, the latency of exposure to disease, the prevalence of exposure and timing (i.e. in some instances, it may be too late to conduct a cohort study).\\nAn intervention refers to an intentional change in some aspect of the individual. 1 Public health interventions against infectious diseases are varied and may range from behavioural (e.g. the promotion of hand-washing and breast-feeding; distribution of condoms to control the spread of sexually transmitted diseases; deployment of insecticide-treated bed nets to prevent malaria); structural (e.g. improvement of water supply and sanitation); to pharmacological (vaccine or drug administration). Rational policy-making in developing countries includes the evaluation of potential interventions in terms of safety, efficacy, effectiveness and financial impact.\\nThe protection afforded by many traditional interventions may be widely known and accepted, while that from newer strategies may need to be evaluated. In the assessment of both pharmacological and non-pharmacological interventions, these need to be carried out under ethical conditions and using robust study designs so as to reach valid conclusions. The evaluation of pharmacological compared to non-pharmacological interventions is more stringent; candidate drugs and vaccines require a very careful, phased approach to minimize the potential risks to participants in trials. 24 If there is an intention to license these drugs or vaccines, regulatory agencies (e.g. the US Food and Drug Administration or the equivalent National Regulatory Agency in a developing country) scrutinize the findings of each step in this process.\\nyears old would prevent 456, 158 and 258 typhoid cases (and 4.6, 1.6 and 2.6 deaths), and avert 126, 44 and 72 DALYs over 3 years in Kolkata, North Jakarta and Karachi, respectively. The cost was calculated at US$160 and US$549, per DALY averted in Kolkata (India) and North Jakarta (Indonesia), respectively, and considered very cost-effective.\\nCost-effective analyses may also compare two or more intervention options. For example, in association with the clinical trial of children with severe malaria in sub-Saharan Africa discussed above, 31 the cost-effectiveness of parenteral artesunate and quinine was compared. The mean cost of treating severe malaria patients was similar in the two study groups: US$63.5 in the quinine and US$66.5 in the artesunate arm. Compared with quinine as a baseline, artesunate showed an incremental cost of US$3.8 per DALY averted and an incremental cost per death averted of US$123. Artesunate was determined to be a highly cost-effective and affordable alternative to quinine for treating children with severe malaria. 38 Cost-utility analysis estimate the ratio between the cost of a health-related intervention and the benefit it produces in terms of quality-adjusted number of years lived by the beneficiaries. A common metric in the denominator allows comparisons of diverse interventions against diverse diseases. An example is the use of DALYs.\\nCost-benefit, cost-effective and cost-utility analyses do not incorporate the populations' stated preferences in decisions to finance new interventions. During the past 20 years, several stated preference studies have been conducted in less-developed countries, some of which have assessed willingness-to-pay for various public health interventions. 39 In general, the studies have shown low willingness to pay for these interventions, which is not surprising considering the competing priorities for food, shelter and other basic necessities. Thus, in the poor regions of the world, local governments and international donors have the obligation to continue to provide and implement much needed interventions.\\nHealth research in developing countries should comply with current international research standards, so as to assure that the rights, safety and wellbeing of participants are protected impact or effectiveness. For example, an effectiveness trial of the typhoid Vi vaccine was conducted to provide evidence for wider-scale implementation. 34 Slum-dwelling residents of Kolkata, India, who were 2 years of age or older were randomly assigned to receive a single dose of either Vi vaccine or inactivated hepatitis A vaccine, according to geographic clusters, and were followed for 2 years. The level of protective effectiveness was 61%. Interestingly, the design of the trial also allowed assessment of the protection of unvaccinated neighbours of vaccinated persons. This indirect protection was estimated at 44%. Not only the direct but also the indirect protection by Vi vaccine should be considered in future deliberations about introducing this vaccine in typhoid fever endemic areas.\\nOther than randomized controlled trials, observational studies such as cohort, household contact, case-control, screening and case-cohort studies may be used to assess the effectiveness of an intervention. 35 Since the intervention is not randomly allocated in observational studies, bias is unavoidable. But it may still be possible to obtain sufficiently good estimates of protection from observational studies for public health purposes. Potential biases should be considered in the design phase and steps taken to minimize them if possible. As many more new and innovative interventions become available and the costs of randomized controlled trials escalate, the role of observational methods will become even more important.\\nThe economic impact of an intervention may be assessed using various methods; all of which weigh the costs of an illness with the expenditure for and benefits from an intervention ( Figure  6 .2). A cost-benefit analysis expresses costs and benefits in terms of money, 36 but monetary value may not be appropriate nor completely capture the benefits from health interventions.\\nIn cost-effectiveness analysis, the cost of the intervention is also measured in monetary units but the benefit gained is expressed in terms of cases, deaths and DALYs averted. For example, in conjunction with the typhoid Vi vaccine effectiveness trial cited above, the cost-effectiveness of vaccination programmes in endemic sites in Asia was calculated. 37 It was estimated that a programme targeting all children ages 2-15 contract research organizations, which are, in some cases, highly lucrative publicly traded enterprises. On the other hand, funders are increasingly demanding adherence to GCP and it is unlikely that regulatory agencies will license a new product without the diligence assured by a contract research organization.\\nWe live in an era in which only a fraction of useful health interventions for populations in developing countries are delivered to these populations. This is true both for interventions that are well established as well as for new interventions. There are many reasons. Perhaps the most obvious obstacle is financial, with highly constrained healthcare budgets in developing countries and limited pools of donor resources being inadequate to fund all potentially useful interventions. Because of financial constraints, policy-makers at the global, regional and national levels increasingly demand hard evidence to compare the potential value of alternative interventions and to justify the expenditure of resources to fund the introduction of interventions.\\nProvision of data on the burden of the disease(s) targeted by the intervention is the most common kind of evidence requested by policy-makers. Evidence about the efficacy of an intervention, often obtained from rigorous clinical trials, is usually also needed. Even for interventions that appear attractive in evaluations of efficacy, there may be uncertainties about the logistic and programmatic feasibility of implementation in public health programmes, as well as about the impact upon desired health outcomes under real-life, public health conditions. Thus, evidence about intervention effectiveness is also relevant to policy deliberations. And, as alluded to above, policy-makers typically require evidence of cost-effectiveness of interventions, ideally expressed as the net cost per DALY averted in order to compare interventions for different diseases, as well as prophylactic versus therapeutic interventions.\\nFinally, it may be helpful to have an indication of the population demand for the intervention, even including assessment of willingness to pay for the intervention. Table 6 .2 provides an outline of a programme of translational research (sometimes called 'implementation research') of this sort used in the Diseases of the Most Impoverished (DOMI) Program -a programme funded by the Bill & Melinda Gates Foundation to accelerate the introduction of new-generation vaccines against cholera, typhoid and shigellosis -for the introduction of killed oral cholera vaccines in several countries of Asia and Africa. 46 and that the study data are credible. The ethics of research in developing countries has been the subject of intense discussion. 40 Issues being deliberated include: choosing the appropriate research question and design; use of placebo control groups in randomised controlled trials; capacity-building of local ethics committees to ensure sound and appropriate local review of the study protocol; ensuring that informed consent and assent is obtained which may be especially challenging in impoverished and less educated populations; the potential coerciveness of the offer to participate in research in locations where this may be the only means of obtaining health care; providing equal consideration to participants (including children and pregnant women) in research that would yield results beneficial to them while ensuring the protection of vulnerable participants; ensuring equal distribution of the burden and benefits of the research and minimizing the risk to participants. 41 These concerns are consistent with principles embraced in the World Medical Association's 'Helsinki Declaration' from 1964 and most recently amended in 2008. Special and continuing vigilance is necessary to safeguard the rights of populations in developing countries.\\nAlthough general ethical principles apply to all types of research, Good Clinical Practice (GCP) guidelines developed by the International Conference on Harmonization 42 are the standard for conduct of clinical research for licensure of pharmacologic interventions (e.g. vaccine and drugs). The GCP guidelines aim to ensure not only ethical conduct of studies, but generation of credible data. The guidelines' objectives of documenting informed consent, participant safety and data integrity are worthy. However, it has been argued that these GCP guidelines are based on consensus of expert opinions and not on evidence. 43 Not surprisingly, the individual standards as well as the concept of standardization of clinical trials of pharmacological interventions are not free of controversy. There have been increasing calls for the guidelines to be made more scientific, up-to-date, flexible and simple through collaborative and evidence-based efforts. 43, 44 For trials in developing countries, rigid adherence to GCP standards as they are now formulated, have both positive and unfavorable consequences. 45 The complexity and expense of clinical trials has risen rapidly in recent years. A portion of this increased expense arises from the extensive documentation and auditing requirements demanded by GCP guidelines. This constitutes a disincentive to the clinical testing of drugs and vaccines for diseases mainly affecting developing countries and for which profitable markets are not foreseen. Furthermore, compliance with the stringent GCP requirements often requires the engagement of expensive policy. This may require development of both detailed 'investment cases' , as well as much shorter policy briefs. Presentation of the information in scientific journals is important, but it is also important to arrange presentations of study findings in both small and large meetings attended by policy-makers. The onus of assuring the attendance of policy-makers at such meetings is on the investigators. Finally, because the WHO has the ear of policy-makers, presentation of the findings at relevant WHO meetings, at both the regional and global levels is critical. An illustration of this approach was provided by efforts to synthesize and communicate the evidence on cholera and oral cholera vaccines. Synthesis of these findings into a white paper, and subsequently an investment case, for WHO's Scientific Advisory Group of Experts led to a greatly strengthened recommendation on oral cholera vaccines by WHO, 48 and provided the background for a recent World Health Assembly resolution, recommending the use of vaccines in the public health armamentarium against cholera. 49\\nPublic health research in developing country populations is complicated and challenging, but necessary. When conducted in a well-planned and focused manner, it can yield many important gains including understanding health problems better, informing policy decisions and rational deployment of interventions resulting in large health benefits.\\nWhile the ensemble of evidence generated by the DOMI Program has now become relatively standard in programmes to generate policy-relevant evidence on interventions for developing countries, it is important to note that several additional strategies are helpful if such evidence is to influence or support policy. Programmes to generate evidence should be formulated and conducted in partnership with policy-makers and healthcare professionals in the countries where the intervention is being contemplated for introduction. For example, the DOMI Program was based on an initial systematic survey of policymakers about evidence needs in targeted countries, and the DOMI field research programme was formulated on the basis of these findings. 47 Also, DOMI's multifaceted research programme to generate evidence was implemented in partnership with Ministries of Health in order to ensure that the decisionmakers would have a sense of ownership of the findings. Moreover, if the evidence is to have an impact at the regional and global levels, as well as the national level, it may be helpful to construct multi-country programmes of research, with study designs and procedures standardized across countries. As shown in Table 6 .2, the DOMI Program employed a standardized, multi-country approach to its studies, to generate evidence to inform policy on the introduction of killed oral cholera vaccines. This led to an evidence base that provided interpretable comparative data across countries.\\nBeyond generating the data, packaging and presentation of the evidence in a way that is convincing for policy-makers is of critical importance if the evidence is to have an impact on\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8997,\n        \"samples\": [\n          \"Tandoc, Edson C.  Kim, Hye Kyung\",\n          \" Ghelichkhani, Samereh.  Jenabi, Ensiyeh. <br>Jalili, Ebrahim.  Alishirzad, Azam.  Shahbazi, Fatemeh\",\n          \" Pujari, Rathin.  Chan, Geoffrey.  Tapply, Ian. <br>Bourne, Rupert R.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9022,\n        \"samples\": [\n          \" A lung rescue team improves survival in obesity<br>with acute respiratory distress syndrome\",\n          \" Associations between Changes in Health<br>Behaviours and Body Weight during the COVID-19 Quarantine<br>in Lithuania: The Lithuanian COVIDiet Study\",\n          \" Issues and Challenges of Public-Health<br>Research in Developing Countries\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"journal\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2785,\n        \"samples\": [\n          \"CJEM\",\n          \"World J Urol\",\n          \"Encyclopedia of Virology\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract_summary\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 7281,\n        \"samples\": [\n          \" This study projects the economic impacts of<br>COVID-19 outbreak on the Brazilian economy using a<br>dynamic interregional computable general<br>equilibrium model. We considered two scenarios. The first<br>scenario has two channels of transmission over the<br>economic system: a negative shock of labor supply due to<br>the rates of morbidity and mortality caused by the<br>pandemic, and a temporary shutdown of nonessential<br>economic activities. The second scenario adds to the<br>first the effects of the government fiscal package<br>adopted to counteract the effects of COVID-19 on the<br>economy. Furthermore, in both scenarios, a sensitive<br>analysis related to the temporality of the...\",\n          \" The management of comprehensive and<br>extensive, multi-lateral and multi-national programmes<br>and projects in the field of coastal and ocean<br>management poses numerous organisational problems<br>encompassing co-ordination between: the actions of<br>individual participating countries at the regional<br>level; the national level actions of institutions<br>from different sectors; and actions that are<br>designed to address issues as diverse as: biological<br>diversity conservation and sustainable use; fisheries<br>management; maritime transport; and the control of land<br>based pollution. Most large multi-lateral projects<br>focus on sound scientific knowledge and<br>information, and pay less attention to the design of a<br>management structure that will ensure coherence...\",\n          \" Citation: Lytton, S.D.; Yeasmin, M.; Ghosh,<br>A.K.; Bulbul, M.R.H.; Molla, M.M.A.; Herr, M.;<br>Duchmann, H.; Sharif, M.M.; Nafisa, T.; Amin, M.R.; et al.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 152
        }
      ],
      "source": [
        "from tqdm import tqdm\n",
        "\n",
        "def get_breaks(content, length):\n",
        "    data = \"\"\n",
        "    words = content.split(' ')\n",
        "    total_chars = 0\n",
        "\n",
        "    # add break every length characters\n",
        "    for i in range(len(words)):\n",
        "        total_chars += len(words[i])\n",
        "        if total_chars > length:\n",
        "            data = data + \"<br>\" + words[i]\n",
        "            total_chars = 0\n",
        "        else:\n",
        "            data = data + \" \" + words[i]\n",
        "    return data\n",
        "\n",
        "\n",
        "dict_ = {'paper_id': [], 'doi':[], 'abstract': [], 'body_text': [],\n",
        "         'authors': [], 'title': [], 'journal': [], 'abstract_summary': []}\n",
        "\n",
        "\n",
        "for idx, entry in tqdm(enumerate(json_files), total=len(json_files)):\n",
        "    try:\n",
        "        content = FileReader(entry)\n",
        "    except Exception as e:\n",
        "        continue  # invalid paper format, skip\n",
        "\n",
        "    # get metadata information\n",
        "    meta_data = meta_df.loc[meta_df['sha'] == content.paper_id]\n",
        "    # no metadata, skip this paper\n",
        "    if len(meta_data) == 0:\n",
        "        continue\n",
        "    if len(content.body_text) == 0:\n",
        "        continue\n",
        "    dict_['abstract'].append(content.abstract)\n",
        "    dict_['paper_id'].append(content.paper_id)\n",
        "    dict_['body_text'].append(content.body_text)\n",
        "    # also create a column for the summary of abstract to be used in a plot\n",
        "    if len(content.abstract) == 0:\n",
        "        # no abstract provided\n",
        "        dict_['abstract_summary'].append(\"Not provided.\")\n",
        "    elif len(content.abstract.split(' ')) > 100:\n",
        "        # abstract provided is too long for plot, take first 300 words append with ...\n",
        "        info = content.abstract.split(' ')[:100]\n",
        "        summary = get_breaks(' '.join(info), 40)\n",
        "        dict_['abstract_summary'].append(summary + \"...\")\n",
        "    else:\n",
        "        # abstract is short enough\n",
        "        summary = get_breaks(content.abstract, 40)\n",
        "        dict_['abstract_summary'].append(summary)\n",
        "\n",
        "    # get metadata information\n",
        "    meta_data = meta_df.loc[meta_df['sha'] == content.paper_id]\n",
        "\n",
        "    try:\n",
        "        # if more than one author\n",
        "        authors = meta_data['authors'].values[0].split(';')\n",
        "        if len(authors) > 2:\n",
        "            # more than 2 authors, may be problem when plotting, so take first 2 append with ...\n",
        "            dict_['authors'].append(get_breaks('. '.join(authors), 40))\n",
        "        else:\n",
        "            # authors will fit in plot\n",
        "            dict_['authors'].append(\". \".join(authors))\n",
        "    except Exception as e:\n",
        "        # if only one author - or Null valie\n",
        "        dict_['authors'].append(meta_data['authors'].values[0])\n",
        "\n",
        "    # add the title information, add breaks when needed\n",
        "    try:\n",
        "        title = get_breaks(meta_data['title'].values[0], 40)\n",
        "        dict_['title'].append(title)\n",
        "    # if title was not provided\n",
        "    except Exception as e:\n",
        "        dict_['title'].append(meta_data['title'].values[0])\n",
        "\n",
        "    # add the journal information\n",
        "    dict_['journal'].append(meta_data['journal'].values[0])\n",
        "\n",
        "    # add doi\n",
        "    dict_['doi'].append(meta_data['doi'].values[0])\n",
        "\n",
        "\n",
        "df_covid = pd.DataFrame(dict_, columns=['paper_id', 'doi', 'abstract', 'body_text',\n",
        "                                        'authors', 'title', 'journal', 'abstract_summary'])\n",
        "df_covid.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "2c2974a3",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2c2974a3",
        "outputId": "9088b3c1-8cb4-466d-b577-34c72b4f2f9d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 9022 entries, 0 to 9021\n",
            "Data columns (total 8 columns):\n",
            " #   Column            Non-Null Count  Dtype \n",
            "---  ------            --------------  ----- \n",
            " 0   paper_id          9022 non-null   object\n",
            " 1   doi               8599 non-null   object\n",
            " 2   abstract          9022 non-null   object\n",
            " 3   body_text         9022 non-null   object\n",
            " 4   authors           9009 non-null   object\n",
            " 5   title             9022 non-null   object\n",
            " 6   journal           8095 non-null   object\n",
            " 7   abstract_summary  9022 non-null   object\n",
            "dtypes: object(8)\n",
            "memory usage: 564.0+ KB\n"
          ]
        }
      ],
      "source": [
        "df_covid.info()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "4f9af672",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4f9af672",
        "outputId": "29653356-9572-4985-f863-3c059c58ed44"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "Index: 8070 entries, 0 to 9021\n",
            "Data columns (total 8 columns):\n",
            " #   Column            Non-Null Count  Dtype \n",
            "---  ------            --------------  ----- \n",
            " 0   paper_id          8070 non-null   object\n",
            " 1   doi               8070 non-null   object\n",
            " 2   abstract          8070 non-null   object\n",
            " 3   body_text         8070 non-null   object\n",
            " 4   authors           8070 non-null   object\n",
            " 5   title             8070 non-null   object\n",
            " 6   journal           8070 non-null   object\n",
            " 7   abstract_summary  8070 non-null   object\n",
            "dtypes: object(8)\n",
            "memory usage: 567.4+ KB\n"
          ]
        }
      ],
      "source": [
        "df = df_covid\n",
        "df.dropna(inplace=True)\n",
        "df.info()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "89d3b3b8",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "89d3b3b8",
        "outputId": "5783d43e-9e5a-4dfe-a53e-f3c503fe473d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: langdetect in /usr/local/lib/python3.10/dist-packages (1.0.9)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from langdetect) (1.16.0)\n"
          ]
        }
      ],
      "source": [
        "!pip install langdetect"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "f2eacd72",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f2eacd72",
        "outputId": "be6ed681-e483-472c-f98f-befdc1f9309b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "100%|██████████| 8070/8070 [01:06<00:00, 121.95it/s]\n"
          ]
        }
      ],
      "source": [
        "from tqdm import tqdm\n",
        "from langdetect import detect\n",
        "from langdetect import DetectorFactory\n",
        "\n",
        "# set seed\n",
        "DetectorFactory.seed = 0\n",
        "\n",
        "# hold label - language\n",
        "languages = []\n",
        "\n",
        "# go through each text\n",
        "for ii in tqdm(range(0,len(df))):\n",
        "    # split by space into list, take the first x intex, join with space\n",
        "    text = df.iloc[ii]['body_text'].split(\" \")\n",
        "\n",
        "    lang = \"en\"\n",
        "    try:\n",
        "        if len(text) > 50:\n",
        "            lang = detect(\" \".join(text[:50]))\n",
        "        elif len(text) > 0:\n",
        "            lang = detect(\" \".join(text[:len(text)]))\n",
        "    # ught... beginning of the document was not in a good format\n",
        "    except Exception as e:\n",
        "        all_words = set(text)\n",
        "        try:\n",
        "            lang = detect(\" \".join(all_words))\n",
        "        # what!! :( let's see if we can find any text in abstract...\n",
        "        except Exception as e:\n",
        "\n",
        "            try:\n",
        "                # let's try to label it through the abstract then\n",
        "                lang = detect(df.iloc[ii]['abstract_summary'])\n",
        "            except Exception as e:\n",
        "                lang = \"unknown\"\n",
        "                pass\n",
        "\n",
        "    # get the language\n",
        "    languages.append(lang)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "7e7396f0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7e7396f0",
        "outputId": "7710ee2c-73d2-4cbb-e5c3-8c19930750fd"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Total: 8070\n",
            "\n",
            "{'de': 20, 'en': 8028, 'es': 13, 'fr': 6, 'id': 1, 'nl': 2}\n"
          ]
        }
      ],
      "source": [
        "from pprint import pprint\n",
        "\n",
        "languages_dict = {}\n",
        "for lang in set(languages):\n",
        "    languages_dict[lang] = languages.count(lang)\n",
        "\n",
        "print(\"Total: {}\\n\".format(len(languages)))\n",
        "pprint(languages_dict)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "e667fa35",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "e667fa35",
        "outputId": "0946d5e6-9b37-4523-9cc7-92b195e2815d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "Index: 8028 entries, 0 to 9021\n",
            "Data columns (total 9 columns):\n",
            " #   Column            Non-Null Count  Dtype \n",
            "---  ------            --------------  ----- \n",
            " 0   paper_id          8028 non-null   object\n",
            " 1   doi               8028 non-null   object\n",
            " 2   abstract          8028 non-null   object\n",
            " 3   body_text         8028 non-null   object\n",
            " 4   authors           8028 non-null   object\n",
            " 5   title             8028 non-null   object\n",
            " 6   journal           8028 non-null   object\n",
            " 7   abstract_summary  8028 non-null   object\n",
            " 8   language          8028 non-null   object\n",
            "dtypes: object(9)\n",
            "memory usage: 627.2+ KB\n"
          ]
        }
      ],
      "source": [
        "df['language'] = languages\n",
        "df = df[df['language'] == 'en']\n",
        "df.info()"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "ad4bf462",
      "metadata": {
        "id": "ad4bf462"
      },
      "source": [
        "# Histogram of year / journal"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "5b707bb3",
      "metadata": {
        "id": "5b707bb3"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "id": "cae116c4",
      "metadata": {
        "id": "cae116c4"
      },
      "source": [
        "# Map-Reduce"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import csv\n",
        "import re\n",
        "from collections import defaultdict\n",
        "from functools import reduce\n",
        "\n",
        "# Sample data: list of documents (each document is a string)\n",
        "documents_MR = []\n",
        "with open(metadata_path, 'r') as csvfile:\n",
        "    reader = csv.reader(csvfile)\n",
        "    for row in reader:\n",
        "        documents_MR.append(row[4])  # Assuming the 4th column is at index 3 (0-based indexing)\n",
        "        documents_MR.append(row[9])  # Assuming the 9th column is at index 4 (0-based indexing)\n",
        "\n",
        "\n",
        "# Specify the words to include in the stoplist\n",
        "stop_words_upper = [\"A\", \"After\", \"Also\", \"An\", \"And\", \"As\", \"At\", \"Base\", \"Between\", \"But\", \"By\", \"Did\", \"Do\", \"Does\", \"During\", \"For\", \"From\", \"Have\", \"Has\", \"Had\", \"However\",\n",
        "                    \"In\", \"Is\", \"It\", \"Its\", \"New\", \"Of\", \"On\", \"Or\", \"That\", \"The\", \"To\", \"Use\", \"Using\", \"Was\", \"We\", \"Were\", \"Which\", \"With\", \"=\"]\n",
        "stop_words_lower = [word.lower() for word in stop_words_upper]\n",
        "\n",
        "# Mapper function\n",
        "def map_function(documents_MR):\n",
        "    # Split the document into words and assign an ID to each word\n",
        "    for idx, word in enumerate(documents_MR.split(), start=1):\n",
        "      yield (word.lower(), 1)\n",
        "\n",
        "# Step 1: Map phase\n",
        "mapped = []\n",
        "for documents_MR in documents_MR:\n",
        "    mapped.extend(map_function(documents_MR))\n",
        "\n",
        "# Step 2: Shuffle and sort phase (group by key)\n",
        "shuffled = defaultdict(list)\n",
        "for word, count in mapped:\n",
        "    shuffled[word].append(count)\n",
        "\n",
        "# Step 3: Reduce phase\n",
        "reduced = {}\n",
        "for word, counts in shuffled.items():\n",
        "    reduced[word] = reduce(lambda x, y: x + y, counts)\n",
        "\n",
        "# Step 4: Merge the counts of words with uppercase and lowercase versions together\n",
        "merged_counts = defaultdict(int)\n",
        "for word, count in reduced.items():\n",
        "    merged_word = re.sub(r'[^a-zA-Z0-9]', '', word)\n",
        "    # Remove non-alphanumeric characters\n",
        "    merged_counts[merged_word.lower()] += count\n",
        "\n",
        "# Step 5.1: filter out the stop words\n",
        "filtered = {word: count for word, count in reduced.items() if word not in stop_words_upper and word not in stop_words_lower}\n",
        "\n",
        "# Step 5.2: Sort the filtered words by count in descending order\n",
        "sorted_words = sorted(filtered.items(), key=lambda x: x[1], reverse=True)\n",
        "sorted_words_unfilter = sorted(reduced.items(), key=lambda x: x[1], reverse=True)\n",
        "\n",
        "# Step 5.3: Get the top 50 prevalent words\n",
        "top50_filtered = sorted_words[:50]\n",
        "top50_unfilter = sorted_words_unfilter[:80]\n",
        "\n",
        "\n",
        "# Step 6.1: Print the table of the top 50 prevalent words without stop words list\n",
        "print(\"Top 80 Prevalent Words without stopwords:\")\n",
        "print(f\"{'ID':<4}{'Word':<20}{'Count'}\")\n",
        "print(\"-----------------------------\")\n",
        "for idx, (word, count) in enumerate(top50_unfilter, start=1):\n",
        "    print(f\"{idx:<4}{word:<20}{count}\")\n",
        "print(\"\\n\"+\"\\n\")\n",
        "\n",
        "# Step 6.2: Print the table of the top 50 prevalent words with IDs\n",
        "print(\"Top 50 Prevalent Words:\")\n",
        "print(f\"{'ID':<4}{'Word':<20}{'Count'}\")\n",
        "print(\"-----------------------------\")\n",
        "for idx, (word, count) in enumerate(top50_filtered, start=1):\n",
        "    print(f\"{idx:<4}{word:<20}{count}\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "p1MWD1KHOv-5",
        "outputId": "7b0823ea-7ca8-4090-ea4b-e78b822f73f0"
      },
      "id": "p1MWD1KHOv-5",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Top 80 Prevalent Words without stopwords:\n",
            "ID  Word                Count\n",
            "-----------------------------\n",
            "1   the                 97935\n",
            "2   of                  75152\n",
            "3   and                 71142\n",
            "4   in                  46376\n",
            "5   to                  40334\n",
            "6   a                   30698\n",
            "7   with                20595\n",
            "8   for                 19646\n",
            "9   is                  12786\n",
            "10  were                12217\n",
            "11  that                11841\n",
            "12  on                  11311\n",
            "13  was                 11006\n",
            "14  covid-19            10343\n",
            "15  this                10102\n",
            "16  as                  10067\n",
            "17  we                  9942\n",
            "18  by                  9331\n",
            "19  from                8221\n",
            "20  are                 7789\n",
            "21  patients            7203\n",
            "22  an                  7006\n",
            "23  be                  6202\n",
            "24  study               5251\n",
            "25  or                  5157\n",
            "26  at                  5110\n",
            "27  have                4582\n",
            "28  during              4384\n",
            "29  health              4244\n",
            "30  has                 4039\n",
            "31  these               3862\n",
            "32  which               3722\n",
            "33  using               3694\n",
            "34  between             3616\n",
            "35  data                3600\n",
            "36  their               3555\n",
            "37  can                 3487\n",
            "38  disease             3458\n",
            "39  =                   3426\n",
            "40  sars-cov-2          3371\n",
            "41  not                 3184\n",
            "42  our                 3003\n",
            "43  clinical            2962\n",
            "44  been                2925\n",
            "45  pandemic            2923\n",
            "46  more                2844\n",
            "47  it                  2831\n",
            "48  infection           2803\n",
            "49  may                 2605\n",
            "50  also                2592\n",
            "51  used                2560\n",
            "52  among               2535\n",
            "53  respiratory         2490\n",
            "54  results             2486\n",
            "55  associated          2484\n",
            "56  coronavirus         2449\n",
            "57  risk                2438\n",
            "58  virus               2395\n",
            "59  analysis            2342\n",
            "60  all                 2340\n",
            "61  after               2298\n",
            "62  than                2267\n",
            "63  most                2246\n",
            "64  viral               2192\n",
            "65  other               2187\n",
            "66  had                 2183\n",
            "67  such                2182\n",
            "68  use                 2136\n",
            "69  model               2127\n",
            "70  severe              2101\n",
            "71  but                 2037\n",
            "72  two                 2019\n",
            "73  results:            1971\n",
            "74  care                1970\n",
            "75  its                 1968\n",
            "76  different           1937\n",
            "77  new                 1911\n",
            "78  human               1886\n",
            "79  there               1876\n",
            "80  social              1876\n",
            "\n",
            "\n",
            "\n",
            "Top 50 Prevalent Words:\n",
            "ID  Word                Count\n",
            "-----------------------------\n",
            "1   covid-19            10343\n",
            "2   this                10102\n",
            "3   are                 7789\n",
            "4   patients            7203\n",
            "5   be                  6202\n",
            "6   study               5251\n",
            "7   health              4244\n",
            "8   these               3862\n",
            "9   data                3600\n",
            "10  their               3555\n",
            "11  can                 3487\n",
            "12  disease             3458\n",
            "13  sars-cov-2          3371\n",
            "14  not                 3184\n",
            "15  our                 3003\n",
            "16  clinical            2962\n",
            "17  been                2925\n",
            "18  pandemic            2923\n",
            "19  more                2844\n",
            "20  infection           2803\n",
            "21  may                 2605\n",
            "22  used                2560\n",
            "23  among               2535\n",
            "24  respiratory         2490\n",
            "25  results             2486\n",
            "26  associated          2484\n",
            "27  coronavirus         2449\n",
            "28  risk                2438\n",
            "29  virus               2395\n",
            "30  analysis            2342\n",
            "31  all                 2340\n",
            "32  than                2267\n",
            "33  most                2246\n",
            "34  viral               2192\n",
            "35  other               2187\n",
            "36  such                2182\n",
            "37  model               2127\n",
            "38  severe              2101\n",
            "39  two                 2019\n",
            "40  results:            1971\n",
            "41  care                1970\n",
            "42  different           1937\n",
            "43  human               1886\n",
            "44  there               1876\n",
            "45  social              1876\n",
            "46  both                1851\n",
            "47  treatment           1847\n",
            "48  however,            1844\n",
            "49  based               1820\n",
            "50  research            1806\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "id": "ac547210",
      "metadata": {
        "id": "ac547210"
      },
      "source": [
        "# Association Analysis"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "f3b12c03",
      "metadata": {
        "id": "f3b12c03"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "id": "48ec6f2e",
      "metadata": {
        "id": "48ec6f2e"
      },
      "source": [
        "# Similarity Analysis"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "4823d5b4",
      "metadata": {
        "id": "4823d5b4"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "id": "adc2e25a",
      "metadata": {
        "id": "adc2e25a"
      },
      "source": [
        "# Clustering Analysis"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.11.7"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}